---
output: html_document
editor_options:
  chunk_output_type: console
css: highlight.css
---

```{r setup, include=FALSE}
knitr::knit_hooks$set(source = function(x, options) {
  if (!is.null(options$hilang)) {
      code_open <- "\n\n<div class=\"sourceCode\">\n<pre class=\"sourceCode\">\n<code class=\"sourceCode\">"
      code_close <- "\n</code>\n</pre>\n</div>\n"
      code_body <- highr::hi_andre(x, language = options$hilang, format = "html")
    stringr::str_c(
      code_open,
      knitr:::indent_block(paste(code_body, collapse = '\n'), ""),
      code_close
    )
  } else {
    stringr::str_c("\n\n```", tolower(options$engine), "\n",
                   paste(x, collapse = '\n'), "\n```\n\n")

  }
})
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = '../')
```


```{r, include=FALSE}
source('R/draw_workflow.R')
prod_recombination <- tibbler(c('step3_ch'))
```

# Analysis VII (Mokeyflowers)

## Summary

The genome wide average differentiation of the study by Stankowski *et al.* (2019) is computed within the  [**nextflow**](https://www.nextflow.io/) script `analysis_stankowski_etal_2019.nf` (located under `$BASE_DIR/nf/analysis_stankowski_etal_2019/`).
(The <span style="color:#4DAF4A">green dot</span> indicates the genotype input, <span style="color:#E41A1C">red arrows</span> depict output that is exported for further use.)

<div style="max-width:800px; margin:auto;">
```{r, echo = FALSE, warning = FALSE, message = FALSE}
girafe( ggobj = dot_plot(file = 'docs/analysis_recombination.dot', git = 6, point_types = prod_recombination),
        width_svg = 14, height_svg = 14)

```
</div>

## Details of `analysis_stankowski_etal_2019.nf`

### Data import

The nextflow script starts by opening the genotype data and the assignment of samples to populations.

<div class="kclass">
```{r , eval = FALSE, hilang = 'nf'}
#!/usr/bin/env nextflow
// This pipelie includes the re-analysis of the monkeyflower differentiation
// git 10.1
// open genotype data
Channel
	.fromFilePairs("../../ressources/other_studies/all_9_taxa_G1_vars.bgziped.vcf.{gz,gz.tbi}")
	.set{ vcf_fl }

// git 10.2
// open population assignment for the samples
Channel
	.fromPath("../../ressources/other_studies/stankowski_etal_2019_spec_short.tsv")
	.set{ pop_fl }
}
```
</div>

Next, we crate an inventory of all populations within the genotype file and create a cross of all possible combinations.

<div class="kclass">
```{r , eval = FALSE, hilang = 'nf'}
	// git 10.3
	// define sepcies set
Channel.from( [[1, "ari"], [2, "aur"], [3, "cal"], [4, "cle"], [5, "gra"], [6, "lon"], [7, "par"], [8, "pur"], [9, "puy"]] ).into{ specs_ch1; specs_ch2 }

// git 10.4
// create all possible species pairs
all_fst_pairs_ch = pop_fl
	.combine(specs_ch1)
	.combine(specs_ch2)
	.filter{ it[1] < it[3] }
	.map{ it[0,2,4]}
	.combine(vcf_fl)
```
</div>

For every species pair, we compute the genome wide differentiation using `VCFtools`.

<div class="kclass">
```{r , eval = FALSE, hilang = 'nf'}
// git 10.5
// comute pairwise fsts
process fst_run {
	label 'L_32g2h_fst_run'
	publishDir "../../ressources/other_studies/stankowski_logs/", mode: 'copy'

	input:
	set file( pop ), val( spec1 ), val( spec2 ), file( vcf_idx ), file( vcf ) from all_fst_pairs_ch

	output:
	file( "${spec1}-${spec2}.log" ) into fst_logs

	script:
	"""
	grep ${spec1} ${pop} > pop1.txt
	grep ${spec2} ${pop} > pop2.txt

	vcftools --gzvcf ${vcf[0]} \
	   --weir-fst-pop pop1.txt \
	   --weir-fst-pop pop2.txt \
	   --out ${spec1}-${spec2} 2> ${spec1}-${spec2}.log
	"""
}
```
</div>

Finally we collect all `VCFtools` log files, extract the genome wide average *F<sub>ST</sub>* value and compile a summary table.

<div class="kclass">
```{r , eval = FALSE, hilang = 'nf'}
// git 10.6
// collect the genome wide average for all species pairs
process fst_collect {
	label 'L_loc_fst_collect'
	publishDir "../../ressources/other_studies/", mode: 'copy'

	input:
	set file( log ) from fst_logs.collect()

	output:
	file( "stankowski_etal_2019.tsv" ) into fst_summary

	script:
	"""
	grep "mean Fst estimate" \$BASE_DIR/ressources/other_studies/stankowski_logs/*.log | \
	  sed "s/.log:Weir and Cockerham mean Fst estimate: /\\t/; s=^\$BASE_DIR/ressources/other_studies/stankowski_logs/==" \
		> stankowski_etal_2019.tsv
	"""
}
```
</div>

The summary table can now be used to create supplementary Figure 1.

---
