<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>17 Figure 5 | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="17 Figure 5 | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="17 Figure 5 | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench" />


<meta name="date" content="2020-06-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-4.html"/>
<link rel="next" href="supplementary-figure-1.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a><ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a><ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a><ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a><ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a><ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a><ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i> &amp; pi)</a><ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a><ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (phylogeny &amp; topology weighting)</a><ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a><ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a><ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a><ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (poptrees)</a><ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#details-of-analysis_fst_poptree.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_fst_poptree.nf</code></a><ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#data-preparation-4"><i class="fa fa-check"></i><b>8.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a><ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a><ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-5"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a><ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a><ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-6"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a><ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a><ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-7"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-visualization.html"><a href="git-11-visualization.html"><i class="fa fa-check"></i><b>12</b> (git 11) Visualization</a></li>
<li class="chapter" data-level="13" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>13</b> Figure 1</a><ul>
<li class="chapter" data-level="13.1" data-path="figure-1.html"><a href="figure-1.html#summary-10"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>13.2</b> Details of <code>plot_F1.R</code></a><ul>
<li class="chapter" data-level="13.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>13.2.1</b> Config</a></li>
<li class="chapter" data-level="13.2.2" data-path="figure-1.html"><a href="figure-1.html#data-import"><i class="fa fa-check"></i><b>13.2.2</b> Data import</a></li>
<li class="chapter" data-level="13.2.3" data-path="figure-1.html"><a href="figure-1.html#plotting"><i class="fa fa-check"></i><b>13.2.3</b> Plotting</a></li>
<li class="chapter" data-level="13.2.4" data-path="figure-1.html"><a href="figure-1.html#table-export"><i class="fa fa-check"></i><b>13.2.4</b> Table export</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>14</b> Figure 2</a><ul>
<li class="chapter" data-level="14.1" data-path="figure-2.html"><a href="figure-2.html#summary-11"><i class="fa fa-check"></i><b>14.1</b> Summary</a></li>
<li class="chapter" data-level="14.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>14.2</b> Details of <code>plot_F2.R</code></a><ul>
<li class="chapter" data-level="14.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>14.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>15</b> Figure 3</a><ul>
<li class="chapter" data-level="15.1" data-path="figure-3.html"><a href="figure-3.html#summary-12"><i class="fa fa-check"></i><b>15.1</b> Summary</a></li>
<li class="chapter" data-level="15.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>15.2</b> Details of <code>plot_F3.R</code></a><ul>
<li class="chapter" data-level="15.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>15.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>16</b> Figure 4</a><ul>
<li class="chapter" data-level="16.1" data-path="figure-4.html"><a href="figure-4.html#summary-13"><i class="fa fa-check"></i><b>16.1</b> Summary</a></li>
<li class="chapter" data-level="16.2" data-path="figure-4.html"><a href="figure-4.html#details-of-plot_f4.r"><i class="fa fa-check"></i><b>16.2</b> Details of <code>plot_F4.R</code></a><ul>
<li class="chapter" data-level="16.2.1" data-path="figure-4.html"><a href="figure-4.html#config-3"><i class="fa fa-check"></i><b>16.2.1</b> Config</a></li>
<li class="chapter" data-level="16.2.2" data-path="figure-4.html"><a href="figure-4.html#data-import-1"><i class="fa fa-check"></i><b>16.2.2</b> Data import</a></li>
<li class="chapter" data-level="16.2.3" data-path="figure-4.html"><a href="figure-4.html#plotting-1"><i class="fa fa-check"></i><b>16.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>17</b> Figure 5</a><ul>
<li class="chapter" data-level="17.1" data-path="figure-5.html"><a href="figure-5.html#summary-14"><i class="fa fa-check"></i><b>17.1</b> Summary</a></li>
<li class="chapter" data-level="17.2" data-path="figure-5.html"><a href="figure-5.html#details-of-plot_f5.r"><i class="fa fa-check"></i><b>17.2</b> Details of <code>plot_F5.R</code></a><ul>
<li class="chapter" data-level="17.2.1" data-path="figure-5.html"><a href="figure-5.html#config-4"><i class="fa fa-check"></i><b>17.2.1</b> Config</a></li>
<li class="chapter" data-level="17.2.2" data-path="figure-5.html"><a href="figure-5.html#data-import-2"><i class="fa fa-check"></i><b>17.2.2</b> Data import</a></li>
<li class="chapter" data-level="17.2.3" data-path="figure-5.html"><a href="figure-5.html#plotting-2"><i class="fa fa-check"></i><b>17.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html"><i class="fa fa-check"></i><b>18</b> Supplementary Figure 1</a><ul>
<li class="chapter" data-level="18.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#summary-15"><i class="fa fa-check"></i><b>18.1</b> Summary</a></li>
<li class="chapter" data-level="18.2" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#details-of-plot_sf1.r"><i class="fa fa-check"></i><b>18.2</b> Details of <code>plot_SF1.R</code></a><ul>
<li class="chapter" data-level="18.2.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#config-5"><i class="fa fa-check"></i><b>18.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html"><i class="fa fa-check"></i><b>19</b> Supplementary Figure 2</a><ul>
<li class="chapter" data-level="19.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#summary-16"><i class="fa fa-check"></i><b>19.1</b> Summary</a></li>
<li class="chapter" data-level="19.2" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#details-of-plot_sf2.r"><i class="fa fa-check"></i><b>19.2</b> Details of <code>plot_SF2.R</code></a><ul>
<li class="chapter" data-level="19.2.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#config-6"><i class="fa fa-check"></i><b>19.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html"><i class="fa fa-check"></i><b>20</b> Supplementary Figure 3</a><ul>
<li class="chapter" data-level="20.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#summary-17"><i class="fa fa-check"></i><b>20.1</b> Summary</a></li>
<li class="chapter" data-level="20.2" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#details-of-plot_sf3.r"><i class="fa fa-check"></i><b>20.2</b> Details of <code>plot_SF3.R</code></a><ul>
<li class="chapter" data-level="20.2.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#config-7"><i class="fa fa-check"></i><b>20.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html"><i class="fa fa-check"></i><b>21</b> Supplementary Figure 4</a><ul>
<li class="chapter" data-level="21.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#summary-18"><i class="fa fa-check"></i><b>21.1</b> Summary</a></li>
<li class="chapter" data-level="21.2" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#details-of-plot_sf4.r"><i class="fa fa-check"></i><b>21.2</b> Details of <code>plot_SF4.R</code></a><ul>
<li class="chapter" data-level="21.2.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#config-8"><i class="fa fa-check"></i><b>21.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html"><i class="fa fa-check"></i><b>22</b> Supplementary Figure 5</a><ul>
<li class="chapter" data-level="22.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#summary-19"><i class="fa fa-check"></i><b>22.1</b> Summary</a></li>
<li class="chapter" data-level="22.2" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#details-of-plot_sf5.r"><i class="fa fa-check"></i><b>22.2</b> Details of <code>plot_SF5.R</code></a><ul>
<li class="chapter" data-level="22.2.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#config-9"><i class="fa fa-check"></i><b>22.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html"><i class="fa fa-check"></i><b>23</b> Supplementary Figure 6</a><ul>
<li class="chapter" data-level="23.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#summary-20"><i class="fa fa-check"></i><b>23.1</b> Summary</a></li>
<li class="chapter" data-level="23.2" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#details-of-plot_sf6.r"><i class="fa fa-check"></i><b>23.2</b> Details of <code>plot_SF6.R</code></a><ul>
<li class="chapter" data-level="23.2.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#config-10"><i class="fa fa-check"></i><b>23.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html"><i class="fa fa-check"></i><b>24</b> Supplementary Figure 7</a><ul>
<li class="chapter" data-level="24.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#summary-21"><i class="fa fa-check"></i><b>24.1</b> Summary</a></li>
<li class="chapter" data-level="24.2" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#details-of-plot_sf7.r"><i class="fa fa-check"></i><b>24.2</b> Details of <code>plot_SF7.R</code></a><ul>
<li class="chapter" data-level="24.2.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#config-11"><i class="fa fa-check"></i><b>24.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html"><i class="fa fa-check"></i><b>25</b> Supplementary Figure 8</a><ul>
<li class="chapter" data-level="25.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#summary-22"><i class="fa fa-check"></i><b>25.1</b> Summary</a></li>
<li class="chapter" data-level="25.2" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#details-of-plot_sf8.r"><i class="fa fa-check"></i><b>25.2</b> Details of <code>plot_SF8.R</code></a><ul>
<li class="chapter" data-level="25.2.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#config-12"><i class="fa fa-check"></i><b>25.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html"><i class="fa fa-check"></i><b>26</b> Supplementary Figure 9</a><ul>
<li class="chapter" data-level="26.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#summary-23"><i class="fa fa-check"></i><b>26.1</b> Summary</a></li>
<li class="chapter" data-level="26.2" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#details-of-plot_sf9.r"><i class="fa fa-check"></i><b>26.2</b> Details of <code>plot_SF9.R</code></a><ul>
<li class="chapter" data-level="26.2.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#config-13"><i class="fa fa-check"></i><b>26.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-5" class="section level1">
<h1><span class="header-section-number">17</span> Figure 5</h1>
<div id="summary-14" class="section level2">
<h2><span class="header-section-number">17.1</span> Summary</h2>
<p>This is the accessory documentation of Figure 5.
The Figure can be recreated by running the <strong>R</strong> script <code>plot_F5.R</code>:</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb76-1"><a href="figure-5.html#cb76-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span></span>
<span id="cb76-2"><a href="figure-5.html#cb76-2"></a></span>
<span id="cb76-3"><a href="figure-5.html#cb76-3"></a><span class="ex">Rscript</span> --vanilla R/fig/plot_F5.R \</span>
<span id="cb76-4"><a href="figure-5.html#cb76-4"></a>  2_analysis/twisst/weights/ ressources/plugin/trees/ \</span>
<span id="cb76-5"><a href="figure-5.html#cb76-5"></a>  https://raw.githubusercontent.com/simonhmartin/twisst/master/plot_twisst.R \</span>
<span id="cb76-6"><a href="figure-5.html#cb76-6"></a>  2_analysis/summaries/fst_outliers_998.tsv 2_analysis/dxy/50k/ \</span>
<span id="cb76-7"><a href="figure-5.html#cb76-7"></a>  2_analysis/fst/50k/ 2_analysis/summaries/fst_globals.txt \</span>
<span id="cb76-8"><a href="figure-5.html#cb76-8"></a>  2_analysis/GxP/50000/ 200 5 2_analysis/fst/poptree/summary/</span></code></pre></div>
</div>
<div id="details-of-plot_f5.r" class="section level2">
<h2><span class="header-section-number">17.2</span> Details of <code>plot_F5.R</code></h2>
<p>In the following, the individual steps of the R script are documented.
It is an executable R script that depends on the accessory R package <a href="https://k-hench.github.io/GenomicOriginsScripts"><strong>GenomicOriginsScripts</strong></a>, as well as on the packages <a href="https://k-hench.github.io/hypoimg"><strong>hypoimg</strong></a>, <a href="https://wilkelab.org/ggtext/"><strong>ggtext</strong></a>, <a href="https://davisvaughan.github.io/furrr/"><strong>furrr</strong></a>, <a href="https://ggraph.data-imaginist.com/"><strong>ggraph</strong></a> and <a href="https://tidygraph.data-imaginist.com/"><strong>tidygraph</strong></a>.</p>
<div id="config-4" class="section level3">
<h3><span class="header-section-number">17.2.1</span> Config</h3>
<p>The scripts start with a header that contains copy &amp; paste templates to execute or debug the script:</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="figure-5.html#cb77-1"></a><span class="co">#!/usr/bin/env Rscript</span></span>
<span id="cb77-2"><a href="figure-5.html#cb77-2"></a><span class="co"># run from terminal:</span></span>
<span id="cb77-3"><a href="figure-5.html#cb77-3"></a><span class="co"># Rscript --vanilla R/fig/plot_F5.R \</span></span>
<span id="cb77-4"><a href="figure-5.html#cb77-4"></a><span class="co">#   2_analysis/twisst/weights/ ressources/plugin/trees/ \</span></span>
<span id="cb77-5"><a href="figure-5.html#cb77-5"></a><span class="co">#   https://raw.githubusercontent.com/simonhmartin/twisst/master/plot_twisst.R \</span></span>
<span id="cb77-6"><a href="figure-5.html#cb77-6"></a><span class="co">#   2_analysis/summaries/fst_outliers_998.tsv 2_analysis/dxy/50k/ \</span></span>
<span id="cb77-7"><a href="figure-5.html#cb77-7"></a><span class="co">#   2_analysis/fst/50k/ 2_analysis/summaries/fst_globals.txt \</span></span>
<span id="cb77-8"><a href="figure-5.html#cb77-8"></a><span class="co">#   2_analysis/GxP/50000/ 200 5 2_analysis/fst/poptree/summary/</span></span>
<span id="cb77-9"><a href="figure-5.html#cb77-9"></a><span class="co"># ===============================================================</span></span>
<span id="cb77-10"><a href="figure-5.html#cb77-10"></a><span class="co"># This script produces Figure 5 of the study &quot;Ancestral variation, hybridization and modularity</span></span>
<span id="cb77-11"><a href="figure-5.html#cb77-11"></a><span class="co"># fuel a marine radiation&quot; by Hench, McMillan and Puebla</span></span>
<span id="cb77-12"><a href="figure-5.html#cb77-12"></a><span class="co"># ---------------------------------------------------------------</span></span>
<span id="cb77-13"><a href="figure-5.html#cb77-13"></a><span class="co"># ===============================================================</span></span>
<span id="cb77-14"><a href="figure-5.html#cb77-14"></a><span class="co"># args &lt;- c(&#39;2_analysis/twisst/weights/&#39;, &#39;ressources/plugin/trees/&#39;,</span></span>
<span id="cb77-15"><a href="figure-5.html#cb77-15"></a><span class="co">#           &#39;https://raw.githubusercontent.com/simonhmartin/twisst/master/plot_twisst.R&#39;,</span></span>
<span id="cb77-16"><a href="figure-5.html#cb77-16"></a><span class="co">#           &#39;2_analysis/summaries/fst_outliers_998.tsv&#39;,</span></span>
<span id="cb77-17"><a href="figure-5.html#cb77-17"></a><span class="co">#           &#39;2_analysis/dxy/50k/&#39;, &#39;2_analysis/fst/50k/&#39;,</span></span>
<span id="cb77-18"><a href="figure-5.html#cb77-18"></a><span class="co">#           &#39;2_analysis/summaries/fst_globals.txt&#39;,</span></span>
<span id="cb77-19"><a href="figure-5.html#cb77-19"></a><span class="co">#           &#39;2_analysis/GxP/50000/&#39;, 200, 5,</span></span>
<span id="cb77-20"><a href="figure-5.html#cb77-20"></a><span class="co">#           &quot;2_analysis/fst/poptree/summary/&quot;)</span></span>
<span id="cb77-21"><a href="figure-5.html#cb77-21"></a><span class="co"># script_name &lt;- &quot;R/fig/plot_F5.R&quot;</span></span></code></pre></div>
<p>The next section processes the input from the command line.
It stores the arguments in the vector <code>args</code>.
The needed R packages are loaded and the script name and the current working directory are stored inside variables (<code>script_name</code>, <code>plot_comment</code>).
This information will later be written into the meta data of the figure to help us tracing back the scripts that created the figures in the future.</p>
<p>Then we drop all the imported information besides the arguments following the script name and print the information to the terminal.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="figure-5.html#cb78-1"></a>args &lt;-<span class="st"> </span><span class="kw">commandArgs</span>(<span class="dt">trailingOnly =</span> <span class="ot">FALSE</span>)</span>
<span id="cb78-2"><a href="figure-5.html#cb78-2"></a><span class="co"># setup -----------------------</span></span>
<span id="cb78-3"><a href="figure-5.html#cb78-3"></a><span class="kw">library</span>(GenomicOriginsScripts)</span>
<span id="cb78-4"><a href="figure-5.html#cb78-4"></a><span class="kw">library</span>(hypoimg)</span>
<span id="cb78-5"><a href="figure-5.html#cb78-5"></a><span class="kw">library</span>(furrr)</span>
<span id="cb78-6"><a href="figure-5.html#cb78-6"></a><span class="kw">library</span>(ggraph)</span>
<span id="cb78-7"><a href="figure-5.html#cb78-7"></a><span class="kw">library</span>(tidygraph)</span>
<span id="cb78-8"><a href="figure-5.html#cb78-8"></a><span class="kw">library</span>(ggtext)</span>
<span id="cb78-9"><a href="figure-5.html#cb78-9"></a><span class="kw">cat</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>)</span>
<span id="cb78-10"><a href="figure-5.html#cb78-10"></a>script_name &lt;-<span class="st"> </span>args[<span class="dv">5</span>] <span class="op">%&gt;%</span></span>
<span id="cb78-11"><a href="figure-5.html#cb78-11"></a><span class="st">  </span><span class="kw">str_remove</span>(.,<span class="st">&#39;--file=&#39;</span>)</span>
<span id="cb78-12"><a href="figure-5.html#cb78-12"></a></span>
<span id="cb78-13"><a href="figure-5.html#cb78-13"></a>plot_comment &lt;-<span class="st"> </span>script_name <span class="op">%&gt;%</span></span>
<span id="cb78-14"><a href="figure-5.html#cb78-14"></a><span class="st">  </span><span class="kw">str_c</span>(<span class="st">&#39;mother-script = &#39;</span>,<span class="kw">getwd</span>(),<span class="st">&#39;/&#39;</span>,.)</span>
<span id="cb78-15"><a href="figure-5.html#cb78-15"></a></span>
<span id="cb78-16"><a href="figure-5.html#cb78-16"></a>args &lt;-<span class="st"> </span><span class="kw">process_input</span>(script_name, args)</span></code></pre></div>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="figure-5.html#cb79-1"></a><span class="co">#&gt; ── Script: scripts/plot_F5.R ────────────────────────────────────────────</span></span>
<span id="cb79-2"><a href="figure-5.html#cb79-2"></a><span class="co">#&gt; Parameters read:</span></span>
<span id="cb79-3"><a href="figure-5.html#cb79-3"></a><span class="co">#&gt; ★ 1: 2_analysis/twisst/weights/</span></span>
<span id="cb79-4"><a href="figure-5.html#cb79-4"></a><span class="co">#&gt; ★ 2: ressources/plugin/trees/</span></span>
<span id="cb79-5"><a href="figure-5.html#cb79-5"></a><span class="co">#&gt; ★ 3: https://raw.githubusercontent.com/simonhmartin/twisst/master/plot_twisst.R</span></span>
<span id="cb79-6"><a href="figure-5.html#cb79-6"></a><span class="co">#&gt; ★ 4: 2_analysis/summaries/fst_outliers_998.tsv</span></span>
<span id="cb79-7"><a href="figure-5.html#cb79-7"></a><span class="co">#&gt; ★ 5: 2_analysis/dxy/50k/</span></span>
<span id="cb79-8"><a href="figure-5.html#cb79-8"></a><span class="co">#&gt; ★ 6: 2_analysis/fst/50k/</span></span>
<span id="cb79-9"><a href="figure-5.html#cb79-9"></a><span class="co">#&gt; ★ 7: 2_analysis/summaries/fst_globals.txt</span></span>
<span id="cb79-10"><a href="figure-5.html#cb79-10"></a><span class="co">#&gt; ★ 8: 2_analysis/GxP/50000/</span></span>
<span id="cb79-11"><a href="figure-5.html#cb79-11"></a><span class="co">#&gt; ★ 9: 200</span></span>
<span id="cb79-12"><a href="figure-5.html#cb79-12"></a><span class="co">#&gt; ★ 10: 5</span></span>
<span id="cb79-13"><a href="figure-5.html#cb79-13"></a><span class="co">#&gt; ★ 11: 2_analysis/fst/poptree/summary/</span></span>
<span id="cb79-14"><a href="figure-5.html#cb79-14"></a><span class="co">#&gt; ─────────────────────────────────────────── /current/working/directory ──</span></span></code></pre></div>
<p>The directories for the different data types are received and stored in respective variables.
Also, we source an external r script from the <a href="https://github.com/simonhmartin/twisst">original twisst github repository</a> that we need to import the twisst data:</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="figure-5.html#cb80-1"></a><span class="co"># config -----------------------</span></span>
<span id="cb80-2"><a href="figure-5.html#cb80-2"></a>w_path &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">1</span>])</span>
<span id="cb80-3"><a href="figure-5.html#cb80-3"></a>d_path &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">2</span>])</span>
<span id="cb80-4"><a href="figure-5.html#cb80-4"></a>twisst_functions &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">3</span>])</span>
<span id="cb80-5"><a href="figure-5.html#cb80-5"></a>out_table &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">4</span>])</span>
<span id="cb80-6"><a href="figure-5.html#cb80-6"></a>dxy_dir &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">5</span>])</span>
<span id="cb80-7"><a href="figure-5.html#cb80-7"></a>fst_dir &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">6</span>])</span>
<span id="cb80-8"><a href="figure-5.html#cb80-8"></a>fst_globals &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">7</span>])</span>
<span id="cb80-9"><a href="figure-5.html#cb80-9"></a>gxp_dir &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">8</span>])</span>
<span id="cb80-10"><a href="figure-5.html#cb80-10"></a>twisst_size &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(args[<span class="dv">9</span>])</span>
<span id="cb80-11"><a href="figure-5.html#cb80-11"></a>resolution &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(args[<span class="dv">10</span>])</span>
<span id="cb80-12"><a href="figure-5.html#cb80-12"></a>fst_summary_path &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">11</span>])</span>
<span id="cb80-13"><a href="figure-5.html#cb80-13"></a><span class="kw">source</span>(twisst_functions, <span class="dt">local =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p>The we define a buffer width.
This is the extent left and right of the <span class="math inline">\(F_{ST}\)</span> outlier windows that is included in the plots.
We also load the R packages <strong>ape</strong> and <strong>igraph</strong> that will help us working with phylogenetic objects (the twisst topologies).</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="figure-5.html#cb81-1"></a><span class="kw">plan</span>(multiprocess)</span>
<span id="cb81-2"><a href="figure-5.html#cb81-2"></a>window_buffer &lt;-<span class="st"> </span><span class="fl">2.5</span><span class="op">*</span><span class="dv">10</span><span class="op">^</span><span class="dv">5</span></span>
<span id="cb81-3"><a href="figure-5.html#cb81-3"></a><span class="co">#-------------------</span></span>
<span id="cb81-4"><a href="figure-5.html#cb81-4"></a><span class="kw">library</span>(ape)</span>
<span id="cb81-5"><a href="figure-5.html#cb81-5"></a><span class="kw">library</span>(igraph)</span></code></pre></div>
</div>
<div id="data-import-2" class="section level3">
<h3><span class="header-section-number">17.2.2</span> Data import</h3>
<p>Then, we start with the data import.
For the figure we are going to need:</p>
<ul>
<li><span class="math inline">\(d_{XY}\)</span> data</li>
<li>genotype <span class="math inline">\(\times\)</span> phenotype data</li>
<li><span class="math inline">\(F_{ST}\)</span> data</li>
<li>topology weighing data</li>
<li>the positions of the genome annotations</li>
<li>the positions of the <span class="math inline">\(F_{ST}\)</span> outlier windows</li>
</ul>
<p>We start by importing <span class="math inline">\(d_{XY}\)</span> by first listing all <span class="math inline">\(d_{XY}\)</span> data files and then iterating the <span class="math inline">\(d_{XY}\)</span> import function over the files.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="figure-5.html#cb82-1"></a><span class="co"># actual script =========================================================</span></span>
<span id="cb82-2"><a href="figure-5.html#cb82-2"></a><span class="co"># locate dxy data files</span></span>
<span id="cb82-3"><a href="figure-5.html#cb82-3"></a>dxy_files &lt;-<span class="st"> </span><span class="kw">dir</span>(dxy_dir, <span class="dt">pattern =</span> <span class="kw">str_c</span>(<span class="st">&#39;dxy.*[a-z]{3}.*.&#39;</span>, resolution ,<span class="st">&#39;0kb-&#39;</span>, resolution ,<span class="st">&#39;kb.tsv.gz&#39;</span>))</span>
<span id="cb82-4"><a href="figure-5.html#cb82-4"></a></span>
<span id="cb82-5"><a href="figure-5.html#cb82-5"></a><span class="co"># import dxy data</span></span>
<span id="cb82-6"><a href="figure-5.html#cb82-6"></a>dxy_data &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">file =</span> <span class="kw">str_c</span>(dxy_dir, dxy_files)) <span class="op">%&gt;%</span></span>
<span id="cb82-7"><a href="figure-5.html#cb82-7"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">pmap_dfr</span>(get_dxy, <span class="dt">kb =</span> <span class="kw">str_c</span>(resolution ,<span class="st">&#39;0kb&#39;</span>))</span></code></pre></div>
<p>Next we iterate the genotype <span class="math inline">\(\times\)</span> phenotype import function over the trait names <em>Bars</em>, <em>Snout</em> and <em>Peduncle</em>.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="figure-5.html#cb83-1"></a><span class="co"># set traits of interest for GxP</span></span>
<span id="cb83-2"><a href="figure-5.html#cb83-2"></a>gxp_traits &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;Bars&#39;</span>, <span class="st">&#39;Snout&#39;</span>, <span class="st">&#39;Peduncle&#39;</span>)</span>
<span id="cb83-3"><a href="figure-5.html#cb83-3"></a></span>
<span id="cb83-4"><a href="figure-5.html#cb83-4"></a><span class="co"># import GxP data</span></span>
<span id="cb83-5"><a href="figure-5.html#cb83-5"></a>gxp_data &lt;-<span class="st"> </span><span class="kw">str_c</span>(gxp_dir,gxp_traits,<span class="st">&#39;.lm.&#39;</span>, resolution ,<span class="st">&#39;0k.&#39;</span>, resolution ,<span class="st">&#39;k.txt.gz&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb83-6"><a href="figure-5.html#cb83-6"></a><span class="st">  </span><span class="kw">future_map_dfr</span>(get_gxp_long, <span class="dt">kb =</span> <span class="dv">50</span>)</span></code></pre></div>
<p>Then, we define two sets of colors - one for the topology highlighting schemes and one for the traits of the genotype <span class="math inline">\(\times\)</span> phenotype association.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="figure-5.html#cb84-1"></a><span class="co"># set topology weighting color scheme</span></span>
<span id="cb84-2"><a href="figure-5.html#cb84-2"></a>twisst_clr &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">Blue =</span> <span class="st">&quot;#0140E5&quot;</span>, <span class="dt">Bars =</span> <span class="st">&quot;#E32210&quot;</span>, <span class="dt">Butter =</span> <span class="st">&quot;#E4E42E&quot;</span>)</span>
<span id="cb84-3"><a href="figure-5.html#cb84-3"></a><span class="co"># set GxP color scheme</span></span>
<span id="cb84-4"><a href="figure-5.html#cb84-4"></a>gxp_clr &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">Bars =</span> <span class="st">&quot;#79009f&quot;</span>, <span class="dt">Snout =</span> <span class="st">&quot;#E48A00&quot;</span>, <span class="dt">Peduncle =</span> <span class="st">&quot;#5B9E2D&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb84-5"><a href="figure-5.html#cb84-5"></a><span class="st">  </span><span class="kw">darken</span>(<span class="dt">factor =</span> <span class="fl">.95</span>) <span class="op">%&gt;%</span></span>
<span id="cb84-6"><a href="figure-5.html#cb84-6"></a><span class="st">  </span><span class="kw">set_names</span>(., <span class="dt">nm =</span> gxp_traits)</span></code></pre></div>
<p>Next, we compute the average genome wide <span class="math inline">\(d_{XY}\)</span> and load the average genome wide <span class="math inline">\(F_{ST}\)</span> values for all 28 pair wise species comparisons.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="figure-5.html#cb85-1"></a><span class="co"># copute genome wide average dxy</span></span>
<span id="cb85-2"><a href="figure-5.html#cb85-2"></a>dxy_globals &lt;-<span class="st"> </span>dxy_data <span class="op">%&gt;%</span></span>
<span id="cb85-3"><a href="figure-5.html#cb85-3"></a><span class="st">  </span><span class="kw">filter</span>(BIN_START <span class="op">%%</span><span class="st"> </span>( resolution <span class="op">*</span><span class="st"> </span><span class="dv">10000</span> ) <span class="op">==</span><span class="st"> </span><span class="dv">1</span> ) <span class="op">%&gt;%</span></span>
<span id="cb85-4"><a href="figure-5.html#cb85-4"></a><span class="st">  </span><span class="kw">group_by</span>( run ) <span class="op">%&gt;%</span></span>
<span id="cb85-5"><a href="figure-5.html#cb85-5"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">mean_global_dxy =</span> <span class="kw">sum</span>(dxy<span class="op">*</span>N_SITES)<span class="op">/</span><span class="kw">sum</span>(N_SITES)) <span class="op">%&gt;%</span></span>
<span id="cb85-6"><a href="figure-5.html#cb85-6"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">run =</span> <span class="kw">fct_reorder</span>(run,mean_global_dxy))</span>
<span id="cb85-7"><a href="figure-5.html#cb85-7"></a></span>
<span id="cb85-8"><a href="figure-5.html#cb85-8"></a><span class="co"># import genome wide average fst</span></span>
<span id="cb85-9"><a href="figure-5.html#cb85-9"></a><span class="co"># and order population pairs by genomewide average fst</span></span>
<span id="cb85-10"><a href="figure-5.html#cb85-10"></a>fst_globals &lt;-<span class="st"> </span>vroom<span class="op">::</span><span class="kw">vroom</span>(fst_globals,<span class="dt">delim =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>,</span>
<span id="cb85-11"><a href="figure-5.html#cb85-11"></a>                        <span class="dt">col_names =</span> <span class="kw">c</span>(<span class="st">&#39;loc&#39;</span>,<span class="st">&#39;run_prep&#39;</span>,<span class="st">&#39;mean_fst&#39;</span>,<span class="st">&#39;weighted_fst&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb85-12"><a href="figure-5.html#cb85-12"></a><span class="st">  </span><span class="kw">separate</span>(run_prep,<span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;pop1&#39;</span>,<span class="st">&#39;pop2&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;-&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb85-13"><a href="figure-5.html#cb85-13"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">run =</span> <span class="kw">str_c</span>(pop1,loc,<span class="st">&#39;-&#39;</span>,pop2,loc),</span>
<span id="cb85-14"><a href="figure-5.html#cb85-14"></a>         <span class="dt">run =</span> <span class="kw">fct_reorder</span>(run,weighted_fst))</span></code></pre></div>
<p>After this, we import <span class="math inline">\(F_{ST}\)</span> by first listing all <span class="math inline">\(F_{ST}\)</span> data files and then iterating the <span class="math inline">\(F_{ST}\)</span> import function over the files.</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="figure-5.html#cb86-1"></a><span class="co"># locate fst data files</span></span>
<span id="cb86-2"><a href="figure-5.html#cb86-2"></a>fst_files &lt;-<span class="st"> </span><span class="kw">dir</span>(fst_dir ,<span class="dt">pattern =</span> <span class="kw">str_c</span>(<span class="st">&#39;.&#39;</span>, resolution ,<span class="st">&#39;0k.windowed.weir.fst.gz&#39;</span>))</span>
<span id="cb86-3"><a href="figure-5.html#cb86-3"></a><span class="co"># import fst data</span></span>
<span id="cb86-4"><a href="figure-5.html#cb86-4"></a>fst_data &lt;-<span class="st"> </span><span class="kw">str_c</span>(fst_dir, fst_files) <span class="op">%&gt;%</span></span>
<span id="cb86-5"><a href="figure-5.html#cb86-5"></a><span class="st">  </span><span class="kw">future_map_dfr</span>(get_fst, <span class="dt">kb =</span> <span class="kw">str_c</span>(resolution ,<span class="st">&#39;0k&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb86-6"><a href="figure-5.html#cb86-6"></a><span class="st">  </span><span class="kw">left_join</span>(dxy_globals) <span class="op">%&gt;%</span></span>
<span id="cb86-7"><a href="figure-5.html#cb86-7"></a><span class="st">  </span><span class="kw">left_join</span>(fst_globals) <span class="op">%&gt;%</span></span>
<span id="cb86-8"><a href="figure-5.html#cb86-8"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">run =</span> <span class="kw">refactor</span>(., fst_globals),</span>
<span id="cb86-9"><a href="figure-5.html#cb86-9"></a>         <span class="dt">BIN_MID =</span> (BIN_START<span class="op">+</span>BIN_END)<span class="op">/</span><span class="dv">2</span>)</span></code></pre></div>
<p>We then add the genome wide averages of <span class="math inline">\(F_{ST}\)</span> and <span class="math inline">\(d_{XY}\)</span> as new columns to the <span class="math inline">\(d_{XY}\)</span> data.
This will be used later for coloring the <span class="math inline">\(d_{XY}\)</span> panel.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="figure-5.html#cb87-1"></a><span class="co"># order dxy population pairs by genomewide average fst</span></span>
<span id="cb87-2"><a href="figure-5.html#cb87-2"></a>dxy_data &lt;-<span class="st"> </span>dxy_data <span class="op">%&gt;%</span></span>
<span id="cb87-3"><a href="figure-5.html#cb87-3"></a><span class="st">  </span><span class="kw">left_join</span>(dxy_globals) <span class="op">%&gt;%</span></span>
<span id="cb87-4"><a href="figure-5.html#cb87-4"></a><span class="st">  </span><span class="kw">left_join</span>(fst_globals) <span class="op">%&gt;%</span></span>
<span id="cb87-5"><a href="figure-5.html#cb87-5"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">run =</span> <span class="kw">refactor</span>(dxy_data, fst_globals),</span>
<span id="cb87-6"><a href="figure-5.html#cb87-6"></a>         <span class="dt">window =</span> <span class="st">&#39;bold(italic(d[xy]))&#39;</span>)</span></code></pre></div>
<p>Then, we summarize the <span class="math inline">\(d_{XY}\)</span> data to compute <span class="math inline">\(\Delta d_{XY}\)</span>.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="figure-5.html#cb88-1"></a><span class="co"># compute delta dxy</span></span>
<span id="cb88-2"><a href="figure-5.html#cb88-2"></a>data_dxy_summary &lt;-<span class="st"> </span>dxy_data <span class="op">%&gt;%</span></span>
<span id="cb88-3"><a href="figure-5.html#cb88-3"></a><span class="st">  </span><span class="kw">group_by</span>(GPOS) <span class="op">%&gt;%</span></span>
<span id="cb88-4"><a href="figure-5.html#cb88-4"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">scaffold =</span> CHROM[<span class="dv">1</span>],</span>
<span id="cb88-5"><a href="figure-5.html#cb88-5"></a>            <span class="dt">start =</span> BIN_START[<span class="dv">1</span>],</span>
<span id="cb88-6"><a href="figure-5.html#cb88-6"></a>            <span class="dt">end =</span> BIN_END[<span class="dv">1</span>],</span>
<span id="cb88-7"><a href="figure-5.html#cb88-7"></a>            <span class="dt">mid =</span> BIN_MID[<span class="dv">1</span>],</span>
<span id="cb88-8"><a href="figure-5.html#cb88-8"></a>            <span class="dt">min_dxy =</span> <span class="kw">min</span>(dxy),</span>
<span id="cb88-9"><a href="figure-5.html#cb88-9"></a>            <span class="dt">max_dxy =</span> <span class="kw">max</span>(dxy),</span>
<span id="cb88-10"><a href="figure-5.html#cb88-10"></a>            <span class="dt">mean_dxy =</span> <span class="kw">mean</span>(dxy),</span>
<span id="cb88-11"><a href="figure-5.html#cb88-11"></a>            <span class="dt">median_dxy =</span> <span class="kw">median</span>(dxy),</span>
<span id="cb88-12"><a href="figure-5.html#cb88-12"></a>            <span class="dt">sd_dxy =</span> <span class="kw">sd</span>(dxy),</span>
<span id="cb88-13"><a href="figure-5.html#cb88-13"></a>            <span class="dt">delta_dxy =</span> <span class="kw">max</span>(dxy)<span class="op">-</span><span class="kw">min</span>(dxy))</span></code></pre></div>
<p>To only load the relevant twisst data, we first load the positions of the <span class="math inline">\(F_{ST}\)</span> outlier regions.
We also define a set of outliers of interest.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="figure-5.html#cb89-1"></a><span class="co"># twisst part ------------------</span></span>
<span id="cb89-2"><a href="figure-5.html#cb89-2"></a><span class="co"># load fst outlier regions</span></span>
<span id="cb89-3"><a href="figure-5.html#cb89-3"></a>outlier_table &lt;-<span class="st"> </span>vroom<span class="op">::</span><span class="kw">vroom</span>(out_table, <span class="dt">delim =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb89-4"><a href="figure-5.html#cb89-4"></a><span class="st">  </span><span class="kw">setNames</span>(., <span class="dt">nm =</span> <span class="kw">c</span>(<span class="st">&quot;outlier_id&quot;</span>,<span class="st">&quot;lg&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>, <span class="st">&quot;gstart&quot;</span>,<span class="st">&quot;gend&quot;</span>,<span class="st">&quot;gpos&quot;</span>))</span>
<span id="cb89-5"><a href="figure-5.html#cb89-5"></a></span>
<span id="cb89-6"><a href="figure-5.html#cb89-6"></a><span class="co"># set outlier regions of interest</span></span>
<span id="cb89-7"><a href="figure-5.html#cb89-7"></a>outlier_pick =<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;LG04_1&#39;</span>, <span class="st">&#39;LG12_3&#39;</span>, <span class="st">&#39;LG12_4&#39;</span>)</span></code></pre></div>
<p>Then we define a set of genes of interest.
These are the ones, that later will be labeled in the annotation panel.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="figure-5.html#cb90-1"></a><span class="co"># select genes to label</span></span>
<span id="cb90-2"><a href="figure-5.html#cb90-2"></a>cool_genes &lt;-<span class="st">  </span><span class="kw">c</span>(<span class="st">&#39;arl3&#39;</span>,<span class="st">&#39;kif16b&#39;</span>,<span class="st">&#39;cdx1&#39;</span>,<span class="st">&#39;hmcn2&#39;</span>,</span>
<span id="cb90-3"><a href="figure-5.html#cb90-3"></a>                 <span class="st">&#39;sox10&#39;</span>,<span class="st">&#39;smarca4&#39;</span>,</span>
<span id="cb90-4"><a href="figure-5.html#cb90-4"></a>                 <span class="st">&#39;rorb&#39;</span>,</span>
<span id="cb90-5"><a href="figure-5.html#cb90-5"></a>                 <span class="st">&#39;alox12b&#39;</span>,<span class="st">&#39;egr1&#39;</span>,</span>
<span id="cb90-6"><a href="figure-5.html#cb90-6"></a>                 <span class="st">&#39;ube4b&#39;</span>,<span class="st">&#39;casz1&#39;</span>,</span>
<span id="cb90-7"><a href="figure-5.html#cb90-7"></a>                 <span class="st">&#39;hoxc8a&#39;</span>,<span class="st">&#39;hoxc9&#39;</span>,<span class="st">&#39;hoxc10a&#39;</span>,</span>
<span id="cb90-8"><a href="figure-5.html#cb90-8"></a>                 <span class="st">&#39;hoxc13a&#39;</span>,<span class="st">&#39;rarga&#39;</span>,<span class="st">&#39;rarg&#39;</span>,</span>
<span id="cb90-9"><a href="figure-5.html#cb90-9"></a>                 <span class="st">&#39;snai1&#39;</span>,<span class="st">&#39;fam83d&#39;</span>,<span class="st">&#39;mafb&#39;</span>,<span class="st">&#39;sws2abeta&#39;</span>,<span class="st">&#39;sws2aalpha&#39;</span>,<span class="st">&#39;sws2b&#39;</span>,<span class="st">&#39;lws&#39;</span>,<span class="st">&#39;grm8&#39;</span>)</span></code></pre></div>
<p>Next, we load the twisst data for both locations and list all species from Belize (This will be needed to calculate their pair wise distances for the topology highlighting).</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="figure-5.html#cb91-1"></a><span class="co"># load twisst data</span></span>
<span id="cb91-2"><a href="figure-5.html#cb91-2"></a>data_tables &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">bel =</span> <span class="kw">prep_data</span>(<span class="dt">loc =</span> <span class="st">&#39;bel&#39;</span>),</span>
<span id="cb91-3"><a href="figure-5.html#cb91-3"></a>                    <span class="dt">hon =</span> <span class="kw">prep_data</span>(<span class="dt">loc =</span> <span class="st">&#39;hon&#39;</span>))</span>
<span id="cb91-4"><a href="figure-5.html#cb91-4"></a></span>
<span id="cb91-5"><a href="figure-5.html#cb91-5"></a><span class="co"># set species sampled in belize</span></span>
<span id="cb91-6"><a href="figure-5.html#cb91-6"></a>pops_bel &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;ind&#39;</span>, <span class="st">&#39;may&#39;</span>, <span class="st">&#39;nig&#39;</span>, <span class="st">&#39;pue&#39;</span>, <span class="st">&#39;uni&#39;</span>)</span></code></pre></div>
</div>
<div id="plotting-2" class="section level3">
<h3><span class="header-section-number">17.2.3</span> Plotting</h3>
<p>As a last step before the actual plotting, we are defining a list of outliers to be included within the final plots.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="figure-5.html#cb92-1"></a><span class="co"># set outlier region label</span></span>
<span id="cb92-2"><a href="figure-5.html#cb92-2"></a>region_label_tibbles &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">outlier_id =</span> outlier_pick,</span>
<span id="cb92-3"><a href="figure-5.html#cb92-3"></a>                            <span class="dt">label =</span> <span class="kw">c</span>(<span class="st">&#39;A&#39;</span>,<span class="st">&#39;B&#39;</span>,<span class="st">&#39;C&#39;</span>))</span></code></pre></div>
<p>We adjust the trait images to show the depicted trait in the same color it is also plotted within figure.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="figure-5.html#cb93-1"></a><span class="co"># load and recolor trait icons</span></span>
<span id="cb93-2"><a href="figure-5.html#cb93-2"></a>trait_grob &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">svg =</span> hypoimg<span class="op">::</span>hypo_trait_img<span class="op">$</span>grob_circle[hypoimg<span class="op">::</span>hypo_trait_img<span class="op">$</span>trait <span class="op">%in%</span><span class="st"> </span>gxp_traits],</span>
<span id="cb93-3"><a href="figure-5.html#cb93-3"></a>                     <span class="dt">layer =</span> <span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">3</span>,<span class="dv">7</span>),</span>
<span id="cb93-4"><a href="figure-5.html#cb93-4"></a>                     <span class="dt">color =</span> gxp_clr[gxp_traits <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>()])<span class="op">%&gt;%</span></span>
<span id="cb93-5"><a href="figure-5.html#cb93-5"></a><span class="st">  </span><span class="kw">pmap</span>(<span class="dt">.f =</span> hypo_recolor_svg) <span class="op">%&gt;%</span></span>
<span id="cb93-6"><a href="figure-5.html#cb93-6"></a><span class="st">  </span><span class="kw">set_names</span>(<span class="dt">nm =</span> gxp_traits <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>())</span>
<span id="cb93-7"><a href="figure-5.html#cb93-7"></a></span>
<span id="cb93-8"><a href="figure-5.html#cb93-8"></a><span class="co"># recolor second bars-layer</span></span>
<span id="cb93-9"><a href="figure-5.html#cb93-9"></a>trait_grob[[<span class="st">&quot;Bars&quot;</span>]] &lt;-<span class="st"> </span>trait_grob[[<span class="st">&quot;Bars&quot;</span>]] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">hypo_recolor_svg</span>(<span class="dt">layer =</span> <span class="dv">7</span>,<span class="dt">color =</span> gxp_clr[[<span class="st">&quot;Bars&quot;</span>]])</span></code></pre></div>
<p>Then we prepare the population trees by computing neighbour joining trees based on pair wise <span class="math inline">\(F_{ST}\)</span>.
(This happens within <code>GenomicOriginsScripts::plot_fst_poptree()</code>.)</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="figure-5.html#cb94-1"></a><span class="co"># prepare population tree trait_panels ----------------------</span></span>
<span id="cb94-2"><a href="figure-5.html#cb94-2"></a><span class="co"># load outler region wide average fst data</span></span>
<span id="cb94-3"><a href="figure-5.html#cb94-3"></a>tree_list &lt;-<span class="st"> </span>outlier_pick <span class="op">%&gt;%</span></span>
<span id="cb94-4"><a href="figure-5.html#cb94-4"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">map_dfr</span>(get_fst_summary_data)</span>
<span id="cb94-5"><a href="figure-5.html#cb94-5"></a></span>
<span id="cb94-6"><a href="figure-5.html#cb94-6"></a><span class="co"># create neighbour-joining trees and plot</span></span>
<span id="cb94-7"><a href="figure-5.html#cb94-7"></a>poptree_plot_list &lt;-<span class="st">  </span>tree_list <span class="op">%&gt;%</span></span>
<span id="cb94-8"><a href="figure-5.html#cb94-8"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">pmap</span>(plot_fst_poptree)</span></code></pre></div>
<p>Then, we iterate the main plotting function over all selected <span class="math inline">\(F_{ST}\)</span> outlier windows and combine the resulting plots into a multi panel plot.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="figure-5.html#cb95-1"></a><span class="co"># compose base figure</span></span>
<span id="cb95-2"><a href="figure-5.html#cb95-2"></a>p_single &lt;-<span class="st"> </span>outlier_table <span class="op">%&gt;%</span></span>
<span id="cb95-3"><a href="figure-5.html#cb95-3"></a><span class="st">  </span><span class="kw">filter</span>(outlier_id <span class="op">%in%</span><span class="st"> </span>outlier_pick) <span class="op">%&gt;%</span></span>
<span id="cb95-4"><a href="figure-5.html#cb95-4"></a><span class="st">  </span><span class="kw">left_join</span>(region_label_tibbles) <span class="op">%&gt;%</span></span>
<span id="cb95-5"><a href="figure-5.html#cb95-5"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">outlier_nr =</span> <span class="kw">row_number</span>(),</span>
<span id="cb95-6"><a href="figure-5.html#cb95-6"></a>         <span class="dt">text =</span> <span class="kw">ifelse</span>(outlier_nr <span class="op">==</span><span class="st"> </span><span class="dv">1</span>,<span class="ot">TRUE</span>,<span class="ot">FALSE</span>),</span>
<span id="cb95-7"><a href="figure-5.html#cb95-7"></a>         <span class="dt">trait =</span> <span class="kw">c</span>(<span class="st">&#39;Snout&#39;</span>, <span class="st">&#39;Bars&#39;</span>, <span class="st">&#39;Peduncle&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb95-8"><a href="figure-5.html#cb95-8"></a><span class="st">  </span><span class="kw">pmap</span>(plot_curtain, <span class="dt">cool_genes =</span> cool_genes) <span class="op">%&gt;%</span></span>
<span id="cb95-9"><a href="figure-5.html#cb95-9"></a><span class="st">  </span><span class="kw">c</span>(., poptree_plot_list) <span class="op">%&gt;%</span></span>
<span id="cb95-10"><a href="figure-5.html#cb95-10"></a><span class="st">  </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> ., <span class="dt">nrow =</span> <span class="dv">2</span>,</span>
<span id="cb95-11"><a href="figure-5.html#cb95-11"></a>                     <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">.17</span>),</span>
<span id="cb95-12"><a href="figure-5.html#cb95-12"></a>                     <span class="dt">labels =</span> letters[<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(outlier_pick)] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">project_case</span>())</span></code></pre></div>
<center>
<img src="plot_F5_files/figure-html/unnamed-chunk-19-1.png" width="1344" />
</center>
<p>At this point all that we miss is the figure legend.
So, for the <span class="math inline">\(F_{ST}\)</span>, <span class="math inline">\(d_{XY}\)</span> and genotype <span class="math inline">\(\times\)</span> phenotype color schemes we create two dummy plots from where we can export the legends.
We combine those two classical color legends into what will become the left column of the legend.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="figure-5.html#cb96-1"></a><span class="co"># compile legend</span></span>
<span id="cb96-2"><a href="figure-5.html#cb96-2"></a><span class="co"># dummy plot for fst legend</span></span>
<span id="cb96-3"><a href="figure-5.html#cb96-3"></a>p_dummy_fst &lt;-<span class="st"> </span>outlier_table <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="kw">row_number</span>() <span class="op">==</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span>purrr<span class="op">::</span><span class="kw">pmap</span>(plot_panel_fst) <span class="op">%&gt;%</span><span class="st"> </span>.[[<span class="dv">1</span>]]</span>
<span id="cb96-4"><a href="figure-5.html#cb96-4"></a><span class="co"># dummy plot for  gxp legend</span></span>
<span id="cb96-5"><a href="figure-5.html#cb96-5"></a>p_dummy_gxp &lt;-<span class="st"> </span>outlier_table <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="kw">row_number</span>() <span class="op">==</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span>purrr<span class="op">::</span><span class="kw">pmap</span>(plot_panel_gxp, <span class="dt">trait =</span> <span class="st">&#39;Bars&#39;</span>) <span class="op">%&gt;%</span><span class="st"> </span>.[[<span class="dv">1</span>]]</span>
<span id="cb96-6"><a href="figure-5.html#cb96-6"></a><span class="co"># fst legend</span></span>
<span id="cb96-7"><a href="figure-5.html#cb96-7"></a>p_leg_fst &lt;-<span class="st"> </span>(p_dummy_fst<span class="op">+</span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&#39;bottom&#39;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">get_legend</span>()</span>
<span id="cb96-8"><a href="figure-5.html#cb96-8"></a><span class="co"># gxp legend</span></span>
<span id="cb96-9"><a href="figure-5.html#cb96-9"></a>p_leg_gxp &lt;-<span class="st"> </span>(p_dummy_gxp<span class="op">+</span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&#39;bottom&#39;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">get_legend</span>()</span>
<span id="cb96-10"><a href="figure-5.html#cb96-10"></a><span class="co"># create poptree legend</span></span>
<span id="cb96-11"><a href="figure-5.html#cb96-11"></a>p_leg_poptree &lt;-<span class="st"> </span>(poptree_plot_list[[<span class="dv">1</span>]] <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">get_legend</span>()</span>
<span id="cb96-12"><a href="figure-5.html#cb96-12"></a><span class="co"># create sub-legend 1</span></span>
<span id="cb96-13"><a href="figure-5.html#cb96-13"></a>p_leg1 &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p_leg_fst,</span>
<span id="cb96-14"><a href="figure-5.html#cb96-14"></a>                             p_leg_gxp,</span>
<span id="cb96-15"><a href="figure-5.html#cb96-15"></a>                             <span class="dt">ncol =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>Then, we construct the topology highlighting color legend.
We first define the three highlighting scenarios, the involved species and their base color and then iterate the legend plotting functions over those configurations.
The resulting legend elements are then combined to create the right side of the figure legend and the two main legend elements are combined.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="figure-5.html#cb97-1"></a><span class="co"># create sub-legend 2 (phylogeny schematics)</span></span>
<span id="cb97-2"><a href="figure-5.html#cb97-2"></a>p_leg2 &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">spec1 =</span> <span class="kw">c</span>(<span class="st">&#39;indigo&#39;</span>, <span class="st">&#39;indigo&#39;</span>,<span class="st">&#39;unicolor&#39;</span>),</span>
<span id="cb97-3"><a href="figure-5.html#cb97-3"></a>                 <span class="dt">spec2 =</span> <span class="kw">c</span>(<span class="st">&#39;maya&#39;</span>, <span class="st">&#39;puella&#39;</span>,<span class="ot">NA</span>),</span>
<span id="cb97-4"><a href="figure-5.html#cb97-4"></a>                 <span class="dt">color =</span> twisst_clr <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unname</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">darken</span>(.,<span class="dt">factor =</span> <span class="fl">.8</span>),</span>
<span id="cb97-5"><a href="figure-5.html#cb97-5"></a>                 <span class="dt">mode =</span> <span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&#39;pair&#39;</span>,<span class="dv">2</span>),<span class="st">&#39;isolation&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb97-6"><a href="figure-5.html#cb97-6"></a><span class="st">  </span><span class="kw">future_pmap</span>(plot_leg) <span class="op">%&gt;%</span></span>
<span id="cb97-7"><a href="figure-5.html#cb97-7"></a><span class="st">  </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> .,</span>
<span id="cb97-8"><a href="figure-5.html#cb97-8"></a>                     <span class="dt">nrow =</span> <span class="dv">1</span>)</span>
<span id="cb97-9"><a href="figure-5.html#cb97-9"></a></span>
<span id="cb97-10"><a href="figure-5.html#cb97-10"></a><span class="co"># create sub-legend 3</span></span>
<span id="cb97-11"><a href="figure-5.html#cb97-11"></a>p_leg_<span class="dv">3</span> &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p_leg1,</span>
<span id="cb97-12"><a href="figure-5.html#cb97-12"></a>                            p_leg2,</span>
<span id="cb97-13"><a href="figure-5.html#cb97-13"></a>                            <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(.<span class="dv">6</span>, <span class="dv">1</span>))</span></code></pre></div>
<p>Lastly, the legend elements from the population trees are added to complete the final legend.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="figure-5.html#cb98-1"></a><span class="co"># complete legend</span></span>
<span id="cb98-2"><a href="figure-5.html#cb98-2"></a>p_leg &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(</span>
<span id="cb98-3"><a href="figure-5.html#cb98-3"></a>  p_leg_poptree,</span>
<span id="cb98-4"><a href="figure-5.html#cb98-4"></a>  p_leg_<span class="dv">3</span>,</span>
<span id="cb98-5"><a href="figure-5.html#cb98-5"></a>  <span class="dt">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb98-6"><a href="figure-5.html#cb98-6"></a>  <span class="dt">rel_heights =</span> <span class="kw">c</span>(.<span class="dv">2</span>, <span class="dv">1</span>))</span></code></pre></div>
<center>
<img src="plot_F5_files/figure-html/unnamed-chunk-23-1.png" width="1344" />
</center>
<p>After adding the legend to the main part, Figure 4 is complete.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="figure-5.html#cb99-1"></a><span class="co"># finalize figure</span></span>
<span id="cb99-2"><a href="figure-5.html#cb99-2"></a>p_done &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p_single, p_leg,</span>
<span id="cb99-3"><a href="figure-5.html#cb99-3"></a>                             <span class="dt">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb99-4"><a href="figure-5.html#cb99-4"></a>                             <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">.17</span>))</span></code></pre></div>
<center>
<img src="plot_F5_files/figure-html/unnamed-chunk-25-1.png" width="1344" />
</center>
<p>Finally, we can export Figure 5.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="figure-5.html#cb100-1"></a><span class="co"># export figure</span></span>
<span id="cb100-2"><a href="figure-5.html#cb100-2"></a>scl &lt;-<span class="st"> </span><span class="fl">.9</span></span>
<span id="cb100-3"><a href="figure-5.html#cb100-3"></a><span class="kw">hypo_save</span>(<span class="dt">plot =</span> p_done, <span class="dt">filename =</span> <span class="st">&#39;figures/F5.pdf&#39;</span>,</span>
<span id="cb100-4"><a href="figure-5.html#cb100-4"></a>          <span class="dt">width =</span> <span class="dv">14</span><span class="op">*</span>scl, <span class="dt">height =</span> <span class="dv">13</span><span class="op">*</span>scl,</span>
<span id="cb100-5"><a href="figure-5.html#cb100-5"></a>          <span class="dt">comment =</span> script_name,</span>
<span id="cb100-6"><a href="figure-5.html#cb100-6"></a>          <span class="dt">device =</span> cairo_pdf)</span></code></pre></div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="figure-4.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="supplementary-figure-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
