<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8 Figure 1 | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘The genomic origins of a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.13 and GitBook 2.6.7" />

  <meta property="og:title" content="8 Figure 1 | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘The genomic origins of a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 Figure 1 | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘The genomic origins of a marine radiation’ by Hench, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench" />


<meta name="date" content="2019-10-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="analysis-iv-rho.html"/>
<link rel="next" href="figure-2.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../">Script repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.2</b> Figures</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.3</b> Prerequesites</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="genotyping-i-snps-only.html"><a href="genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> Genotyping I (SNPs only)</a><ul>
<li class="chapter" data-level="2.1" data-path="genotyping-i-snps-only.html"><a href="genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="genotyping-i-snps-only.html"><a href="genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a><ul>
<li class="chapter" data-level="2.2.1" data-path="genotyping-i-snps-only.html"><a href="genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="genotyping-i-snps-only.html"><a href="genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="genotyping-i-snps-only.html"><a href="genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genotyping-ii-all-callable-sites.html"><a href="genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> Genotyping II (all callable sites)</a><ul>
<li class="chapter" data-level="3.1" data-path="genotyping-ii-all-callable-sites.html"><a href="genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="genotyping-ii-all-callable-sites.html"><a href="genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a><ul>
<li class="chapter" data-level="3.2.1" data-path="genotyping-ii-all-callable-sites.html"><a href="genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis-i-fst-gxp.html"><a href="analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a><ul>
<li class="chapter" data-level="4.1" data-path="analysis-i-fst-gxp.html"><a href="analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="analysis-i-fst-gxp.html"><a href="analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a><ul>
<li class="chapter" data-level="4.2.1" data-path="analysis-i-fst-gxp.html"><a href="analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="analysis-ii-dxy-pi.html"><a href="analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> Analysis II (<i>d<sub>XY</sub></i> &amp; <span class="math inline">\(\pi\)</span>)</a><ul>
<li class="chapter" data-level="5.1" data-path="analysis-ii-dxy-pi.html"><a href="analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="analysis-ii-dxy-pi.html"><a href="analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a><ul>
<li class="chapter" data-level="5.2.1" data-path="analysis-ii-dxy-pi.html"><a href="analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="analysis-iii-phylogeny-topology-weighting.html"><a href="analysis-iii-phylogeny-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> Analysis III (phylogeny &amp; topology weighting)</a><ul>
<li class="chapter" data-level="6.1" data-path="analysis-iii-phylogeny-topology-weighting.html"><a href="analysis-iii-phylogeny-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="analysis-iii-phylogeny-topology-weighting.html"><a href="analysis-iii-phylogeny-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a><ul>
<li class="chapter" data-level="6.2.1" data-path="analysis-iii-phylogeny-topology-weighting.html"><a href="analysis-iii-phylogeny-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="analysis-iv-rho.html"><a href="analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> Analysis IV (<span class="math inline">\(\rho\)</span>)</a><ul>
<li class="chapter" data-level="7.1" data-path="analysis-iv-rho.html"><a href="analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="analysis-iv-rho.html"><a href="analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a><ul>
<li class="chapter" data-level="7.2.1" data-path="analysis-iv-rho.html"><a href="analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>8</b> Figure 1</a><ul>
<li class="chapter" data-level="8.1" data-path="figure-1.html"><a href="figure-1.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>8.2</b> Details of <code>plot_F1.R</code></a><ul>
<li class="chapter" data-level="8.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>8.2.1</b> Config</a></li>
<li class="chapter" data-level="8.2.2" data-path="figure-1.html"><a href="figure-1.html#data-import"><i class="fa fa-check"></i><b>8.2.2</b> Data import</a></li>
<li class="chapter" data-level="8.2.3" data-path="figure-1.html"><a href="figure-1.html#plotting"><i class="fa fa-check"></i><b>8.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>9</b> Figure 2</a><ul>
<li class="chapter" data-level="9.1" data-path="figure-2.html"><a href="figure-2.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>9.2</b> Details of <code>plot_F2.R</code></a><ul>
<li class="chapter" data-level="9.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>9.2.1</b> Config</a></li>
<li class="chapter" data-level="9.2.2" data-path="figure-2.html"><a href="figure-2.html#data-import-1"><i class="fa fa-check"></i><b>9.2.2</b> Data import</a></li>
<li class="chapter" data-level="9.2.3" data-path="figure-2.html"><a href="figure-2.html#plotting-1"><i class="fa fa-check"></i><b>9.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>10</b> Figure 3</a><ul>
<li class="chapter" data-level="10.1" data-path="figure-3.html"><a href="figure-3.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>10.2</b> Details of <code>plot_F3.R</code></a><ul>
<li class="chapter" data-level="10.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>10.2.1</b> Config</a></li>
<li class="chapter" data-level="10.2.2" data-path="figure-3.html"><a href="figure-3.html#data-import-2"><i class="fa fa-check"></i><b>10.2.2</b> Data import</a></li>
<li class="chapter" data-level="10.2.3" data-path="figure-3.html"><a href="figure-3.html#plotting-2"><i class="fa fa-check"></i><b>10.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-1" class="section level1">
<h1><span class="header-section-number">8</span> Figure 1</h1>
<div id="summary-6" class="section level2">
<h2><span class="header-section-number">8.1</span> Summary</h2>
<p>This is the accessory documentation of Figure 1.
The Figure can be recreated by running the <strong>R</strong> script <code>plot_F1.R</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="bu">cd</span> <span class="va">$BASE_DIR</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="ex">Rscript</span> --vanilla R/fig/plot_F1.R 2_analysis/dxy/50k/ 2_analysis/fst/50k/</a></code></pre></div>
</div>
<div id="details-of-plot_f1.r" class="section level2">
<h2><span class="header-section-number">8.2</span> Details of <code>plot_F1.R</code></h2>
<p>In the following, the individual steps of the R script are documented.
It is an executable R script that depends on the accessory R package <strong>GenomicOriginsScripts</strong>.</p>
<div id="config" class="section level3">
<h3><span class="header-section-number">8.2.1</span> Config</h3>
<p>The scripts start with a header that contains copy &amp; paste templates to execute or debug the script:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co">#!/usr/bin/env Rscript</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="co"># run from terminal:</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co"># Rscript --vanilla R/fig/plot_F1.R 2_analysis/dxy/50k/ 2_analysis/fst/50k/</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co"># ===============================================================</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co"># This script produces Figure 1 of the study &quot;The genomic origins of a marine radiation&quot;</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co"># by Hench, McMillan an Puebla</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co"># ---------------------------------------------------------------</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co"># ===============================================================</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co"># args &lt;- c(&#39;2_analysis/dxy/50k/&#39;, &#39;2_analysis/fst/50k/&#39;)</span></a></code></pre></div>
<p>The next section processes the input from the command line.
It stores the arguments in the vector <code>args</code>.
The R package <strong>GenomicOriginsScripts</strong> is loaded and the script name and the current working directory are stored inside variables (<code>script_name</code>, <code>plot_comment</code>).
This information will later be written into the meta data of the figure to help us tracing back the scripts that created the figures in the future.</p>
<p>Then we drop all the imported information besides the arguments following the script name and print the information to the terminal.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">args =<span class="st"> </span><span class="kw">commandArgs</span>(<span class="dt">trailingOnly=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co"># setup -----------------------</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="kw">library</span>(GenomicOriginsScripts)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">library</span>(hypoimg)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="kw">cat</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>)</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">script_name &lt;-<span class="st"> </span>args[<span class="dv">5</span>] <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="st">  </span><span class="kw">str_remove</span>(.,<span class="st">&#39;--file=&#39;</span>)</a>
<a class="sourceLine" id="cb3-9" data-line-number="9"></a>
<a class="sourceLine" id="cb3-10" data-line-number="10">plot_comment &lt;-<span class="st"> </span>script_name <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="st">  </span><span class="kw">str_c</span>(<span class="st">&#39;mother-script = &#39;</span>,<span class="kw">getwd</span>(),<span class="st">&#39;/&#39;</span>,.) </a>
<a class="sourceLine" id="cb3-12" data-line-number="12"></a>
<a class="sourceLine" id="cb3-13" data-line-number="13">args &lt;-<span class="st"> </span><span class="kw">process_input</span>(script_name, args)</a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co">#&gt; ── Script: scripts/plot_F1.R ────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">#&gt; Parameters read:</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt;  ★ 1: 2_analysis/dxy/50k/</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt;  ★ 2: 2_analysis/fst/50k/</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt; ─────────────────────────────────────────── /current/working/directory ──</span></a></code></pre></div>
<p>The directories for the different data types are received and stored in respective variables.
Also, we set a few parameters for the plot layout:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># config -----------------------</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">dxy_dir &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">1</span>])</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">fst_dir &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">2</span>])</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">wdh &lt;-<span class="st"> </span><span class="fl">.3</span>          <span class="co"># The width of the boxplots</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5">scaler &lt;-<span class="st"> </span><span class="dv">20</span>       <span class="co"># the ratio of the Fst and the dxy axis</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6">clr_sec &lt;-<span class="st"> &#39;gray&#39;</span>  <span class="co"># the color of the secondary axis (dxy)</span></a></code></pre></div>
</div>
<div id="data-import" class="section level3">
<h3><span class="header-section-number">8.2.2</span> Data import</h3>
<p>We begin with the data import by first collecting the paths to all files containing either <em>F<sub>ST</sub></em> or <em>d<sub>XY</sub></em> data (<code>dir()</code>), then iterating the import function over all files (<code>map(summarize_fst)</code>) and finally combining the outputs into a single tibble (<code>bind_rows()</code>).
This is done for both <em>F<sub>ST</sub></em> and <em>d<sub>XY</sub></em>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># start script -------------------</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co"># import Fst </span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4">fst_files &lt;-<span class="st"> </span><span class="kw">dir</span>(fst_dir, <span class="dt">pattern =</span> <span class="st">&#39;.50k.windowed.weir.fst.gz&#39;</span>)</a>
<a class="sourceLine" id="cb6-5" data-line-number="5"></a>
<a class="sourceLine" id="cb6-6" data-line-number="6">fst_data &lt;-<span class="st"> </span><span class="kw">str_c</span>(fst_dir,fst_files) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="st">  </span>purrr<span class="op">::</span><span class="kw">map</span>(summarize_fst) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="st">  </span><span class="kw">bind_rows</span>()</a>
<a class="sourceLine" id="cb6-9" data-line-number="9"></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co"># import dxy </span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11">dxy_files &lt;-<span class="st"> </span><span class="kw">dir</span>(dxy_dir)</a>
<a class="sourceLine" id="cb6-12" data-line-number="12"></a>
<a class="sourceLine" id="cb6-13" data-line-number="13">dxy_data &lt;-<span class="st">  </span><span class="kw">str_c</span>(dxy_dir,dxy_files) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="st">  </span>purrr<span class="op">::</span><span class="kw">map</span>(summarize_dxy) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="st">  </span><span class="kw">bind_rows</span>()</a></code></pre></div>
<p>We use the genome wide average <em>F<sub>ST</sub></em> to rank the individual pair wise comparisons.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">fst_order &lt;-<span class="st"> </span>fst_data <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="st">  </span><span class="kw">select</span>(run, <span class="st">`</span><span class="dt">mean_weighted-fst</span><span class="st">`</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">run =</span> <span class="kw">fct_reorder</span>(run,<span class="st">`</span><span class="dt">mean_weighted-fst</span><span class="st">`</span>))</a></code></pre></div>
<p>Then, we merge the <em>F<sub>ST</sub></em> and <em>d<sub>XY</sub></em> data sets and do quite a bit of data wrangling to create a rescaled <em>d<sub>XY</sub></em> value and to prepare the placement of the boxplots.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">data &lt;-<span class="st"> </span><span class="kw">left_join</span>(fst_data, dxy_data) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="st">  </span><span class="kw">select</span>(<span class="kw">c</span>(<span class="dv">8</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">7</span>,<span class="dv">9</span><span class="op">:</span><span class="dv">15</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="st">  </span><span class="co"># transfrom table from short to long format </span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="st">  </span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&#39;stat&#39;</span>, <span class="dt">value =</span> <span class="st">&#39;val&#39;</span>,<span class="dv">2</span><span class="op">:</span><span class="dv">15</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="st">  </span><span class="co"># separating the boxplot element from the population genetic parameter</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="st">  </span><span class="kw">separate</span>(stat, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;sumstat&#39;</span>,<span class="st">&#39;popstat&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="st">  </span><span class="co"># rescaleing the dxy values to fall in the range of Fst</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">val_scaled =</span> <span class="kw">ifelse</span>(popstat <span class="op">==</span><span class="st"> &#39;dxy&#39;</span>, val <span class="op">*</span><span class="st"> </span>scaler , val)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="st">  </span><span class="co"># combining original and rescaled values </span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="st">  </span><span class="kw">unite</span>(temp, val, val_scaled) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="st">  </span><span class="co"># back transforming to short format</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="st">  </span><span class="kw">spread</span>(.,<span class="dt">key =</span> <span class="st">&#39;sumstat&#39;</span>,<span class="dt">value =</span> <span class="st">&#39;temp&#39;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="st">  </span><span class="co"># separating original and rescaled values for all boxplot elements</span></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="st">  </span><span class="kw">separate</span>(mean, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;mean&#39;</span>,<span class="st">&#39;mean_scaled&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-15" data-line-number="15"><span class="st">  </span><span class="kw">separate</span>(median, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;median&#39;</span>,<span class="st">&#39;median_scaled&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="st">  </span><span class="kw">separate</span>(sd, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;sd&#39;</span>,<span class="st">&#39;sd_scaled&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17"><span class="st">  </span><span class="kw">separate</span>(lower, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;lower&#39;</span>,<span class="st">&#39;lower_scaled&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-18" data-line-number="18"><span class="st">  </span><span class="kw">separate</span>(upper, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;upper&#39;</span>,<span class="st">&#39;upper_scaled&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-19" data-line-number="19"><span class="st">  </span><span class="kw">separate</span>(lowpoint, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;lowpoint&#39;</span>,<span class="st">&#39;lowpoint_scaled&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-20" data-line-number="20"><span class="st">  </span><span class="kw">separate</span>(highpoint, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;highpoint&#39;</span>,<span class="st">&#39;highpoint_scaled&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-21" data-line-number="21"><span class="st">  </span><span class="co"># adding preparations for plotting</span></a>
<a class="sourceLine" id="cb8-22" data-line-number="22"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">loc =</span> <span class="kw">str_sub</span>(run,<span class="dv">4</span>,<span class="dv">6</span>),                             <span class="co"># extract location from run</span></a>
<a class="sourceLine" id="cb8-23" data-line-number="23">         <span class="dt">run =</span> <span class="kw">factor</span>(run, <span class="dt">levels =</span> <span class="kw">levels</span>(fst_order<span class="op">$</span>run)),  <span class="co"># introduce the run ranking</span></a>
<a class="sourceLine" id="cb8-24" data-line-number="24">         <span class="dt">x =</span> <span class="kw">as.numeric</span>(run) ,                               <span class="co"># extract rank of runs (base x)</span></a>
<a class="sourceLine" id="cb8-25" data-line-number="25">         <span class="dt">x_dodge =</span> <span class="kw">ifelse</span>(popstat <span class="op">==</span><span class="st"> &#39;dxy&#39;</span>,x <span class="op">+</span><span class="st"> </span><span class="fl">.25</span>,x <span class="op">-</span><span class="st"> </span><span class="fl">.25</span>), <span class="co"># shift dxy to the right, fst left</span></a>
<a class="sourceLine" id="cb8-26" data-line-number="26">         <span class="dt">x_start_dodge =</span> x_dodge <span class="op">-</span><span class="st"> </span>wdh<span class="op">/</span><span class="dv">2</span>,                    <span class="co"># boxplot left border</span></a>
<a class="sourceLine" id="cb8-27" data-line-number="27">         <span class="dt">x_end_dodge =</span> x_dodge <span class="op">+</span><span class="st"> </span>wdh<span class="op">/</span><span class="dv">2</span>,                      <span class="co"># boxplot right border</span></a>
<a class="sourceLine" id="cb8-28" data-line-number="28">         <span class="dt">popstat_loc =</span> <span class="kw">str_c</span>(popstat,<span class="st">&#39;[&#39;</span>,loc,<span class="st">&#39;]&#39;</span>))           <span class="co"># parameter X location (for color) </span></a></code></pre></div>
<p>At this point, the data is ready for the boxplots.
But first we are going to prepare the networks of pairwise comparisons.</p>
<p>For this we create a tibble of the runs with their respective rank.
Then, we prepare a config table with one row per location, storing the parameters neede for the layout function for the networks.
We need to define the location, the number of species at the location, the short three letter ID of those species and a wheight parameter that is shifting the comparison label on the link within the networks.</p>
<p>Finally, we create one network plot per location.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># preparing a run/rnk lookup table</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">run_ord &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">run =</span> <span class="kw">levels</span>(data<span class="op">$</span>run),</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">       <span class="dt">run_ord =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">levels</span>(data<span class="op">$</span>run)))</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co"># network config and layout</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6">networx &lt;-<span class="st"> </span><span class="kw">tibble</span>( <span class="dt">loc =</span> <span class="kw">c</span>(<span class="st">&#39;bel&#39;</span>,<span class="st">&#39;hon&#39;</span>, <span class="st">&#39;pan&#39;</span>), </a>
<a class="sourceLine" id="cb9-7" data-line-number="7">                   <span class="dt">n =</span> <span class="kw">c</span>(<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">3</span>),</a>
<a class="sourceLine" id="cb9-8" data-line-number="8">                   <span class="dt">label =</span> <span class="kw">list</span>(<span class="kw">str_c</span>(<span class="kw">c</span>(<span class="st">&#39;ind&#39;</span>,<span class="st">&#39;may&#39;</span>,<span class="st">&#39;nig&#39;</span>,<span class="st">&#39;pue&#39;</span>,<span class="st">&#39;uni&#39;</span>),<span class="st">&#39;bel&#39;</span>),</a>
<a class="sourceLine" id="cb9-9" data-line-number="9">                                <span class="kw">str_c</span>(<span class="kw">c</span>(<span class="st">&#39;abe&#39;</span>,<span class="st">&#39;gum&#39;</span>,<span class="st">&#39;nig&#39;</span>,<span class="st">&#39;pue&#39;</span>,<span class="st">&#39;ran&#39;</span>,<span class="st">&#39;uni&#39;</span>),<span class="st">&#39;hon&#39;</span>),</a>
<a class="sourceLine" id="cb9-10" data-line-number="10">                                <span class="kw">str_c</span>(<span class="kw">c</span>(<span class="st">&#39;nig&#39;</span>,<span class="st">&#39;pue&#39;</span>,<span class="st">&#39;uni&#39;</span>),<span class="st">&#39;pan&#39;</span>)),</a>
<a class="sourceLine" id="cb9-11" data-line-number="11">                   <span class="dt">weight =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="fl">1.45</span>,<span class="dv">1</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="st">  </span>purrr<span class="op">::</span><span class="kw">pmap</span>(network_layout) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="st">  </span><span class="kw">bind_rows</span>()</a>
<a class="sourceLine" id="cb9-14" data-line-number="14"></a>
<a class="sourceLine" id="cb9-15" data-line-number="15">plot_list &lt;-<span class="st"> </span>networx <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="st">  </span>purrr<span class="op">::</span><span class="kw">pmap</span>(plot_network, <span class="dt">node_lab_shift =</span> <span class="fl">.2</span>)</a></code></pre></div>
</div>
<div id="plotting" class="section level3">
<h3><span class="header-section-number">8.2.3</span> Plotting</h3>
<p>To create the first panel of Figure 1, we combine the three networks and label the locations.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">p1 &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">  grid<span class="op">::</span><span class="kw">textGrob</span>(<span class="st">&#39;Belize&#39;</span>),</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">  grid<span class="op">::</span><span class="kw">textGrob</span>(<span class="st">&#39;Honduras&#39;</span>),</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">  grid<span class="op">::</span><span class="kw">textGrob</span>(<span class="st">&#39;Panama&#39;</span>),</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">  plot_list[[<span class="dv">1</span>]], plot_list[[<span class="dv">2</span>]], plot_list[[<span class="dv">3</span>]], </a>
<a class="sourceLine" id="cb10-6" data-line-number="6">  <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(.<span class="dv">1</span>,<span class="dv">1</span>))</a></code></pre></div>
<center>
<img src="plot_F1_files/figure-html/unnamed-chunk-10-1.png" width="864" />
</center>
<p>Now, we can create the second panel of Figure 1, by plotting our prepared data tibble.
We are going to plot each boxplot element as a single layer.
(This, might seem a little cumbersome given <code>geom_boxplot()</code>, but this approach was choosen for specific fine tuning of the postioning, dropping of outliers and reducing runtime during the plotting phase - otherwise the entire genome wide data set would have been carried though whole script.)</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">p2 &lt;-<span class="st"> </span>data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">color =</span> popstat_loc)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="st">  </span><span class="co"># adding whiskers</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x_dodge, <span class="dt">xend =</span> x_dodge, </a>
<a class="sourceLine" id="cb11-5" data-line-number="5">                   <span class="dt">y =</span> lowpoint_scaled,<span class="dt">yend =</span> highpoint_scaled)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="st">  </span><span class="co"># adding box (inter quartile range)</span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_rect</span>(<span class="kw">aes</span>(<span class="dt">xmin =</span> x_start_dodge, <span class="dt">xmax =</span> x_end_dodge,</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">                <span class="dt">ymin =</span> lower_scaled, <span class="dt">ymax =</span> upper_scaled), </a>
<a class="sourceLine" id="cb11-9" data-line-number="9">             <span class="dt">fill=</span><span class="st">&#39;white&#39;</span>)<span class="op">+</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="st">  </span><span class="co"># adding bar (median)</span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x_start_dodge,</a>
<a class="sourceLine" id="cb11-12" data-line-number="12">                   <span class="dt">xend =</span> x_end_dodge,</a>
<a class="sourceLine" id="cb11-13" data-line-number="13">                   <span class="dt">y =</span> median_scaled,</a>
<a class="sourceLine" id="cb11-14" data-line-number="14">                   <span class="dt">yend =</span> median_scaled),<span class="dt">lwd =</span> <span class="fl">.9</span>)<span class="op">+</span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="st">  </span><span class="co"># adding point (mean)</span></a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x_dodge, <span class="dt">y =</span> mean_scaled),<span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="fl">.7</span>, <span class="dt">fill =</span> <span class="st">&#39;white&#39;</span>)<span class="op">+</span></a>
<a class="sourceLine" id="cb11-17" data-line-number="17"><span class="st">  </span><span class="co"># layout x ayis</span></a>
<a class="sourceLine" id="cb11-18" data-line-number="18"><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">28</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-19" data-line-number="19"><span class="st">  </span><span class="co"># layout primary y axis (Fst)</span></a>
<a class="sourceLine" id="cb11-20" data-line-number="20"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">05</span>,.<span class="dv">1</span>,.<span class="dv">15</span>),</a>
<a class="sourceLine" id="cb11-21" data-line-number="21">                     <span class="dt">name =</span> <span class="kw">expression</span>(<span class="kw">italic</span>(F[ST])),</a>
<a class="sourceLine" id="cb11-22" data-line-number="22">                     <span class="co"># layout secondary y axis (dxy)</span></a>
<a class="sourceLine" id="cb11-23" data-line-number="23">                     <span class="dt">sec.axis =</span> <span class="kw">sec_axis</span>(<span class="op">~</span><span class="st"> </span>. <span class="op">/</span>scaler,</a>
<a class="sourceLine" id="cb11-24" data-line-number="24">                                         <span class="dt">name =</span> <span class="kw">expression</span>(<span class="kw">italic</span>(d[XY])), </a>
<a class="sourceLine" id="cb11-25" data-line-number="25">                                         <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">005</span>,.<span class="dv">01</span>)))<span class="op">+</span></a>
<a class="sourceLine" id="cb11-26" data-line-number="26"><span class="st">  </span><span class="co"># define color scheme</span></a>
<a class="sourceLine" id="cb11-27" data-line-number="27"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="kw">make_faint_clr</span>(<span class="st">&#39;bel&#39;</span>),</a>
<a class="sourceLine" id="cb11-28" data-line-number="28">                                <span class="kw">make_faint_clr</span>(<span class="st">&#39;hon&#39;</span>),</a>
<a class="sourceLine" id="cb11-29" data-line-number="29">                                <span class="kw">make_faint_clr</span>(<span class="st">&#39;pan&#39;</span>))[<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">6</span>)])<span class="op">+</span></a>
<a class="sourceLine" id="cb11-30" data-line-number="30"><span class="st">  </span><span class="co"># set plot range</span></a>
<a class="sourceLine" id="cb11-31" data-line-number="31"><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">29</span>),</a>
<a class="sourceLine" id="cb11-32" data-line-number="32">                  <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="op">+</span></a>
<a class="sourceLine" id="cb11-33" data-line-number="33"><span class="st">  </span><span class="co"># tune plot appreance</span></a>
<a class="sourceLine" id="cb11-34" data-line-number="34"><span class="st">  </span><span class="kw">theme_minimal</span>()<span class="op">+</span></a>
<a class="sourceLine" id="cb11-35" data-line-number="35"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb11-36" data-line-number="36">        <span class="dt">legend.position =</span> <span class="st">&#39;none&#39;</span>,</a>
<a class="sourceLine" id="cb11-37" data-line-number="37">        <span class="dt">strip.placement =</span> <span class="st">&#39;outside&#39;</span>,</a>
<a class="sourceLine" id="cb11-38" data-line-number="38">        <span class="dt">strip.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>),</a>
<a class="sourceLine" id="cb11-39" data-line-number="39">        <span class="dt">panel.grid.major.x =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb11-40" data-line-number="40">        <span class="dt">panel.grid.minor.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb11-41" data-line-number="41">        <span class="dt">axis.text.y.right =</span> <span class="kw">element_text</span>(<span class="dt">color =</span> clr_sec),</a>
<a class="sourceLine" id="cb11-42" data-line-number="42">        <span class="dt">axis.title.y.right =</span> <span class="kw">element_text</span>(<span class="dt">color =</span> clr_sec))</a></code></pre></div>
<center>
<img src="plot_F1_files/figure-html/unnamed-chunk-12-1.png" width="864" />
</center>
<p>To create the final figure, we combine the two sub panels unsing the <strong>cowplot</strong> package.
Here, we add the panel labels and define the space each panel is going to take.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">p_done &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1,p2,</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">                             <span class="dt">ncol =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">                             <span class="dt">rel_heights =</span> <span class="kw">c</span>(.<span class="dv">7</span>,<span class="dv">1</span>),</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">                             <span class="dt">labels =</span> letters[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">project_case</span>())</a></code></pre></div>
<center>
<img src="plot_F1_files/figure-html/unnamed-chunk-14-1.png" width="864" />
</center>
<p>Finally, we can export Figure 1.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">hypo_save</span>(p_done, <span class="dt">filename =</span> <span class="st">&#39;figures/F1.pdf&#39;</span>,</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">          <span class="dt">width =</span> <span class="dv">9</span>, <span class="dt">height =</span> <span class="dv">7</span>,</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">          <span class="dt">comment =</span> plot_comment)</a></code></pre></div>
<p>The function <code>hypo_save()</code> is simply a wrapper around <code>ggsave()</code>, that will write the name of the currently running script into the meta data of the plot (after the plot has been exported).
The benfit of this is that you can read this information later to remember how a specifc plot was created using <code>hypo_show_metadata()</code>.
This is done using <a href="https://www.sno.phy.queensu.ca/~phil/exiftool/">exiftool</a> and has currently only been tested on my linux system.
If this does not work for you, simple replace <code>hypo_save()</code> with <code>ggsave()</code> and drop the <code>comment</code> parameter.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">hypo_show_metadata</span>(<span class="st">&#39;figures/F1.pdf&#39;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="co">#&gt; [1] &quot;mother-script = /path/to/scripts/plot_F1.R&quot;</span></a></code></pre></div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="analysis-iv-rho.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
