<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>16 Figure 4 | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="16 Figure 4 | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="16 Figure 4 | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench" />


<meta name="date" content="2020-06-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-3.html"/>
<link rel="next" href="figure-5.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a><ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a><ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a><ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a><ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a><ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a><ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i> &amp; pi)</a><ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a><ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (phylogeny &amp; topology weighting)</a><ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a><ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a><ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a><ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (poptrees)</a><ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#details-of-analysis_fst_poptree.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_fst_poptree.nf</code></a><ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#data-preparation-4"><i class="fa fa-check"></i><b>8.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a><ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a><ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-5"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a><ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a><ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-6"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a><ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a><ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-7"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-visualization.html"><a href="git-11-visualization.html"><i class="fa fa-check"></i><b>12</b> (git 11) Visualization</a></li>
<li class="chapter" data-level="13" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>13</b> Figure 1</a><ul>
<li class="chapter" data-level="13.1" data-path="figure-1.html"><a href="figure-1.html#summary-10"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>13.2</b> Details of <code>plot_F1.R</code></a><ul>
<li class="chapter" data-level="13.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>13.2.1</b> Config</a></li>
<li class="chapter" data-level="13.2.2" data-path="figure-1.html"><a href="figure-1.html#data-import"><i class="fa fa-check"></i><b>13.2.2</b> Data import</a></li>
<li class="chapter" data-level="13.2.3" data-path="figure-1.html"><a href="figure-1.html#plotting"><i class="fa fa-check"></i><b>13.2.3</b> Plotting</a></li>
<li class="chapter" data-level="13.2.4" data-path="figure-1.html"><a href="figure-1.html#table-export"><i class="fa fa-check"></i><b>13.2.4</b> Table export</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>14</b> Figure 2</a><ul>
<li class="chapter" data-level="14.1" data-path="figure-2.html"><a href="figure-2.html#summary-11"><i class="fa fa-check"></i><b>14.1</b> Summary</a></li>
<li class="chapter" data-level="14.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>14.2</b> Details of <code>plot_F2.R</code></a><ul>
<li class="chapter" data-level="14.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>14.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>15</b> Figure 3</a><ul>
<li class="chapter" data-level="15.1" data-path="figure-3.html"><a href="figure-3.html#summary-12"><i class="fa fa-check"></i><b>15.1</b> Summary</a></li>
<li class="chapter" data-level="15.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>15.2</b> Details of <code>plot_F3.R</code></a><ul>
<li class="chapter" data-level="15.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>15.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>16</b> Figure 4</a><ul>
<li class="chapter" data-level="16.1" data-path="figure-4.html"><a href="figure-4.html#summary-13"><i class="fa fa-check"></i><b>16.1</b> Summary</a></li>
<li class="chapter" data-level="16.2" data-path="figure-4.html"><a href="figure-4.html#details-of-plot_f4.r"><i class="fa fa-check"></i><b>16.2</b> Details of <code>plot_F4.R</code></a><ul>
<li class="chapter" data-level="16.2.1" data-path="figure-4.html"><a href="figure-4.html#config-3"><i class="fa fa-check"></i><b>16.2.1</b> Config</a></li>
<li class="chapter" data-level="16.2.2" data-path="figure-4.html"><a href="figure-4.html#data-import-1"><i class="fa fa-check"></i><b>16.2.2</b> Data import</a></li>
<li class="chapter" data-level="16.2.3" data-path="figure-4.html"><a href="figure-4.html#plotting-1"><i class="fa fa-check"></i><b>16.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>17</b> Figure 5</a><ul>
<li class="chapter" data-level="17.1" data-path="figure-5.html"><a href="figure-5.html#summary-14"><i class="fa fa-check"></i><b>17.1</b> Summary</a></li>
<li class="chapter" data-level="17.2" data-path="figure-5.html"><a href="figure-5.html#details-of-plot_f5.r"><i class="fa fa-check"></i><b>17.2</b> Details of <code>plot_F5.R</code></a><ul>
<li class="chapter" data-level="17.2.1" data-path="figure-5.html"><a href="figure-5.html#config-4"><i class="fa fa-check"></i><b>17.2.1</b> Config</a></li>
<li class="chapter" data-level="17.2.2" data-path="figure-5.html"><a href="figure-5.html#data-import-2"><i class="fa fa-check"></i><b>17.2.2</b> Data import</a></li>
<li class="chapter" data-level="17.2.3" data-path="figure-5.html"><a href="figure-5.html#plotting-2"><i class="fa fa-check"></i><b>17.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-4" class="section level1">
<h1><span class="header-section-number">16</span> Figure 4</h1>
<div id="summary-13" class="section level2">
<h2><span class="header-section-number">16.1</span> Summary</h2>
<p>This is the accessory documentation of Figure 4.
The Figure can be recreated by running the <strong>R</strong> script <code>plot_F4.R</code>:</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb57-1"><a href="figure-4.html#cb57-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span></span>
<span id="cb57-2"><a href="figure-4.html#cb57-2"></a></span>
<span id="cb57-3"><a href="figure-4.html#cb57-3"></a><span class="ex">Rscript</span> --vanilla R/fig/plot_F4.R 2_analysis/dxy/50k/ \</span>
<span id="cb57-4"><a href="figure-4.html#cb57-4"></a>   2_analysis/fst/50k/multi_fst.50k.tsv.gz 2_analysis/GxP/50000/ \</span>
<span id="cb57-5"><a href="figure-4.html#cb57-5"></a>   2_analysis/summaries/fst_outliers_998.tsv \</span>
<span id="cb57-6"><a href="figure-4.html#cb57-6"></a>   https://raw.githubusercontent.com/simonhmartin/twisst/master/plot_twisst.R \</span>
<span id="cb57-7"><a href="figure-4.html#cb57-7"></a>   2_analysis/twisst/weights/ ressources/plugin/trees/ \</span>
<span id="cb57-8"><a href="figure-4.html#cb57-8"></a>   2_analysis/fasteprr/step4/fasteprr.all.rho.txt.gz \</span>
<span id="cb57-9"><a href="figure-4.html#cb57-9"></a>   2_analysis/summaries/fst_globals.txt</span></code></pre></div>
</div>
<div id="details-of-plot_f4.r" class="section level2">
<h2><span class="header-section-number">16.2</span> Details of <code>plot_F4.R</code></h2>
<p>In the following, the individual steps of the R script are documented.
It is an executable R script that depends on the accessory R package <a href="https://k-hench.github.io/GenomicOriginsScripts"><strong>GenomicOriginsScripts</strong></a>.</p>
<div id="config-3" class="section level3">
<h3><span class="header-section-number">16.2.1</span> Config</h3>
<p>The scripts start with a header that contains copy &amp; paste templates to execute or debug the script:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="figure-4.html#cb58-1"></a><span class="co">#!/usr/bin/env Rscript</span></span>
<span id="cb58-2"><a href="figure-4.html#cb58-2"></a><span class="co"># run from terminal:</span></span>
<span id="cb58-3"><a href="figure-4.html#cb58-3"></a><span class="co"># Rscript --vanilla R/fig/plot_F4.R 2_analysis/dxy/50k/ \</span></span>
<span id="cb58-4"><a href="figure-4.html#cb58-4"></a><span class="co">#    2_analysis/fst/50k/multi_fst.50k.tsv.gz 2_analysis/GxP/50000/ \</span></span>
<span id="cb58-5"><a href="figure-4.html#cb58-5"></a><span class="co">#    2_analysis/summaries/fst_outliers_998.tsv \</span></span>
<span id="cb58-6"><a href="figure-4.html#cb58-6"></a><span class="co">#    https://raw.githubusercontent.com/simonhmartin/twisst/master/plot_twisst.R \</span></span>
<span id="cb58-7"><a href="figure-4.html#cb58-7"></a><span class="co">#    2_analysis/twisst/weights/ ressources/plugin/trees/ \</span></span>
<span id="cb58-8"><a href="figure-4.html#cb58-8"></a><span class="co">#    2_analysis/fasteprr/step4/fasteprr.all.rho.txt.gz \</span></span>
<span id="cb58-9"><a href="figure-4.html#cb58-9"></a><span class="co">#    2_analysis/summaries/fst_globals.txt</span></span>
<span id="cb58-10"><a href="figure-4.html#cb58-10"></a><span class="co"># ===============================================================</span></span>
<span id="cb58-11"><a href="figure-4.html#cb58-11"></a><span class="co"># This script produces Figure 4 of the study &quot;Ancestral variation, hybridization and modularity</span></span>
<span id="cb58-12"><a href="figure-4.html#cb58-12"></a><span class="co"># fuel a marine radiation&quot; by Hench, McMillan and Puebla</span></span>
<span id="cb58-13"><a href="figure-4.html#cb58-13"></a><span class="co"># ---------------------------------------------------------------</span></span>
<span id="cb58-14"><a href="figure-4.html#cb58-14"></a><span class="co"># ===============================================================</span></span>
<span id="cb58-15"><a href="figure-4.html#cb58-15"></a><span class="co"># args &lt;- c(&#39;2_analysis/dxy/50k/&#39;,&#39;2_analysis/fst/50k/multi_fst.50k.tsv.gz&#39;,</span></span>
<span id="cb58-16"><a href="figure-4.html#cb58-16"></a><span class="co"># &#39;2_analysis/GxP/50000/&#39;, &#39;2_analysis/summaries/fst_outliers_998.tsv&#39;,</span></span>
<span id="cb58-17"><a href="figure-4.html#cb58-17"></a><span class="co"># &#39;https://raw.githubusercontent.com/simonhmartin/twisst/master/plot_twisst.R&#39;,</span></span>
<span id="cb58-18"><a href="figure-4.html#cb58-18"></a><span class="co"># &#39;2_analysis/twisst/weights/&#39;, &#39;ressources/plugin/trees/&#39;,</span></span>
<span id="cb58-19"><a href="figure-4.html#cb58-19"></a><span class="co"># &#39;2_analysis/fasteprr/step4/fasteprr.all.rho.txt.gz&#39;, &#39;2_analysis/summaries/fst_globals.txt&#39;)</span></span>
<span id="cb58-20"><a href="figure-4.html#cb58-20"></a><span class="co"># script_name &lt;- &quot;R/fig/plot_F4.R&quot;</span></span></code></pre></div>
<p>The next section processes the input from the command line.
It stores the arguments in the vector <code>args</code>.
The R package <a href="https://k-hench.github.io/GenomicOriginsScripts"><strong>GenomicOriginsScripts</strong></a> is loaded and the script name and the current working directory are stored inside variables (<code>script_name</code>, <code>plot_comment</code>).
This information will later be written into the meta data of the figure to help us tracing back the scripts that created the figures in the future.</p>
<p>Then we drop all the imported information besides the arguments following the script name and print the information to the terminal.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="figure-4.html#cb59-1"></a>args &lt;-<span class="st"> </span><span class="kw">commandArgs</span>(<span class="dt">trailingOnly =</span> <span class="ot">FALSE</span>)</span>
<span id="cb59-2"><a href="figure-4.html#cb59-2"></a><span class="co"># setup -----------------------</span></span>
<span id="cb59-3"><a href="figure-4.html#cb59-3"></a><span class="kw">library</span>(GenomicOriginsScripts)</span>
<span id="cb59-4"><a href="figure-4.html#cb59-4"></a><span class="kw">library</span>(hypoimg)</span>
<span id="cb59-5"><a href="figure-4.html#cb59-5"></a></span>
<span id="cb59-6"><a href="figure-4.html#cb59-6"></a><span class="kw">cat</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>)</span>
<span id="cb59-7"><a href="figure-4.html#cb59-7"></a>script_name &lt;-<span class="st"> </span>args[<span class="dv">5</span>] <span class="op">%&gt;%</span></span>
<span id="cb59-8"><a href="figure-4.html#cb59-8"></a><span class="st">  </span><span class="kw">str_remove</span>(.,<span class="st">&#39;--file=&#39;</span>)</span>
<span id="cb59-9"><a href="figure-4.html#cb59-9"></a></span>
<span id="cb59-10"><a href="figure-4.html#cb59-10"></a>plot_comment &lt;-<span class="st"> </span>script_name <span class="op">%&gt;%</span></span>
<span id="cb59-11"><a href="figure-4.html#cb59-11"></a><span class="st">  </span><span class="kw">str_c</span>(<span class="st">&#39;mother-script = &#39;</span>,<span class="kw">getwd</span>(),<span class="st">&#39;/&#39;</span>,.)</span>
<span id="cb59-12"><a href="figure-4.html#cb59-12"></a></span>
<span id="cb59-13"><a href="figure-4.html#cb59-13"></a>args &lt;-<span class="st"> </span><span class="kw">process_input</span>(script_name, args)</span></code></pre></div>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="figure-4.html#cb60-1"></a><span class="co">#&gt; ── Script: scripts/plot_F4.R ────────────────────────────────────────────</span></span>
<span id="cb60-2"><a href="figure-4.html#cb60-2"></a><span class="co">#&gt; Parameters read:</span></span>
<span id="cb60-3"><a href="figure-4.html#cb60-3"></a><span class="co">#&gt; ★ 1: 2_analysis/dxy/50k/</span></span>
<span id="cb60-4"><a href="figure-4.html#cb60-4"></a><span class="co">#&gt; ★ 2: 2_analysis/fst/50k/multi_fst.50k.tsv.gz</span></span>
<span id="cb60-5"><a href="figure-4.html#cb60-5"></a><span class="co">#&gt; ★ 3: 2_analysis/GxP/50000/</span></span>
<span id="cb60-6"><a href="figure-4.html#cb60-6"></a><span class="co">#&gt; ★ 4: 2_analysis/summaries/fst_outliers_998.tsv</span></span>
<span id="cb60-7"><a href="figure-4.html#cb60-7"></a><span class="co">#&gt; ★ 5: https://raw.githubusercontent.com/simonhmartin/twisst/master/plot_twisst.R</span></span>
<span id="cb60-8"><a href="figure-4.html#cb60-8"></a><span class="co">#&gt; ★ 6: 2_analysis/twisst/weights/</span></span>
<span id="cb60-9"><a href="figure-4.html#cb60-9"></a><span class="co">#&gt; ★ 7: ressources/plugin/trees/</span></span>
<span id="cb60-10"><a href="figure-4.html#cb60-10"></a><span class="co">#&gt; ★ 8: 2_analysis/fasteprr/step4/fasteprr.all.rho.txt.gz</span></span>
<span id="cb60-11"><a href="figure-4.html#cb60-11"></a><span class="co">#&gt; ★ 9: 2_analysis/summaries/fst_globals.txt</span></span>
<span id="cb60-12"><a href="figure-4.html#cb60-12"></a><span class="co">#&gt; ─────────────────────────────────────────── /current/working/directory ──</span></span></code></pre></div>
<p>The directories for the different data types are received and stored in respective variables.
Also, we source an external r script from the <a href="https://github.com/simonhmartin/twisst">original twisst github repository</a> that we need to import the twisst data:</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="figure-4.html#cb61-1"></a><span class="co"># config -----------------------</span></span>
<span id="cb61-2"><a href="figure-4.html#cb61-2"></a>dxy_dir &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">1</span>])</span>
<span id="cb61-3"><a href="figure-4.html#cb61-3"></a>fst_file &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">2</span>])</span>
<span id="cb61-4"><a href="figure-4.html#cb61-4"></a>gxp_dir &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">3</span>])</span>
<span id="cb61-5"><a href="figure-4.html#cb61-5"></a>outlier_table &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">4</span>])</span>
<span id="cb61-6"><a href="figure-4.html#cb61-6"></a>twisst_script &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">5</span>])</span>
<span id="cb61-7"><a href="figure-4.html#cb61-7"></a>w_path &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">6</span>])</span>
<span id="cb61-8"><a href="figure-4.html#cb61-8"></a>d_path &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">7</span>])</span>
<span id="cb61-9"><a href="figure-4.html#cb61-9"></a>recombination_file &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">8</span>])</span>
<span id="cb61-10"><a href="figure-4.html#cb61-10"></a>global_fst_file &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">9</span>])</span>
<span id="cb61-11"><a href="figure-4.html#cb61-11"></a><span class="kw">source</span>(twisst_script)</span></code></pre></div>
</div>
<div id="data-import-1" class="section level3">
<h3><span class="header-section-number">16.2.2</span> Data import</h3>
<p>Supplementary Figure 4 contains quite a lot of different data sets.
The main part of this script is just importing and organizing all of this data:
In the following we’ll go step by step through the import of:</p>
<ul>
<li>differentiation data (<span class="math inline">\(F_{ST}\)</span>)</li>
<li>divergence data (<span class="math inline">\(d_{XY}\)</span>, also containing diversity data - <span class="math inline">\(\pi\)</span>)</li>
<li>genotype <span class="math inline">\(\times\)</span> phenotype association data (<span class="math inline">\(p_{Wald}\)</span>)</li>
<li>recombination data (<span class="math inline">\(\rho\)</span>)</li>
<li>topology weighting data</li>
</ul>
<p>We start with the import of the <span class="math inline">\(F_{ST}\)</span> data, specifically the data set containing the genome wide <span class="math inline">\(F_{ST}\)</span> computed for all populations simultaneously (joint <span class="math inline">\(F_{ST}\)</span>).</p>
<p>The data file is read, the columns are renamed and the genomic position is added.
Then, only the genomic position and the <span class="math inline">\(F_{ST}\)</span> columns are selected and a window column is added for faceting in <code>ggplot()</code>.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="figure-4.html#cb62-1"></a><span class="co"># start script -------------------</span></span>
<span id="cb62-2"><a href="figure-4.html#cb62-2"></a><span class="co"># import fst data</span></span>
<span id="cb62-3"><a href="figure-4.html#cb62-3"></a>fst_data &lt;-<span class="st"> </span>vroom<span class="op">::</span><span class="kw">vroom</span>(fst_file, <span class="dt">delim =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb62-4"><a href="figure-4.html#cb62-4"></a><span class="st">  </span><span class="kw">select</span>(CHROM, BIN_START, BIN_END, N_VARIANTS, WEIGHTED_FST) <span class="op">%&gt;%</span></span>
<span id="cb62-5"><a href="figure-4.html#cb62-5"></a><span class="st">  </span><span class="kw">setNames</span>(., <span class="dt">nm =</span> <span class="kw">c</span>(<span class="st">&#39;CHROM&#39;</span>, <span class="st">&#39;BIN_START&#39;</span>, <span class="st">&#39;BIN_END&#39;</span>, <span class="st">&#39;n_snps&#39;</span>, <span class="st">&#39;fst&#39;</span>) ) <span class="op">%&gt;%</span></span>
<span id="cb62-6"><a href="figure-4.html#cb62-6"></a><span class="st">  </span><span class="kw">add_gpos</span>() <span class="op">%&gt;%</span></span>
<span id="cb62-7"><a href="figure-4.html#cb62-7"></a><span class="st">  </span><span class="kw">select</span>(GPOS, fst) <span class="op">%&gt;%</span></span>
<span id="cb62-8"><a href="figure-4.html#cb62-8"></a><span class="st">  </span><span class="kw">setNames</span>(., <span class="dt">nm =</span> <span class="kw">c</span>(<span class="st">&#39;GPOS&#39;</span>,<span class="st">&#39;value&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb62-9"><a href="figure-4.html#cb62-9"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">window =</span> <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;a&#39;</span>),<span class="st">&#39;):joint~italic(F[ST])&#39;</span>))</span></code></pre></div>
<p>Next, we import the <span class="math inline">\(d_{XY}\)</span> data.
Here we are importing all 28 pairwise comparisons, so we first collect all the file paths and the iterate the data import over all files.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="figure-4.html#cb63-1"></a><span class="co"># locate dxy data files</span></span>
<span id="cb63-2"><a href="figure-4.html#cb63-2"></a>dxy_files &lt;-<span class="st"> </span><span class="kw">dir</span>(dxy_dir)</span>
<span id="cb63-3"><a href="figure-4.html#cb63-3"></a></span>
<span id="cb63-4"><a href="figure-4.html#cb63-4"></a><span class="co"># import dxy data</span></span>
<span id="cb63-5"><a href="figure-4.html#cb63-5"></a>dxy_data &lt;-<span class="st">  </span><span class="kw">str_c</span>(dxy_dir,dxy_files) <span class="op">%&gt;%</span></span>
<span id="cb63-6"><a href="figure-4.html#cb63-6"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">map</span>(get_dxy) <span class="op">%&gt;%</span></span>
<span id="cb63-7"><a href="figure-4.html#cb63-7"></a><span class="st">  </span><span class="kw">bind_rows</span>() <span class="op">%&gt;%</span></span>
<span id="cb63-8"><a href="figure-4.html#cb63-8"></a><span class="st">  </span><span class="kw">select</span>(N_SITES<span class="op">:</span>GPOS, run) <span class="op">%&gt;%</span></span>
<span id="cb63-9"><a href="figure-4.html#cb63-9"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">pop1 =</span> <span class="kw">str_sub</span>(run,<span class="dv">1</span>,<span class="dv">6</span>),</span>
<span id="cb63-10"><a href="figure-4.html#cb63-10"></a>         <span class="dt">pop2 =</span> <span class="kw">str_sub</span>(run,<span class="dv">8</span>,<span class="dv">13</span>))</span></code></pre></div>
<p>From this data, we compute the divergence difference (<span class="math inline">\(\Delta d_{XY}\)</span>).</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="figure-4.html#cb64-1"></a><span class="co"># compute delta dxy</span></span>
<span id="cb64-2"><a href="figure-4.html#cb64-2"></a>dxy_summary &lt;-<span class="st"> </span>dxy_data <span class="op">%&gt;%</span></span>
<span id="cb64-3"><a href="figure-4.html#cb64-3"></a><span class="st">  </span><span class="kw">group_by</span>(GPOS) <span class="op">%&gt;%</span></span>
<span id="cb64-4"><a href="figure-4.html#cb64-4"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">delta_dxy =</span> <span class="kw">max</span>(dxy)<span class="op">-</span><span class="kw">min</span>(dxy),</span>
<span id="cb64-5"><a href="figure-4.html#cb64-5"></a>            <span class="dt">sd_dxy =</span> <span class="kw">sd</span>(dxy),</span>
<span id="cb64-6"><a href="figure-4.html#cb64-6"></a>            <span class="dt">delt_pi =</span> <span class="kw">max</span>(<span class="kw">c</span>(<span class="kw">max</span>(PI_POP1),<span class="kw">max</span>(PI_POP2))) <span class="op">-</span><span class="st"> </span><span class="kw">min</span>(<span class="kw">c</span>(<span class="kw">min</span>(PI_POP1),<span class="kw">min</span>(PI_POP2)))) <span class="op">%&gt;%</span></span>
<span id="cb64-7"><a href="figure-4.html#cb64-7"></a><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></span>
<span id="cb64-8"><a href="figure-4.html#cb64-8"></a><span class="st">  </span><span class="kw">setNames</span>(., <span class="dt">nm =</span> <span class="kw">c</span>(<span class="st">&#39;GPOS&#39;</span>,</span>
<span id="cb64-9"><a href="figure-4.html#cb64-9"></a>                     <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;e&#39;</span>),<span class="st">&#39;):\u0394~italic(d[xy])&#39;</span>),</span>
<span id="cb64-10"><a href="figure-4.html#cb64-10"></a>                     <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;e&#39;</span>),<span class="st">&#39;):italic(d[xy])~(sd)&#39;</span>),</span>
<span id="cb64-11"><a href="figure-4.html#cb64-11"></a>                     <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;e&#39;</span>),<span class="st">&#39;):\u0394~italic(\u03C0)&#39;</span>))) <span class="op">%&gt;%</span></span>
<span id="cb64-12"><a href="figure-4.html#cb64-12"></a><span class="st">  </span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&#39;window&#39;</span>, <span class="dt">value =</span> <span class="st">&#39;value&#39;</span>,<span class="dv">2</span><span class="op">:</span><span class="dv">4</span>) <span class="op">%&gt;%</span></span>
<span id="cb64-13"><a href="figure-4.html#cb64-13"></a><span class="st">  </span><span class="kw">filter</span>(window <span class="op">==</span><span class="st"> </span><span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;e&#39;</span>),<span class="st">&#39;):\u0394~italic(d[xy])&#39;</span>))</span></code></pre></div>
<p>Then we import the genotype <span class="math inline">\(\times\)</span> phenotype association data.
For this, we list all the traits we want to include and then iterate the import function over all traits.
We combine the data sets and transform the table to <em>long</em> format.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="figure-4.html#cb65-1"></a><span class="co"># set G x P traits to be imported</span></span>
<span id="cb65-2"><a href="figure-4.html#cb65-2"></a>traits &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Bars.lm.50k.5k.txt.gz&quot;</span>, <span class="st">&quot;Peduncle.lm.50k.5k.txt.gz&quot;</span>, <span class="st">&quot;Snout.lm.50k.5k.txt.gz&quot;</span>)</span>
<span id="cb65-3"><a href="figure-4.html#cb65-3"></a></span>
<span id="cb65-4"><a href="figure-4.html#cb65-4"></a><span class="co"># set trait figure panels</span></span>
<span id="cb65-5"><a href="figure-4.html#cb65-5"></a>trait_panels &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">Bars =</span> <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;h&#39;</span>),<span class="st">&#39;)&#39;</span>),</span>
<span id="cb65-6"><a href="figure-4.html#cb65-6"></a>                  <span class="dt">Peduncle =</span> <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;i&#39;</span>),<span class="st">&#39;)&#39;</span>),</span>
<span id="cb65-7"><a href="figure-4.html#cb65-7"></a>                  <span class="dt">Snout =</span> <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;j&#39;</span>),<span class="st">&#39;)&#39;</span>))</span>
<span id="cb65-8"><a href="figure-4.html#cb65-8"></a></span>
<span id="cb65-9"><a href="figure-4.html#cb65-9"></a><span class="co"># import G x P data</span></span>
<span id="cb65-10"><a href="figure-4.html#cb65-10"></a>gxp_data &lt;-<span class="st"> </span><span class="kw">str_c</span>(gxp_dir,traits) <span class="op">%&gt;%</span></span>
<span id="cb65-11"><a href="figure-4.html#cb65-11"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">map</span>(get_gxp) <span class="op">%&gt;%</span></span>
<span id="cb65-12"><a href="figure-4.html#cb65-12"></a><span class="st">  </span><span class="kw">join_list</span>() <span class="op">%&gt;%</span></span>
<span id="cb65-13"><a href="figure-4.html#cb65-13"></a><span class="st">  </span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&#39;window&#39;</span>, <span class="dt">value =</span> <span class="st">&#39;value&#39;</span>,<span class="dv">2</span><span class="op">:</span><span class="dv">4</span>)</span></code></pre></div>
<p>Then, we import the genome wide <span class="math inline">\(F_{ST}\)</span> summary for all 28 pair wise comparisons to be able to pick a divergence data set of an intermediately differentiated species pair (the species pair of rank 15, close to 14.5 - the median of rank 1 to 28).</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="figure-4.html#cb66-1"></a><span class="co"># import genome wide Fst data summary  --------</span></span>
<span id="cb66-2"><a href="figure-4.html#cb66-2"></a>globals &lt;-<span class="st"> </span>vroom<span class="op">::</span><span class="kw">vroom</span>(global_fst_file, <span class="dt">delim =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>,</span>
<span id="cb66-3"><a href="figure-4.html#cb66-3"></a>                        <span class="dt">col_names =</span> <span class="kw">c</span>(<span class="st">&#39;loc&#39;</span>,<span class="st">&#39;run&#39;</span>,<span class="st">&#39;mean&#39;</span>,<span class="st">&#39;weighted&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb66-4"><a href="figure-4.html#cb66-4"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">run =</span> <span class="kw">str_c</span>(<span class="kw">str_sub</span>(run,<span class="dv">1</span>,<span class="dv">3</span>),loc,<span class="st">&#39;-&#39;</span>,<span class="kw">str_sub</span>(run,<span class="dv">5</span>,<span class="dv">7</span>),loc),</span>
<span id="cb66-5"><a href="figure-4.html#cb66-5"></a>         <span class="dt">run =</span> <span class="kw">fct_reorder</span>(run,weighted))</span>
<span id="cb66-6"><a href="figure-4.html#cb66-6"></a></span>
<span id="cb66-7"><a href="figure-4.html#cb66-7"></a><span class="co"># dxy and pi are only shown for one exemplary population (/pair)</span></span>
<span id="cb66-8"><a href="figure-4.html#cb66-8"></a><span class="co"># select dxy pair run (15 is one of the two central runs of the 28 pairs)</span></span>
<span id="cb66-9"><a href="figure-4.html#cb66-9"></a><span class="co"># here, the 15th lowest fst value is identified as &quot;selector&quot;</span></span>
<span id="cb66-10"><a href="figure-4.html#cb66-10"></a>selectors_dxy &lt;-<span class="st"> </span>globals <span class="op">%&gt;%</span></span>
<span id="cb66-11"><a href="figure-4.html#cb66-11"></a><span class="st">  </span><span class="kw">arrange</span>(weighted) <span class="op">%&gt;%</span></span>
<span id="cb66-12"><a href="figure-4.html#cb66-12"></a><span class="st">  </span>.<span class="op">$</span>weighted <span class="op">%&gt;%</span></span>
<span id="cb66-13"><a href="figure-4.html#cb66-13"></a><span class="st">  </span>.[<span class="dv">15</span>]</span>
<span id="cb66-14"><a href="figure-4.html#cb66-14"></a></span>
<span id="cb66-15"><a href="figure-4.html#cb66-15"></a><span class="co"># the dxy population pair corresponding to the selector is identified</span></span>
<span id="cb66-16"><a href="figure-4.html#cb66-16"></a>select_dxy_runs &lt;-<span class="st"> </span>globals <span class="op">%&gt;%</span></span>
<span id="cb66-17"><a href="figure-4.html#cb66-17"></a><span class="st">  </span><span class="kw">filter</span>(weighted <span class="op">%in%</span><span class="st"> </span>selectors_dxy) <span class="op">%&gt;%</span></span>
<span id="cb66-18"><a href="figure-4.html#cb66-18"></a><span class="st">  </span>.<span class="op">$</span>run <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.character</span>()</span>
<span id="cb66-19"><a href="figure-4.html#cb66-19"></a></span>
<span id="cb66-20"><a href="figure-4.html#cb66-20"></a><span class="co"># then thne dxy data is subset based on the selector</span></span>
<span id="cb66-21"><a href="figure-4.html#cb66-21"></a>dxy_select &lt;-<span class="st"> </span>dxy_data <span class="op">%&gt;%</span></span>
<span id="cb66-22"><a href="figure-4.html#cb66-22"></a><span class="st">  </span><span class="kw">filter</span>(run <span class="op">%in%</span><span class="st"> </span>select_dxy_runs) <span class="op">%&gt;%</span></span>
<span id="cb66-23"><a href="figure-4.html#cb66-23"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">window =</span> <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;b&#39;</span>),<span class="st">&#39;): italic(d[XY])&#39;</span>))</span></code></pre></div>
<p>The <span class="math inline">\(d_{XY}\)</span> data set includes also <span class="math inline">\(\pi\)</span> of the involved populations.
We first extract the diversity data for each population (pop1 &amp; pop2), combine them and compute the statistics needed for ranking the populations based on their diversity.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="figure-4.html#cb67-1"></a><span class="co"># the pi data is filtered on a similar logic to the dxy data</span></span>
<span id="cb67-2"><a href="figure-4.html#cb67-2"></a><span class="co"># first a table with the genome wide average pi for each population is compiled</span></span>
<span id="cb67-3"><a href="figure-4.html#cb67-3"></a><span class="co"># (based on the first populations from the dxy data table</span></span>
<span id="cb67-4"><a href="figure-4.html#cb67-4"></a><span class="co"># which contains pi for both populations)</span></span>
<span id="cb67-5"><a href="figure-4.html#cb67-5"></a>pi_summary_<span class="dv">1</span> &lt;-<span class="st"> </span>dxy_data <span class="op">%&gt;%</span></span>
<span id="cb67-6"><a href="figure-4.html#cb67-6"></a><span class="st">  </span><span class="kw">group_by</span>(pop1,run) <span class="op">%&gt;%</span></span>
<span id="cb67-7"><a href="figure-4.html#cb67-7"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">avg_pi =</span> <span class="kw">mean</span>(PI_POP1)) <span class="op">%&gt;%</span></span>
<span id="cb67-8"><a href="figure-4.html#cb67-8"></a><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></span>
<span id="cb67-9"><a href="figure-4.html#cb67-9"></a><span class="st">  </span><span class="kw">set_names</span>(., <span class="dt">nm =</span> <span class="kw">c</span>(<span class="st">&#39;pop&#39;</span>,<span class="st">&#39;run&#39;</span>,<span class="st">&#39;avg_pi&#39;</span>))</span>
<span id="cb67-10"><a href="figure-4.html#cb67-10"></a></span>
<span id="cb67-11"><a href="figure-4.html#cb67-11"></a><span class="co"># the mean genome wide average pi is compiled for all the second populations</span></span>
<span id="cb67-12"><a href="figure-4.html#cb67-12"></a><span class="co"># from the dxy data</span></span>
<span id="cb67-13"><a href="figure-4.html#cb67-13"></a><span class="co"># then, the average of all the comparisons is computed for each population</span></span>
<span id="cb67-14"><a href="figure-4.html#cb67-14"></a>pi_summary &lt;-<span class="st"> </span>dxy_data <span class="op">%&gt;%</span></span>
<span id="cb67-15"><a href="figure-4.html#cb67-15"></a><span class="st">  </span><span class="kw">group_by</span>(pop2,run) <span class="op">%&gt;%</span></span>
<span id="cb67-16"><a href="figure-4.html#cb67-16"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">avg_pi =</span> <span class="kw">mean</span>(PI_POP2)) <span class="op">%&gt;%</span></span>
<span id="cb67-17"><a href="figure-4.html#cb67-17"></a><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></span>
<span id="cb67-18"><a href="figure-4.html#cb67-18"></a><span class="st">  </span><span class="kw">set_names</span>(., <span class="dt">nm =</span> <span class="kw">c</span>(<span class="st">&#39;pop&#39;</span>,<span class="st">&#39;run&#39;</span>,<span class="st">&#39;avg_pi&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb67-19"><a href="figure-4.html#cb67-19"></a><span class="st">  </span><span class="kw">bind_rows</span>(pi_summary_<span class="dv">1</span>)  <span class="op">%&gt;%</span></span>
<span id="cb67-20"><a href="figure-4.html#cb67-20"></a><span class="st">  </span><span class="kw">group_by</span>(pop) <span class="op">%&gt;%</span></span>
<span id="cb67-21"><a href="figure-4.html#cb67-21"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">n =</span> <span class="kw">length</span>(pop),</span>
<span id="cb67-22"><a href="figure-4.html#cb67-22"></a>            <span class="dt">mean_pi =</span> <span class="kw">mean</span>(avg_pi),</span>
<span id="cb67-23"><a href="figure-4.html#cb67-23"></a>            <span class="dt">min_pi =</span> <span class="kw">min</span>(avg_pi),</span>
<span id="cb67-24"><a href="figure-4.html#cb67-24"></a>            <span class="dt">max_pi =</span> <span class="kw">max</span>(avg_pi),</span>
<span id="cb67-25"><a href="figure-4.html#cb67-25"></a>            <span class="dt">sd_pi =</span> <span class="kw">sd</span>(avg_pi)) <span class="op">%&gt;%</span></span>
<span id="cb67-26"><a href="figure-4.html#cb67-26"></a><span class="st">  </span><span class="kw">arrange</span>(n)</span></code></pre></div>
<p>Then, we determine an intermediately diverse candidate of our 14 populations (rank 7, again: <span class="math inline">\(7 \approx median(1:14)\)</span>) and average over the diversities estimated in all pairwise comparisons this population was involved in.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="figure-4.html#cb68-1"></a><span class="co"># one of the central populations with respect to average genome</span></span>
<span id="cb68-2"><a href="figure-4.html#cb68-2"></a><span class="co"># wide pi is identified</span></span>
<span id="cb68-3"><a href="figure-4.html#cb68-3"></a><span class="co"># for this, the 7th lowest pi value of the 14 populations is</span></span>
<span id="cb68-4"><a href="figure-4.html#cb68-4"></a><span class="co"># determined as &quot;selector&quot;</span></span>
<span id="cb68-5"><a href="figure-4.html#cb68-5"></a>selectors_pi &lt;-<span class="st"> </span>pi_summary <span class="op">%&gt;%</span></span>
<span id="cb68-6"><a href="figure-4.html#cb68-6"></a><span class="st">  </span>.<span class="op">$</span>mean_pi <span class="op">%&gt;%</span></span>
<span id="cb68-7"><a href="figure-4.html#cb68-7"></a><span class="st">  </span><span class="kw">sort</span>() <span class="op">%&gt;%</span></span>
<span id="cb68-8"><a href="figure-4.html#cb68-8"></a><span class="st">  </span>.[<span class="dv">7</span>]</span>
<span id="cb68-9"><a href="figure-4.html#cb68-9"></a></span>
<span id="cb68-10"><a href="figure-4.html#cb68-10"></a><span class="co"># the respective population is identified</span></span>
<span id="cb68-11"><a href="figure-4.html#cb68-11"></a>select_pi_pops &lt;-<span class="st"> </span>pi_summary <span class="op">%&gt;%</span></span>
<span id="cb68-12"><a href="figure-4.html#cb68-12"></a><span class="st">  </span><span class="kw">filter</span>(mean_pi <span class="op">%in%</span><span class="st"> </span>selectors_pi) <span class="op">%&gt;%</span></span>
<span id="cb68-13"><a href="figure-4.html#cb68-13"></a><span class="st">  </span>.<span class="op">$</span>pop <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.character</span>()</span>
<span id="cb68-14"><a href="figure-4.html#cb68-14"></a></span>
<span id="cb68-15"><a href="figure-4.html#cb68-15"></a><span class="co"># then the dxy data is subset by that population and the average pi over</span></span>
<span id="cb68-16"><a href="figure-4.html#cb68-16"></a><span class="co"># all pair-wise runs is calculated for each window</span></span>
<span id="cb68-17"><a href="figure-4.html#cb68-17"></a>pi_data_select &lt;-<span class="st"> </span>dxy_data <span class="op">%&gt;%</span></span>
<span id="cb68-18"><a href="figure-4.html#cb68-18"></a><span class="st">  </span><span class="kw">select</span>(GPOS, PI_POP1, pop1 )<span class="op">%&gt;%</span></span>
<span id="cb68-19"><a href="figure-4.html#cb68-19"></a><span class="st">  </span><span class="kw">set_names</span>(., <span class="dt">nm =</span> <span class="kw">c</span>(<span class="st">&#39;GPOS&#39;</span>,<span class="st">&#39;pi&#39;</span>,<span class="st">&#39;pop&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb68-20"><a href="figure-4.html#cb68-20"></a><span class="st">  </span><span class="kw">bind_rows</span>(.,dxy_data <span class="op">%&gt;%</span></span>
<span id="cb68-21"><a href="figure-4.html#cb68-21"></a><span class="st">              </span><span class="kw">select</span>(GPOS, PI_POP2, pop2 )<span class="op">%&gt;%</span></span>
<span id="cb68-22"><a href="figure-4.html#cb68-22"></a><span class="st">              </span><span class="kw">set_names</span>(., <span class="dt">nm =</span> <span class="kw">c</span>(<span class="st">&#39;GPOS&#39;</span>,<span class="st">&#39;pi&#39;</span>,<span class="st">&#39;pop&#39;</span>))) <span class="op">%&gt;%</span></span>
<span id="cb68-23"><a href="figure-4.html#cb68-23"></a><span class="st">  </span><span class="kw">group_by</span>(GPOS,pop) <span class="op">%&gt;%</span></span>
<span id="cb68-24"><a href="figure-4.html#cb68-24"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">n =</span> <span class="kw">length</span>(pop),</span>
<span id="cb68-25"><a href="figure-4.html#cb68-25"></a>            <span class="dt">mean_pi =</span> <span class="kw">mean</span>(pi),</span>
<span id="cb68-26"><a href="figure-4.html#cb68-26"></a>            <span class="dt">min_pi =</span> <span class="kw">min</span>(pi),</span>
<span id="cb68-27"><a href="figure-4.html#cb68-27"></a>            <span class="dt">max_pi =</span> <span class="kw">max</span>(pi),</span>
<span id="cb68-28"><a href="figure-4.html#cb68-28"></a>            <span class="dt">sd_pi =</span> <span class="kw">sd</span>(pi)) <span class="op">%&gt;%</span></span>
<span id="cb68-29"><a href="figure-4.html#cb68-29"></a><span class="st">  </span><span class="kw">filter</span>(pop <span class="op">%in%</span><span class="st"> </span>select_pi_pops) <span class="op">%&gt;%</span></span>
<span id="cb68-30"><a href="figure-4.html#cb68-30"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">window =</span> <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;c&#39;</span>),<span class="st">&#39;):~\u03C0&#39;</span>))</span></code></pre></div>
<p>The import of the recombination data is pretty straight forward:
Reading one file, adding genomic position and window column for faceting.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="figure-4.html#cb69-1"></a><span class="co"># import recombination data</span></span>
<span id="cb69-2"><a href="figure-4.html#cb69-2"></a>recombination_data &lt;-<span class="st"> </span>vroom<span class="op">::</span><span class="kw">vroom</span>(recombination_file,<span class="dt">delim =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb69-3"><a href="figure-4.html#cb69-3"></a><span class="st">  </span><span class="kw">add_gpos</span>() <span class="op">%&gt;%</span></span>
<span id="cb69-4"><a href="figure-4.html#cb69-4"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">window =</span> <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;d&#39;</span>),<span class="st">&#39;):~\u03C1&#39;</span>))</span></code></pre></div>
<p>Then we import the topology weighting data.
This is done once per location, the data sets are combined and specific columns are selected:
The gnomic position, the topology number (format: three digits with leading zeros, hence “topo3”), relative topology rank ranging from 0 to 1, the faceting column and the actual weight data.</p>
<p>We also create a dummy tibble that contains the null expectation of the topology weight for the two locations (1/n, with n = number of possible topologies - n = 15 for Belize and 105 for Honduras).</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="figure-4.html#cb70-1"></a><span class="co"># import topology weighting data</span></span>
<span id="cb70-2"><a href="figure-4.html#cb70-2"></a>twisst_data &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">loc =</span> <span class="kw">c</span>(<span class="st">&#39;bel&#39;</span>,<span class="st">&#39;hon&#39;</span>),</span>
<span id="cb70-3"><a href="figure-4.html#cb70-3"></a>                      <span class="dt">panel =</span> <span class="kw">c</span>(<span class="st">&#39;f&#39;</span>,<span class="st">&#39;g&#39;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">project_case</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,.,<span class="st">&#39;)&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb70-4"><a href="figure-4.html#cb70-4"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">pmap</span>(match_twisst_files) <span class="op">%&gt;%</span></span>
<span id="cb70-5"><a href="figure-4.html#cb70-5"></a><span class="st">  </span><span class="kw">bind_rows</span>() <span class="op">%&gt;%</span></span>
<span id="cb70-6"><a href="figure-4.html#cb70-6"></a><span class="st">  </span><span class="kw">select</span>(GPOS, topo3,topo_rel,window,weight)</span>
<span id="cb70-7"><a href="figure-4.html#cb70-7"></a></span>
<span id="cb70-8"><a href="figure-4.html#cb70-8"></a><span class="co"># the &quot;null-weighting&quot; is computed for both locations</span></span>
<span id="cb70-9"><a href="figure-4.html#cb70-9"></a>twisst_null &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">window =</span> <span class="kw">c</span>(<span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;f&#39;</span>),<span class="st">&#39;):~italic(w)[bel]&#39;</span>),</span>
<span id="cb70-10"><a href="figure-4.html#cb70-10"></a>                                 <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;g&#39;</span>),<span class="st">&#39;):~italic(w)[hon]&#39;</span>)),</span>
<span id="cb70-11"><a href="figure-4.html#cb70-11"></a>                      <span class="dt">weight =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">/</span><span class="dv">15</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">105</span>))</span></code></pre></div>
<p>We create a single data set for <span class="math inline">\(d_{XY}\)</span>, <span class="math inline">\(F_{ST}\)</span> and genotype <span class="math inline">\(\times\)</span> phenotype data.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="figure-4.html#cb71-1"></a><span class="co"># combine data types --------</span></span>
<span id="cb71-2"><a href="figure-4.html#cb71-2"></a>data &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(dxy_summary, fst_data, gxp_data)</span></code></pre></div>
<p>Then we load the positions of the the <span class="math inline">\(F_{ST}\)</span> outlier windows, select the focal outliers that will receive individual labels and create a tibble and two parameters for the label placement within the plot.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="figure-4.html#cb72-1"></a><span class="co"># import fst outliers</span></span>
<span id="cb72-2"><a href="figure-4.html#cb72-2"></a>outliers &lt;-<span class="st">  </span>vroom<span class="op">::</span><span class="kw">vroom</span>(outlier_table, <span class="dt">delim =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>)</span>
<span id="cb72-3"><a href="figure-4.html#cb72-3"></a></span>
<span id="cb72-4"><a href="figure-4.html#cb72-4"></a><span class="co"># the focal outlier IDs are set</span></span>
<span id="cb72-5"><a href="figure-4.html#cb72-5"></a>outlier_pick &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;LG04_1&#39;</span>, <span class="st">&#39;LG12_3&#39;</span>, <span class="st">&#39;LG12_4&#39;</span>)</span>
<span id="cb72-6"><a href="figure-4.html#cb72-6"></a></span>
<span id="cb72-7"><a href="figure-4.html#cb72-7"></a><span class="co"># the table for the outlier labels is created</span></span>
<span id="cb72-8"><a href="figure-4.html#cb72-8"></a>outlier_label &lt;-<span class="st"> </span>outliers <span class="op">%&gt;%</span></span>
<span id="cb72-9"><a href="figure-4.html#cb72-9"></a><span class="st">  </span><span class="kw">filter</span>(gid <span class="op">%in%</span><span class="st"> </span>outlier_pick) <span class="op">%&gt;%</span></span>
<span id="cb72-10"><a href="figure-4.html#cb72-10"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">label =</span> letters[<span class="kw">row_number</span>()] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">project_inv_case</span>(),</span>
<span id="cb72-11"><a href="figure-4.html#cb72-11"></a>         <span class="dt">x_shift_label =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="op">-</span><span class="fl">1.2</span>,<span class="dv">1</span>)<span class="op">*</span><span class="dv">10</span><span class="op">^</span><span class="dv">7</span>,</span>
<span id="cb72-12"><a href="figure-4.html#cb72-12"></a>         <span class="dt">gpos_label =</span> gpos <span class="op">+</span><span class="st"> </span>x_shift_label,</span>
<span id="cb72-13"><a href="figure-4.html#cb72-13"></a>         <span class="dt">gpos_label2 =</span> gpos_label <span class="op">-</span><span class="st"> </span><span class="kw">sign</span>(x_shift_label) <span class="op">*</span>.<span class="dv">5</span><span class="op">*</span><span class="dv">10</span><span class="op">^</span><span class="dv">7</span>,</span>
<span id="cb72-14"><a href="figure-4.html#cb72-14"></a>         <span class="dt">window =</span> <span class="kw">str_c</span>(<span class="st">&#39;bold(&#39;</span>,<span class="kw">project_case</span>(<span class="st">&#39;a&#39;</span>),<span class="st">&#39;):joint~italic(F[ST])&#39;</span>))</span>
<span id="cb72-15"><a href="figure-4.html#cb72-15"></a></span>
<span id="cb72-16"><a href="figure-4.html#cb72-16"></a><span class="co"># the y height of the outlier labels and the corresponding tags is set</span></span>
<span id="cb72-17"><a href="figure-4.html#cb72-17"></a>outlier_y &lt;-<span class="st"> </span><span class="fl">.45</span></span>
<span id="cb72-18"><a href="figure-4.html#cb72-18"></a>outlier_yend &lt;-<span class="st"> </span><span class="fl">.475</span></span></code></pre></div>
<p>We prepare a set of trait annotations for the genotype <span class="math inline">\(\times\)</span> phenotype association panels.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="figure-4.html#cb73-1"></a><span class="co"># the icons for the traits of the GxP are loaded</span></span>
<span id="cb73-2"><a href="figure-4.html#cb73-2"></a>trait_tibble &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">window =</span> <span class="kw">c</span>(<span class="st">&quot;bold(h):italic(p)[Bars]&quot;</span>,</span>
<span id="cb73-3"><a href="figure-4.html#cb73-3"></a>                                  <span class="st">&quot;bold(i):italic(p)[Peduncle]&quot;</span>,</span>
<span id="cb73-4"><a href="figure-4.html#cb73-4"></a>                                  <span class="st">&quot;bold(j):italic(p)[Snout]&quot;</span>),</span>
<span id="cb73-5"><a href="figure-4.html#cb73-5"></a>                       <span class="dt">grob =</span> hypo_trait_img<span class="op">$</span>grob_circle[hypo_trait_img<span class="op">$</span>trait <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;Bars&#39;</span>, <span class="st">&#39;Peduncle&#39;</span>, <span class="st">&#39;Snout&#39;</span>)])</span></code></pre></div>
</div>
<div id="plotting-1" class="section level3">
<h3><span class="header-section-number">16.2.3</span> Plotting</h3>
<p>Finally it is time to put the pieces together with one giant <code>ggplot():</code></p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="figure-4.html#cb74-1"></a><span class="co"># finally, the figure is being put together</span></span>
<span id="cb74-2"><a href="figure-4.html#cb74-2"></a>p_done &lt;-<span class="st"> </span><span class="kw">ggplot</span>()<span class="op">+</span></span>
<span id="cb74-3"><a href="figure-4.html#cb74-3"></a><span class="st">  </span><span class="co"># add gray/white LGs background</span></span>
<span id="cb74-4"><a href="figure-4.html#cb74-4"></a><span class="st">  </span><span class="kw">geom_hypo_LG</span>()<span class="op">+</span></span>
<span id="cb74-5"><a href="figure-4.html#cb74-5"></a><span class="st">  </span><span class="co"># the red highlights for the outlier regions are added</span></span>
<span id="cb74-6"><a href="figure-4.html#cb74-6"></a><span class="st">  </span><span class="kw">geom_vline</span>(<span class="dt">data =</span> outliers, <span class="kw">aes</span>(<span class="dt">xintercept =</span> gpos), <span class="dt">color =</span> outlr_clr)<span class="op">+</span></span>
<span id="cb74-7"><a href="figure-4.html#cb74-7"></a><span class="st">  </span><span class="co"># the tags of the outlier labels are added</span></span>
<span id="cb74-8"><a href="figure-4.html#cb74-8"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="dt">data =</span> outlier_label,</span>
<span id="cb74-9"><a href="figure-4.html#cb74-9"></a>               <span class="kw">aes</span>(<span class="dt">x =</span> gpos,</span>
<span id="cb74-10"><a href="figure-4.html#cb74-10"></a>                   <span class="dt">xend =</span> gpos_label2, <span class="dt">y =</span> outlier_y, <span class="dt">yend =</span> outlier_yend),</span>
<span id="cb74-11"><a href="figure-4.html#cb74-11"></a>               <span class="dt">color =</span> <span class="kw">alpha</span>(outlr_clr,<span class="dv">1</span>),<span class="dt">size =</span> <span class="fl">.2</span>)<span class="op">+</span></span>
<span id="cb74-12"><a href="figure-4.html#cb74-12"></a><span class="st">  </span><span class="co"># the outlier labels are added</span></span>
<span id="cb74-13"><a href="figure-4.html#cb74-13"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">data =</span> outlier_label, <span class="kw">aes</span>(<span class="dt">x =</span> gpos_label, <span class="dt">y =</span> outlier_yend, <span class="dt">label =</span> label),</span>
<span id="cb74-14"><a href="figure-4.html#cb74-14"></a>            <span class="dt">color =</span> <span class="kw">alpha</span>(outlr_clr,<span class="dv">1</span>), <span class="dt">fontface =</span> <span class="st">&#39;bold&#39;</span>)<span class="op">+</span></span>
<span id="cb74-15"><a href="figure-4.html#cb74-15"></a><span class="st">  </span><span class="co"># the fst, delta dxy and gxp data is plotted</span></span>
<span id="cb74-16"><a href="figure-4.html#cb74-16"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">data =</span> data, <span class="kw">aes</span>(<span class="dt">x =</span> GPOS, <span class="dt">y =</span> value),<span class="dt">size =</span> plot_size, <span class="dt">color =</span> plot_clr) <span class="op">+</span></span>
<span id="cb74-17"><a href="figure-4.html#cb74-17"></a><span class="st">  </span><span class="co"># the dxy data is plotted</span></span>
<span id="cb74-18"><a href="figure-4.html#cb74-18"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">data =</span> dxy_select,<span class="kw">aes</span>(<span class="dt">x =</span> GPOS, <span class="dt">y =</span> dxy),<span class="dt">size =</span> plot_size, <span class="dt">color =</span> plot_clr)<span class="op">+</span></span>
<span id="cb74-19"><a href="figure-4.html#cb74-19"></a><span class="st">  </span><span class="co"># the pi data is plotted</span></span>
<span id="cb74-20"><a href="figure-4.html#cb74-20"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">data =</span> pi_data_select, <span class="kw">aes</span>(<span class="dt">x =</span> GPOS, <span class="dt">y =</span> mean_pi),<span class="dt">size =</span> plot_size, <span class="dt">color =</span> plot_clr) <span class="op">+</span></span>
<span id="cb74-21"><a href="figure-4.html#cb74-21"></a><span class="st">  </span><span class="co"># the roh data is plotted</span></span>
<span id="cb74-22"><a href="figure-4.html#cb74-22"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">data =</span> recombination_data, <span class="kw">aes</span>(<span class="dt">x =</span> GPOS, <span class="dt">y =</span> RHO),<span class="dt">size =</span> plot_size, <span class="dt">color =</span> plot_clr) <span class="op">+</span></span>
<span id="cb74-23"><a href="figure-4.html#cb74-23"></a><span class="st">  </span><span class="co"># the smoothed rho is plotted</span></span>
<span id="cb74-24"><a href="figure-4.html#cb74-24"></a><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">data =</span> recombination_data, <span class="kw">aes</span>(<span class="dt">x =</span> GPOS, <span class="dt">y =</span> RHO, <span class="dt">group =</span> CHROM),</span>
<span id="cb74-25"><a href="figure-4.html#cb74-25"></a>                <span class="dt">color =</span> <span class="st">&#39;red&#39;</span>, <span class="dt">se =</span> <span class="ot">FALSE</span>, <span class="dt">size =</span> <span class="fl">.7</span>) <span class="op">+</span></span>
<span id="cb74-26"><a href="figure-4.html#cb74-26"></a><span class="st">  </span><span class="co"># the topology weighting data is plotted</span></span>
<span id="cb74-27"><a href="figure-4.html#cb74-27"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">data =</span> twisst_data, <span class="kw">aes</span>(<span class="dt">x =</span> GPOS, <span class="dt">y =</span> weight, <span class="dt">color =</span> topo_rel), <span class="dt">size =</span> <span class="fl">.4</span>) <span class="op">+</span></span>
<span id="cb74-28"><a href="figure-4.html#cb74-28"></a><span class="st">  </span><span class="co"># the null weighting is added</span></span>
<span id="cb74-29"><a href="figure-4.html#cb74-29"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">data =</span> twisst_null, <span class="kw">aes</span>(<span class="dt">yintercept =</span> weight), <span class="dt">color =</span> <span class="kw">rgb</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>), <span class="dt">size =</span> <span class="fl">.4</span>) <span class="op">+</span></span>
<span id="cb74-30"><a href="figure-4.html#cb74-30"></a><span class="st">  </span><span class="co"># the trait icons are added</span></span>
<span id="cb74-31"><a href="figure-4.html#cb74-31"></a><span class="st">  </span><span class="kw">geom_hypo_grob</span>(<span class="dt">data =</span> trait_tibble,</span>
<span id="cb74-32"><a href="figure-4.html#cb74-32"></a>                 <span class="kw">aes</span>(<span class="dt">grob =</span> grob, <span class="dt">angle =</span> <span class="dv">0</span>, <span class="dt">height =</span> <span class="fl">.65</span>),</span>
<span id="cb74-33"><a href="figure-4.html#cb74-33"></a>                 <span class="dt">inherit.aes =</span> <span class="ot">FALSE</span>, <span class="dt">x =</span> <span class="fl">.95</span>, <span class="dt">y =</span> <span class="fl">0.65</span>)<span class="op">+</span></span>
<span id="cb74-34"><a href="figure-4.html#cb74-34"></a><span class="st">  </span><span class="co"># setting the scales</span></span>
<span id="cb74-35"><a href="figure-4.html#cb74-35"></a><span class="st">  </span><span class="kw">scale_fill_hypo_LG_bg</span>() <span class="op">+</span></span>
<span id="cb74-36"><a href="figure-4.html#cb74-36"></a><span class="st">  </span><span class="kw">scale_x_hypo_LG</span>()<span class="op">+</span></span>
<span id="cb74-37"><a href="figure-4.html#cb74-37"></a><span class="st">  </span><span class="kw">scale_color_gradient</span>( <span class="dt">low =</span> <span class="st">&quot;#f0a830ff&quot;</span>, <span class="dt">high =</span> <span class="st">&quot;#084082ff&quot;</span>, <span class="dt">guide =</span> <span class="ot">FALSE</span>)<span class="op">+</span></span>
<span id="cb74-38"><a href="figure-4.html#cb74-38"></a><span class="st">  </span><span class="co"># organizing the plot across panels</span></span>
<span id="cb74-39"><a href="figure-4.html#cb74-39"></a><span class="st">  </span><span class="kw">facet_grid</span>(window<span class="op">~</span>.,<span class="dt">scales =</span> <span class="st">&#39;free&#39;</span>,<span class="dt">switch =</span> <span class="st">&#39;y&#39;</span>, <span class="dt">labeller =</span> label_parsed)<span class="op">+</span></span>
<span id="cb74-40"><a href="figure-4.html#cb74-40"></a><span class="st">  </span><span class="co"># tweak plot appreance</span></span>
<span id="cb74-41"><a href="figure-4.html#cb74-41"></a><span class="st">  </span><span class="kw">theme_hypo</span>()<span class="op">+</span></span>
<span id="cb74-42"><a href="figure-4.html#cb74-42"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&#39;bottom&#39;</span>,</span>
<span id="cb74-43"><a href="figure-4.html#cb74-43"></a>    <span class="dt">axis.title =</span> <span class="kw">element_blank</span>(),<span class="dt">strip.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>),</span>
<span id="cb74-44"><a href="figure-4.html#cb74-44"></a>    <span class="dt">strip.background =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb74-45"><a href="figure-4.html#cb74-45"></a>    <span class="dt">strip.placement =</span> <span class="st">&#39;outside&#39;</span>)</span></code></pre></div>
<center>
<img src="plot_F4_files/figure-html/unnamed-chunk-17-1.png" width="897.6" />
</center>
<p>The final figure is then exported using <code>hypo_save()</code>.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="figure-4.html#cb75-1"></a><span class="co"># export figure 4</span></span>
<span id="cb75-2"><a href="figure-4.html#cb75-2"></a>scl &lt;-<span class="st"> </span><span class="fl">.8</span></span>
<span id="cb75-3"><a href="figure-4.html#cb75-3"></a><span class="kw">hypo_save</span>(p_done, <span class="dt">filename =</span> <span class="st">&#39;figures/F4.png&#39;</span>,</span>
<span id="cb75-4"><a href="figure-4.html#cb75-4"></a>          <span class="dt">width =</span> <span class="dv">297</span><span class="op">*</span>scl, <span class="dt">height =</span> <span class="dv">275</span><span class="op">*</span>scl,</span>
<span id="cb75-5"><a href="figure-4.html#cb75-5"></a>          <span class="dt">units =</span> <span class="st">&#39;mm&#39;</span>,</span>
<span id="cb75-6"><a href="figure-4.html#cb75-6"></a>          <span class="dt">type =</span> <span class="st">&quot;cairo&quot;</span>,</span>
<span id="cb75-7"><a href="figure-4.html#cb75-7"></a>          <span class="dt">comment =</span> plot_comment)</span></code></pre></div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="figure-3.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
