<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>13 Figure 1 | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="13 Figure 1 | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="13 Figure 1 | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench" />


<meta name="date" content="2020-06-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="git-11-visualization.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a><ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a><ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a><ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a><ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a><ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a><ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i> &amp; pi)</a><ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a><ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (phylogeny &amp; topology weighting)</a><ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a><ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a><ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a><ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (poptrees)</a><ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#details-of-analysis_fst_poptree.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_fst_poptree.nf</code></a><ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#data-preparation-4"><i class="fa fa-check"></i><b>8.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a><ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a><ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-5"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a><ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a><ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-6"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a><ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a><ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-7"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-visualization.html"><a href="git-11-visualization.html"><i class="fa fa-check"></i><b>12</b> (git 11) Visualization</a></li>
<li class="chapter" data-level="13" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>13</b> Figure 1</a><ul>
<li class="chapter" data-level="13.1" data-path="figure-1.html"><a href="figure-1.html#summary-10"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>13.2</b> Details of <code>plot_F1.R</code></a><ul>
<li class="chapter" data-level="13.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>13.2.1</b> Config</a></li>
<li class="chapter" data-level="13.2.2" data-path="figure-1.html"><a href="figure-1.html#data-import"><i class="fa fa-check"></i><b>13.2.2</b> Data import</a></li>
<li class="chapter" data-level="13.2.3" data-path="figure-1.html"><a href="figure-1.html#plotting"><i class="fa fa-check"></i><b>13.2.3</b> Plotting</a></li>
<li class="chapter" data-level="13.2.4" data-path="figure-1.html"><a href="figure-1.html#table-export"><i class="fa fa-check"></i><b>13.2.4</b> Table export</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-1" class="section level1">
<h1><span class="header-section-number">13</span> Figure 1</h1>
<div id="summary-10" class="section level2">
<h2><span class="header-section-number">13.1</span> Summary</h2>
<p>This is the accessory documentation of Figure 1.
The Figure can be recreated by running the <strong>R</strong> script <code>plot_F1.R</code>:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb14-1"><a href="figure-1.html#cb14-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span></span>
<span id="cb14-2"><a href="figure-1.html#cb14-2"></a></span>
<span id="cb14-3"><a href="figure-1.html#cb14-3"></a><span class="ex">Rscript</span> --vanilla R/fig/plot_F1.R 2_analysis/dxy/50k/ 2_analysis/fst/50k/</span></code></pre></div>
</div>
<div id="details-of-plot_f1.r" class="section level2">
<h2><span class="header-section-number">13.2</span> Details of <code>plot_F1.R</code></h2>
<p>In the following, the individual steps of the R script are documented.
It is an executable R script that depends on the accessory R package <strong>GenomicOriginsScripts</strong>.</p>
<div id="config" class="section level3">
<h3><span class="header-section-number">13.2.1</span> Config</h3>
<p>The scripts start with a header that contains copy &amp; paste templates to execute or debug the script:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="figure-1.html#cb15-1"></a><span class="co">#!/usr/bin/env Rscript</span></span>
<span id="cb15-2"><a href="figure-1.html#cb15-2"></a><span class="co"># run from terminal:</span></span>
<span id="cb15-3"><a href="figure-1.html#cb15-3"></a><span class="co"># Rscript --vanilla R/fig/plot_F1.R \</span></span>
<span id="cb15-4"><a href="figure-1.html#cb15-4"></a><span class="co">#    2_analysis/dxy/50k/ 2_analysis/fst/50k/ 2_analysis/summaries/fst_globals.txt</span></span>
<span id="cb15-5"><a href="figure-1.html#cb15-5"></a><span class="co"># ===============================================================</span></span>
<span id="cb15-6"><a href="figure-1.html#cb15-6"></a><span class="co"># This script produces Figure 1 of the study &quot;Ancestral variation, hybridization and modularity</span></span>
<span id="cb15-7"><a href="figure-1.html#cb15-7"></a><span class="co"># fuel a marine radiation&quot; by Hench, McMillan and Puebla</span></span>
<span id="cb15-8"><a href="figure-1.html#cb15-8"></a><span class="co"># ---------------------------------------------------------------</span></span>
<span id="cb15-9"><a href="figure-1.html#cb15-9"></a><span class="co"># ===============================================================</span></span>
<span id="cb15-10"><a href="figure-1.html#cb15-10"></a><span class="co"># args &lt;- c(&#39;2_analysis/dxy/50k/&#39;, &#39;2_analysis/fst/50k/&#39;, &#39;2_analysis/summaries/fst_globals.txt&#39;)</span></span>
<span id="cb15-11"><a href="figure-1.html#cb15-11"></a><span class="co"># script_name &lt;- &quot;R/fig/plot_F1.R&quot;</span></span></code></pre></div>
<p>The next section processes the input from the command line.
It stores the arguments in the vector <code>args</code>.
The R package <strong>GenomicOriginsScripts</strong> is loaded and the script name and the current working directory are stored inside variables (<code>script_name</code>, <code>plot_comment</code>).
This information will later be written into the meta data of the figure to help us tracing back the scripts that created the figures in the future.</p>
<p>Then we drop all the imported information besides the arguments following the script name and print the information to the terminal.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="figure-1.html#cb16-1"></a>args &lt;-<span class="st"> </span><span class="kw">commandArgs</span>(<span class="dt">trailingOnly=</span><span class="ot">FALSE</span>)</span>
<span id="cb16-2"><a href="figure-1.html#cb16-2"></a><span class="co"># setup -----------------------</span></span>
<span id="cb16-3"><a href="figure-1.html#cb16-3"></a><span class="kw">library</span>(GenomicOriginsScripts)</span>
<span id="cb16-4"><a href="figure-1.html#cb16-4"></a><span class="kw">library</span>(hypoimg)</span>
<span id="cb16-5"><a href="figure-1.html#cb16-5"></a></span>
<span id="cb16-6"><a href="figure-1.html#cb16-6"></a><span class="kw">cat</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>)</span>
<span id="cb16-7"><a href="figure-1.html#cb16-7"></a>script_name &lt;-<span class="st"> </span>args[<span class="dv">5</span>] <span class="op">%&gt;%</span></span>
<span id="cb16-8"><a href="figure-1.html#cb16-8"></a><span class="st">  </span><span class="kw">str_remove</span>(., <span class="st">&#39;--file=&#39;</span>)</span>
<span id="cb16-9"><a href="figure-1.html#cb16-9"></a></span>
<span id="cb16-10"><a href="figure-1.html#cb16-10"></a>plot_comment &lt;-<span class="st"> </span>script_name <span class="op">%&gt;%</span></span>
<span id="cb16-11"><a href="figure-1.html#cb16-11"></a><span class="st">  </span><span class="kw">str_c</span>(<span class="st">&#39;mother-script = &#39;</span>, <span class="kw">getwd</span>(), <span class="st">&#39;/&#39;</span>, .)</span>
<span id="cb16-12"><a href="figure-1.html#cb16-12"></a></span>
<span id="cb16-13"><a href="figure-1.html#cb16-13"></a>args &lt;-<span class="st"> </span><span class="kw">process_input</span>(script_name, args)</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="figure-1.html#cb17-1"></a><span class="co">#&gt; ── Script: R/fig/plot_F1.R ──────────────────────────────────────────────</span></span>
<span id="cb17-2"><a href="figure-1.html#cb17-2"></a><span class="co">#&gt; Parameters read:</span></span>
<span id="cb17-3"><a href="figure-1.html#cb17-3"></a><span class="co">#&gt;  ★ 1: 2_analysis/dxy/50k/</span></span>
<span id="cb17-4"><a href="figure-1.html#cb17-4"></a><span class="co">#&gt;  ★ 2: 2_analysis/fst/50k/</span></span>
<span id="cb17-5"><a href="figure-1.html#cb17-5"></a><span class="co">#&gt;  ★ 3: 2_analysis/summaries/fst_globals.txt</span></span>
<span id="cb17-6"><a href="figure-1.html#cb17-6"></a><span class="co">#&gt; ─────────────────────────────────────────── /current/working/directory ──</span></span></code></pre></div>
<p>The directories for the different data types are received and stored in respective variables.
Also, we set a few parameters for the plot layout:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="figure-1.html#cb18-1"></a><span class="co"># config -----------------------</span></span>
<span id="cb18-2"><a href="figure-1.html#cb18-2"></a>dxy_dir &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">1</span>])</span>
<span id="cb18-3"><a href="figure-1.html#cb18-3"></a>fst_dir &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">2</span>])</span>
<span id="cb18-4"><a href="figure-1.html#cb18-4"></a>fst_globals &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">3</span>])</span>
<span id="cb18-5"><a href="figure-1.html#cb18-5"></a>wdh &lt;-<span class="st"> </span><span class="fl">.3</span>          <span class="co"># The width of the boxplots</span></span>
<span id="cb18-6"><a href="figure-1.html#cb18-6"></a>scaler &lt;-<span class="st"> </span><span class="dv">20</span>       <span class="co"># the ratio of the Fst and the dxy axis</span></span>
<span id="cb18-7"><a href="figure-1.html#cb18-7"></a>clr_sec &lt;-<span class="st"> &#39;gray&#39;</span>  <span class="co"># the color of the secondary axis (dxy)</span></span></code></pre></div>
</div>
<div id="data-import" class="section level3">
<h3><span class="header-section-number">13.2.2</span> Data import</h3>
<p>We begin with the data import by first collecting the paths to all files containing either <em>F<sub>ST</sub></em> or <em>d<sub>XY</sub></em> data (<code>dir()</code>), then iterating the import function over all files (<code>map(summarize_fst)</code>) and finally combining the outputs into a single tibble (<code>bind_rows()</code>).
This is done for both <em>F<sub>ST</sub></em> and <em>d<sub>XY</sub></em>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="figure-1.html#cb19-1"></a><span class="co"># start script -------------------</span></span>
<span id="cb19-2"><a href="figure-1.html#cb19-2"></a></span>
<span id="cb19-3"><a href="figure-1.html#cb19-3"></a><span class="co"># import Fst</span></span>
<span id="cb19-4"><a href="figure-1.html#cb19-4"></a>fst_files &lt;-<span class="st"> </span><span class="kw">dir</span>(fst_dir, <span class="dt">pattern =</span> <span class="st">&#39;.50k.windowed.weir.fst.gz&#39;</span>)</span>
<span id="cb19-5"><a href="figure-1.html#cb19-5"></a></span>
<span id="cb19-6"><a href="figure-1.html#cb19-6"></a>fst_data &lt;-<span class="st"> </span><span class="kw">str_c</span>(fst_dir,fst_files) <span class="op">%&gt;%</span></span>
<span id="cb19-7"><a href="figure-1.html#cb19-7"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">map</span>(summarize_fst) <span class="op">%&gt;%</span></span>
<span id="cb19-8"><a href="figure-1.html#cb19-8"></a><span class="st">  </span><span class="kw">bind_rows</span>()</span>
<span id="cb19-9"><a href="figure-1.html#cb19-9"></a></span>
<span id="cb19-10"><a href="figure-1.html#cb19-10"></a><span class="co"># lookup dxy files</span></span>
<span id="cb19-11"><a href="figure-1.html#cb19-11"></a>dxy_files &lt;-<span class="st"> </span><span class="kw">dir</span>(dxy_dir)</span>
<span id="cb19-12"><a href="figure-1.html#cb19-12"></a></span>
<span id="cb19-13"><a href="figure-1.html#cb19-13"></a><span class="co"># import dxy</span></span>
<span id="cb19-14"><a href="figure-1.html#cb19-14"></a>dxy_data &lt;-<span class="st">  </span><span class="kw">str_c</span>(dxy_dir,dxy_files) <span class="op">%&gt;%</span></span>
<span id="cb19-15"><a href="figure-1.html#cb19-15"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">map</span>(summarize_dxy) <span class="op">%&gt;%</span></span>
<span id="cb19-16"><a href="figure-1.html#cb19-16"></a><span class="st">  </span><span class="kw">bind_rows</span>()</span></code></pre></div>
<p>We use the genome wide average <em>F<sub>ST</sub></em> to rank the individual pair wise comparisons.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="figure-1.html#cb20-1"></a><span class="co"># determine fst ranking</span></span>
<span id="cb20-2"><a href="figure-1.html#cb20-2"></a>fst_order &lt;-<span class="st"> </span>fst_data <span class="op">%&gt;%</span></span>
<span id="cb20-3"><a href="figure-1.html#cb20-3"></a><span class="st">  </span><span class="kw">select</span>(run, <span class="st">`</span><span class="dt">mean_weighted-fst</span><span class="st">`</span>) <span class="op">%&gt;%</span></span>
<span id="cb20-4"><a href="figure-1.html#cb20-4"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">run =</span> <span class="kw">fct_reorder</span>(run, <span class="st">`</span><span class="dt">mean_weighted-fst</span><span class="st">`</span>))</span></code></pre></div>
<p>Then, we merge the <em>F<sub>ST</sub></em> and <em>d<sub>XY</sub></em> data sets and do quite a bit of data wrangling to create a rescaled <em>d<sub>XY</sub></em> value and to prepare the placement of the boxplots.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="figure-1.html#cb21-1"></a><span class="co"># merge fst and dxy cc_data</span></span>
<span id="cb21-2"><a href="figure-1.html#cb21-2"></a><span class="co"># (large parts of this code are now unnecessary after the separation of dxy and</span></span>
<span id="cb21-3"><a href="figure-1.html#cb21-3"></a><span class="co">#  fst plots into separate panels b &amp; c)</span></span>
<span id="cb21-4"><a href="figure-1.html#cb21-4"></a>data &lt;-<span class="st"> </span><span class="kw">left_join</span>(fst_data, dxy_data) <span class="op">%&gt;%</span></span>
<span id="cb21-5"><a href="figure-1.html#cb21-5"></a><span class="st">  </span><span class="kw">select</span>(<span class="kw">c</span>(<span class="dv">8</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">7</span>,<span class="dv">9</span><span class="op">:</span><span class="dv">15</span>)) <span class="op">%&gt;%</span></span>
<span id="cb21-6"><a href="figure-1.html#cb21-6"></a><span class="st">  </span><span class="co"># reformat table to enable parallel plotting (with secondary axis)</span></span>
<span id="cb21-7"><a href="figure-1.html#cb21-7"></a><span class="st">  </span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&#39;stat&#39;</span>, <span class="dt">value =</span> <span class="st">&#39;val&#39;</span>, <span class="dv">2</span><span class="op">:</span><span class="dv">15</span>) <span class="op">%&gt;%</span></span>
<span id="cb21-8"><a href="figure-1.html#cb21-8"></a><span class="st">  </span><span class="co"># sumstat contains the values needed to plot the boxplots (quartiles, etc)</span></span>
<span id="cb21-9"><a href="figure-1.html#cb21-9"></a><span class="st">  </span><span class="kw">separate</span>(stat, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;sumstat&#39;</span>, <span class="st">&#39;popstat&#39;</span>), <span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb21-10"><a href="figure-1.html#cb21-10"></a><span class="st">  </span><span class="co"># duplicate dxy values scaled to fst range</span></span>
<span id="cb21-11"><a href="figure-1.html#cb21-11"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">val_scaled =</span> <span class="kw">ifelse</span>(popstat <span class="op">==</span><span class="st"> &#39;dxy&#39;</span>, val <span class="op">*</span><span class="st"> </span>scaler , val)) <span class="op">%&gt;%</span></span>
<span id="cb21-12"><a href="figure-1.html#cb21-12"></a><span class="st">  </span><span class="kw">unite</span>(temp, val, val_scaled) <span class="op">%&gt;%</span></span>
<span id="cb21-13"><a href="figure-1.html#cb21-13"></a><span class="st">  </span><span class="co"># separate th eoriginal values from the scales ons (scaled = secondary axis)</span></span>
<span id="cb21-14"><a href="figure-1.html#cb21-14"></a><span class="st">  </span><span class="kw">spread</span>(.,<span class="dt">key =</span> <span class="st">&#39;sumstat&#39;</span>,<span class="dt">value =</span> <span class="st">&#39;temp&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb21-15"><a href="figure-1.html#cb21-15"></a><span class="st">  </span><span class="kw">separate</span>(mean, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;mean&#39;</span>,<span class="st">&#39;mean_scaled&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb21-16"><a href="figure-1.html#cb21-16"></a><span class="st">  </span><span class="kw">separate</span>(median, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;median&#39;</span>,<span class="st">&#39;median_scaled&#39;</span>), <span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb21-17"><a href="figure-1.html#cb21-17"></a><span class="st">  </span><span class="kw">separate</span>(sd, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;sd&#39;</span>,<span class="st">&#39;sd_scaled&#39;</span>),<span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb21-18"><a href="figure-1.html#cb21-18"></a><span class="st">  </span><span class="kw">separate</span>(lower, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;lower&#39;</span>,<span class="st">&#39;lower_scaled&#39;</span>), <span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb21-19"><a href="figure-1.html#cb21-19"></a><span class="st">  </span><span class="kw">separate</span>(upper, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;upper&#39;</span>,<span class="st">&#39;upper_scaled&#39;</span>), <span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb21-20"><a href="figure-1.html#cb21-20"></a><span class="st">  </span><span class="kw">separate</span>(lowpoint, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;lowpoint&#39;</span>,<span class="st">&#39;lowpoint_scaled&#39;</span>), <span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb21-21"><a href="figure-1.html#cb21-21"></a><span class="st">  </span><span class="kw">separate</span>(highpoint, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;highpoint&#39;</span>,<span class="st">&#39;highpoint_scaled&#39;</span>), <span class="dt">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="dt">convert =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb21-22"><a href="figure-1.html#cb21-22"></a><span class="st">  </span><span class="co"># include &quot;dodge&quot;-positions for side-by-side plotting (secondary axis)</span></span>
<span id="cb21-23"><a href="figure-1.html#cb21-23"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">loc =</span> <span class="kw">str_sub</span>(run,<span class="dv">4</span>,<span class="dv">6</span>),</span>
<span id="cb21-24"><a href="figure-1.html#cb21-24"></a>         <span class="dt">run =</span> <span class="kw">factor</span>(run, <span class="dt">levels =</span> <span class="kw">levels</span>(fst_order<span class="op">$</span>run)),</span>
<span id="cb21-25"><a href="figure-1.html#cb21-25"></a>         <span class="dt">x =</span> <span class="kw">as.numeric</span>(run) ,</span>
<span id="cb21-26"><a href="figure-1.html#cb21-26"></a>         <span class="dt">x_dodge =</span> <span class="kw">ifelse</span>(popstat <span class="op">==</span><span class="st"> &#39;dxy&#39;</span>, x <span class="op">+</span><span class="st"> </span><span class="fl">.25</span>, x <span class="op">-</span><span class="st"> </span><span class="fl">.25</span>),</span>
<span id="cb21-27"><a href="figure-1.html#cb21-27"></a>         <span class="dt">x_start_dodge =</span> x_dodge <span class="op">-</span><span class="st"> </span>wdh<span class="op">/</span><span class="dv">2</span>,</span>
<span id="cb21-28"><a href="figure-1.html#cb21-28"></a>         <span class="dt">x_end_dodge =</span> x_dodge <span class="op">+</span><span class="st"> </span>wdh<span class="op">/</span><span class="dv">2</span>,</span>
<span id="cb21-29"><a href="figure-1.html#cb21-29"></a>         <span class="dt">popstat_loc =</span> <span class="kw">str_c</span>(popstat,<span class="st">&#39;[&#39;</span>,loc,<span class="st">&#39;]&#39;</span>))</span></code></pre></div>
<p>At this point, the data is ready for the boxplots.
But first we are going to prepare the networks of pairwise comparisons.</p>
<p>For this we create a tibble of the runs with their respective rank.
Then, we prepare a config table with one row per location, storing the parameters needed for the layout function for the networks.
We need to define the location, the number of species at the location, the short three letter ID of those species and a weight parameter that is shifting the comparison label on the link within the networks.</p>
<p>Finally, we create one network plot per location.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="figure-1.html#cb22-1"></a><span class="co"># sort run by average genome wide Fst</span></span>
<span id="cb22-2"><a href="figure-1.html#cb22-2"></a>run_ord &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">run =</span> <span class="kw">levels</span>(data<span class="op">$</span>run),</span>
<span id="cb22-3"><a href="figure-1.html#cb22-3"></a>                  <span class="dt">run_ord =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">levels</span>(data<span class="op">$</span>run)))</span>
<span id="cb22-4"><a href="figure-1.html#cb22-4"></a></span>
<span id="cb22-5"><a href="figure-1.html#cb22-5"></a><span class="co"># onderlying structure for the network plots</span></span>
<span id="cb22-6"><a href="figure-1.html#cb22-6"></a>networx &lt;-<span class="st"> </span><span class="kw">tibble</span>( <span class="dt">loc =</span> <span class="kw">c</span>(<span class="st">&#39;bel&#39;</span>,<span class="st">&#39;hon&#39;</span>, <span class="st">&#39;pan&#39;</span>),</span>
<span id="cb22-7"><a href="figure-1.html#cb22-7"></a>                   <span class="dt">n =</span> <span class="kw">c</span>(<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">3</span>),</span>
<span id="cb22-8"><a href="figure-1.html#cb22-8"></a>                   <span class="dt">label =</span> <span class="kw">list</span>(<span class="kw">str_c</span>(<span class="kw">c</span>(<span class="st">&#39;ind&#39;</span>,<span class="st">&#39;may&#39;</span>,<span class="st">&#39;nig&#39;</span>,<span class="st">&#39;pue&#39;</span>,<span class="st">&#39;uni&#39;</span>),<span class="st">&#39;bel&#39;</span>),</span>
<span id="cb22-9"><a href="figure-1.html#cb22-9"></a>                                <span class="kw">str_c</span>(<span class="kw">c</span>(<span class="st">&#39;abe&#39;</span>,<span class="st">&#39;gum&#39;</span>,<span class="st">&#39;nig&#39;</span>,<span class="st">&#39;pue&#39;</span>,<span class="st">&#39;ran&#39;</span>,<span class="st">&#39;uni&#39;</span>),<span class="st">&#39;hon&#39;</span>),</span>
<span id="cb22-10"><a href="figure-1.html#cb22-10"></a>                                <span class="kw">str_c</span>(<span class="kw">c</span>(<span class="st">&#39;nig&#39;</span>,<span class="st">&#39;pue&#39;</span>,<span class="st">&#39;uni&#39;</span>),<span class="st">&#39;pan&#39;</span>)),</span>
<span id="cb22-11"><a href="figure-1.html#cb22-11"></a>                   <span class="dt">weight =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="fl">1.45</span>,<span class="dv">1</span>)) <span class="op">%&gt;%</span></span>
<span id="cb22-12"><a href="figure-1.html#cb22-12"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">pmap</span>(network_layout) <span class="op">%&gt;%</span></span>
<span id="cb22-13"><a href="figure-1.html#cb22-13"></a><span class="st">  </span><span class="kw">bind_rows</span>()</span>
<span id="cb22-14"><a href="figure-1.html#cb22-14"></a></span>
<span id="cb22-15"><a href="figure-1.html#cb22-15"></a><span class="co"># plot the individual networks by location</span></span>
<span id="cb22-16"><a href="figure-1.html#cb22-16"></a>plot_list &lt;-<span class="st"> </span>networx <span class="op">%&gt;%</span></span>
<span id="cb22-17"><a href="figure-1.html#cb22-17"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">pmap</span>(plot_network, <span class="dt">node_lab_shift =</span> <span class="fl">.2</span>)</span></code></pre></div>
</div>
<div id="plotting" class="section level3">
<h3><span class="header-section-number">13.2.3</span> Plotting</h3>
<p>To create the first panel of Figure 1, we combine the three networks and label the locations.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="figure-1.html#cb23-1"></a><span class="co"># assemble panel a</span></span>
<span id="cb23-2"><a href="figure-1.html#cb23-2"></a>p1 &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(</span>
<span id="cb23-3"><a href="figure-1.html#cb23-3"></a>  grid<span class="op">::</span><span class="kw">textGrob</span>(<span class="st">&#39;Belize&#39;</span>),</span>
<span id="cb23-4"><a href="figure-1.html#cb23-4"></a>  grid<span class="op">::</span><span class="kw">textGrob</span>(<span class="st">&#39;Honduras&#39;</span>),</span>
<span id="cb23-5"><a href="figure-1.html#cb23-5"></a>  grid<span class="op">::</span><span class="kw">textGrob</span>(<span class="st">&#39;Panama&#39;</span>),</span>
<span id="cb23-6"><a href="figure-1.html#cb23-6"></a>  plot_list[[<span class="dv">1</span>]], plot_list[[<span class="dv">2</span>]], plot_list[[<span class="dv">3</span>]],</span>
<span id="cb23-7"><a href="figure-1.html#cb23-7"></a>  <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(.<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<center>
<img src="plot_F1_files/figure-html/unnamed-chunk-9-1.png" width="864" />
</center>
<p>Now, we can create the second panel of Figure 1, by plotting our prepared data tibble.
We are going to plot each boxplot element as a single layer.</p>
<blockquote>
<p>(This, might seem a little cumbersome given <code>geom_boxplot()</code>, but this approach was chosen for specific fine tuning of the positioning, dropping of outliers and reducing runtime during the plotting phase - otherwise the entire genome wide data set would have been carried though whole script. By now, this indirect approach is also somewhat obsolete since the <i>F<sub>ST</sub></i> and <i>d<sub>XY</sub></i> boxplots are now separated into several panels)</p>
</blockquote>
<p>The <i>F<sub>ST</sub></i> boxplots are created for panel <strong>b</strong>…</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="figure-1.html#cb24-1"></a><span class="co"># assemble panel b</span></span>
<span id="cb24-2"><a href="figure-1.html#cb24-2"></a>p2 &lt;-<span class="st"> </span>data <span class="op">%&gt;%</span></span>
<span id="cb24-3"><a href="figure-1.html#cb24-3"></a><span class="st">  </span><span class="kw">filter</span>(popstat <span class="op">==</span><span class="st"> &quot;weighted-fst&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb24-4"><a href="figure-1.html#cb24-4"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">color =</span> loc)) <span class="op">+</span></span>
<span id="cb24-5"><a href="figure-1.html#cb24-5"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">xend =</span> x,</span>
<span id="cb24-6"><a href="figure-1.html#cb24-6"></a>                   <span class="dt">y =</span> lowpoint, <span class="dt">yend =</span> highpoint))<span class="op">+</span></span>
<span id="cb24-7"><a href="figure-1.html#cb24-7"></a><span class="st">  </span><span class="kw">geom_rect</span>(<span class="kw">aes</span>(<span class="dt">xmin =</span> x <span class="op">-</span><span class="st"> </span>wdh, <span class="dt">xmax =</span> x <span class="op">+</span><span class="st"> </span>wdh,</span>
<span id="cb24-8"><a href="figure-1.html#cb24-8"></a>                <span class="dt">ymin =</span> lower, <span class="dt">ymax =</span> upper),</span>
<span id="cb24-9"><a href="figure-1.html#cb24-9"></a>             <span class="dt">fill =</span> <span class="st">&#39;white&#39;</span>)<span class="op">+</span></span>
<span id="cb24-10"><a href="figure-1.html#cb24-10"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x <span class="op">-</span><span class="st"> </span>wdh,</span>
<span id="cb24-11"><a href="figure-1.html#cb24-11"></a>                   <span class="dt">xend =</span> x <span class="op">+</span><span class="st"> </span>wdh,</span>
<span id="cb24-12"><a href="figure-1.html#cb24-12"></a>                   <span class="dt">y =</span> median,</span>
<span id="cb24-13"><a href="figure-1.html#cb24-13"></a>                   <span class="dt">yend =</span> median), <span class="dt">lwd =</span> <span class="fl">.9</span>)<span class="op">+</span></span>
<span id="cb24-14"><a href="figure-1.html#cb24-14"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> mean),</span>
<span id="cb24-15"><a href="figure-1.html#cb24-15"></a>             <span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="fl">.7</span>, <span class="dt">fill =</span> <span class="st">&#39;white&#39;</span>)<span class="op">+</span></span>
<span id="cb24-16"><a href="figure-1.html#cb24-16"></a><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">28</span>) <span class="op">+</span></span>
<span id="cb24-17"><a href="figure-1.html#cb24-17"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="co">#breaks = c(0,.05,.1,.15),</span></span>
<span id="cb24-18"><a href="figure-1.html#cb24-18"></a>                     <span class="dt">name =</span> <span class="kw">expression</span>(<span class="kw">italic</span>(F[ST])))<span class="op">+</span></span>
<span id="cb24-19"><a href="figure-1.html#cb24-19"></a><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="kw">make_faint_clr</span>(<span class="st">&#39;bel&#39;</span>),</span>
<span id="cb24-20"><a href="figure-1.html#cb24-20"></a>                                <span class="kw">make_faint_clr</span>(<span class="st">&#39;hon&#39;</span>),</span>
<span id="cb24-21"><a href="figure-1.html#cb24-21"></a>                                <span class="kw">make_faint_clr</span>(<span class="st">&#39;pan&#39;</span>))[<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">6</span>)])<span class="op">+</span></span>
<span id="cb24-22"><a href="figure-1.html#cb24-22"></a><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">29</span>),</span>
<span id="cb24-23"><a href="figure-1.html#cb24-23"></a>                  <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="op">+</span></span>
<span id="cb24-24"><a href="figure-1.html#cb24-24"></a><span class="st">  </span><span class="kw">theme_minimal</span>()<span class="op">+</span></span>
<span id="cb24-25"><a href="figure-1.html#cb24-25"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb24-26"><a href="figure-1.html#cb24-26"></a>        <span class="dt">legend.position =</span> <span class="st">&#39;none&#39;</span>,</span>
<span id="cb24-27"><a href="figure-1.html#cb24-27"></a>        <span class="dt">strip.placement =</span> <span class="st">&#39;outside&#39;</span>,</span>
<span id="cb24-28"><a href="figure-1.html#cb24-28"></a>        <span class="dt">strip.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>),</span>
<span id="cb24-29"><a href="figure-1.html#cb24-29"></a>        <span class="dt">panel.grid.major.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb24-30"><a href="figure-1.html#cb24-30"></a>        <span class="dt">panel.grid.minor.y =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb24-31"><a href="figure-1.html#cb24-31"></a>        <span class="dt">axis.text.y.right =</span> <span class="kw">element_text</span>(<span class="dt">color =</span> clr_sec),</span>
<span id="cb24-32"><a href="figure-1.html#cb24-32"></a>        <span class="dt">axis.title.y.right =</span> <span class="kw">element_text</span>(<span class="dt">color =</span> clr_sec))</span></code></pre></div>
<center>
<img src="plot_F1_files/figure-html/unnamed-chunk-11-1.png" width="432" />
</center>
<p>… and the <i>d<sub>XY</sub></i> boxplotsfor panel <strong>c</strong>:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="figure-1.html#cb25-1"></a><span class="co"># assemble panel c</span></span>
<span id="cb25-2"><a href="figure-1.html#cb25-2"></a>p3 &lt;-<span class="st"> </span>data <span class="op">%&gt;%</span></span>
<span id="cb25-3"><a href="figure-1.html#cb25-3"></a><span class="st">  </span><span class="kw">filter</span>(popstat <span class="op">==</span><span class="st"> &quot;dxy&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb25-4"><a href="figure-1.html#cb25-4"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">color =</span> loc)) <span class="op">+</span></span>
<span id="cb25-5"><a href="figure-1.html#cb25-5"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">xend =</span> x,</span>
<span id="cb25-6"><a href="figure-1.html#cb25-6"></a>                   <span class="dt">y =</span> lowpoint, <span class="dt">yend =</span> highpoint))<span class="op">+</span></span>
<span id="cb25-7"><a href="figure-1.html#cb25-7"></a><span class="st">  </span><span class="kw">geom_rect</span>(<span class="kw">aes</span>(<span class="dt">xmin =</span> x <span class="op">-</span><span class="st"> </span>wdh, <span class="dt">xmax =</span> x <span class="op">+</span><span class="st"> </span>wdh,</span>
<span id="cb25-8"><a href="figure-1.html#cb25-8"></a>                <span class="dt">ymin =</span> lower, <span class="dt">ymax =</span> upper),</span>
<span id="cb25-9"><a href="figure-1.html#cb25-9"></a>            <span class="dt">fill =</span> <span class="st">&#39;white&#39;</span>)<span class="op">+</span></span>
<span id="cb25-10"><a href="figure-1.html#cb25-10"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x <span class="op">-</span><span class="st"> </span>wdh,</span>
<span id="cb25-11"><a href="figure-1.html#cb25-11"></a>                   <span class="dt">xend =</span> x <span class="op">+</span><span class="st"> </span>wdh,</span>
<span id="cb25-12"><a href="figure-1.html#cb25-12"></a>                   <span class="dt">y =</span> median,</span>
<span id="cb25-13"><a href="figure-1.html#cb25-13"></a>                   <span class="dt">yend =</span> median),<span class="dt">lwd =</span> <span class="fl">.9</span>)<span class="op">+</span></span>
<span id="cb25-14"><a href="figure-1.html#cb25-14"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> mean),</span>
<span id="cb25-15"><a href="figure-1.html#cb25-15"></a>             <span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="fl">.7</span>, <span class="dt">fill =</span> <span class="st">&#39;white&#39;</span>)<span class="op">+</span></span>
<span id="cb25-16"><a href="figure-1.html#cb25-16"></a><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">28</span>) <span class="op">+</span></span>
<span id="cb25-17"><a href="figure-1.html#cb25-17"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>( <span class="kw">expression</span>(<span class="kw">italic</span>(d[XY])),</span>
<span id="cb25-18"><a href="figure-1.html#cb25-18"></a>                      <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">0025</span>,.<span class="dv">005</span>,.<span class="dv">0075</span>,.<span class="dv">01</span>),</span>
<span id="cb25-19"><a href="figure-1.html#cb25-19"></a>                      <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">01</span>))<span class="op">+</span></span>
<span id="cb25-20"><a href="figure-1.html#cb25-20"></a><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="kw">make_faint_clr</span>(<span class="st">&#39;bel&#39;</span>),</span>
<span id="cb25-21"><a href="figure-1.html#cb25-21"></a>                                <span class="kw">make_faint_clr</span>(<span class="st">&#39;hon&#39;</span>),</span>
<span id="cb25-22"><a href="figure-1.html#cb25-22"></a>                                <span class="kw">make_faint_clr</span>(<span class="st">&#39;pan&#39;</span>))[<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">6</span>)])<span class="op">+</span></span>
<span id="cb25-23"><a href="figure-1.html#cb25-23"></a><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">29</span>),</span>
<span id="cb25-24"><a href="figure-1.html#cb25-24"></a>                  <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="op">+</span></span>
<span id="cb25-25"><a href="figure-1.html#cb25-25"></a><span class="st">  </span><span class="kw">theme_minimal</span>()<span class="op">+</span></span>
<span id="cb25-26"><a href="figure-1.html#cb25-26"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb25-27"><a href="figure-1.html#cb25-27"></a>        <span class="dt">legend.position =</span> <span class="st">&#39;none&#39;</span>,</span>
<span id="cb25-28"><a href="figure-1.html#cb25-28"></a>        <span class="dt">strip.placement =</span> <span class="st">&#39;outside&#39;</span>,</span>
<span id="cb25-29"><a href="figure-1.html#cb25-29"></a>        <span class="dt">strip.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>),</span>
<span id="cb25-30"><a href="figure-1.html#cb25-30"></a>        <span class="dt">panel.grid.major.x =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb25-31"><a href="figure-1.html#cb25-31"></a>        <span class="dt">panel.grid.minor.y =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb25-32"><a href="figure-1.html#cb25-32"></a>        <span class="dt">axis.text.y.right =</span> <span class="kw">element_text</span>(<span class="dt">color =</span> clr_sec),</span>
<span id="cb25-33"><a href="figure-1.html#cb25-33"></a>        <span class="dt">axis.title.y.right =</span> <span class="kw">element_text</span>(<span class="dt">color =</span> clr_sec))</span></code></pre></div>
<center>
<img src="plot_F1_files/figure-html/unnamed-chunk-13-1.png" width="432" />
</center>
<p>The panels <strong>b</strong> and <strong>c</strong> are merged to crate the lower half of the figure and then the entire figure is being brought together.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="figure-1.html#cb26-1"></a><span class="co"># merge panel b &amp; c</span></span>
<span id="cb26-2"><a href="figure-1.html#cb26-2"></a>p23 &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p2,p3,</span>
<span id="cb26-3"><a href="figure-1.html#cb26-3"></a>                          <span class="dt">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb26-4"><a href="figure-1.html#cb26-4"></a>                          <span class="dt">labels =</span> letters[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>] <span class="op">%&gt;%</span></span>
<span id="cb26-5"><a href="figure-1.html#cb26-5"></a><span class="st">                            </span><span class="kw">project_case</span>())</span>
<span id="cb26-6"><a href="figure-1.html#cb26-6"></a><span class="co"># merge all panels</span></span>
<span id="cb26-7"><a href="figure-1.html#cb26-7"></a>p_done &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p23,</span>
<span id="cb26-8"><a href="figure-1.html#cb26-8"></a>                             <span class="dt">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb26-9"><a href="figure-1.html#cb26-9"></a>                             <span class="dt">rel_heights =</span> <span class="kw">c</span>(.<span class="dv">9</span>,<span class="dv">1</span>),</span>
<span id="cb26-10"><a href="figure-1.html#cb26-10"></a>                             <span class="dt">labels =</span> <span class="kw">c</span>(letters[<span class="dv">1</span>], <span class="ot">NULL</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">project_case</span>())</span></code></pre></div>
<center>
<img src="plot_F1_files/figure-html/unnamed-chunk-15-1.png" width="1056" />
</center>
<p>Finally, we can export Figure 1.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="figure-1.html#cb27-1"></a><span class="kw">hypo_save</span>(p_done, <span class="dt">filename =</span> <span class="st">&#39;figures/F1.pdf&#39;</span>,</span>
<span id="cb27-2"><a href="figure-1.html#cb27-2"></a>          <span class="dt">width =</span> <span class="dv">11</span>, <span class="dt">height =</span> <span class="fl">6.5</span>,</span>
<span id="cb27-3"><a href="figure-1.html#cb27-3"></a>          <span class="dt">comment =</span> plot_comment)</span></code></pre></div>
<blockquote>
<p>The function <code>hypo_save()</code> is simply a wrapper around <code>ggsave()</code>, that will write the name of the currently running script into the meta data of the plot (after the plot has been exported).
The benefit of this is that you can read this information later to remember how a specific plot was created using <code>hypo_show_metadata()</code>.
This is done using <a href="https://www.sno.phy.queensu.ca/~phil/exiftool/">exiftool</a> and has currently only been tested on my linux system.
If this does not work for you, simple replace <code>hypo_save()</code> with <code>ggsave()</code> and drop the <code>comment</code> parameter.</p>
</blockquote>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="figure-1.html#cb28-1"></a><span class="kw">hypo_show_metadata</span>(<span class="st">&#39;figures/F1.pdf&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="figure-1.html#cb29-1"></a><span class="co">#&gt; [1] &quot;mother-script = /current/working/directory/R/fig/plot_F1.R&quot;</span></span></code></pre></div>
</div>
<div id="table-export" class="section level3">
<h3><span class="header-section-number">13.2.4</span> Table export</h3>
<p>This script is also used to compile the sub-tables Suppl. Table 3 <strong>a</strong> - <strong>c</strong>.
The table is compiled from the weighted average <i>F<sub>ST</sub></i> values which are imported from the <code>nextflow</code> results and the average <i>d<sub>XY</sub></i> values which are computed within this <code>R</code> script.</p>
<p>First, a proto-version of Suppl. Table 3 is created.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="figure-1.html#cb30-1"></a><span class="co"># compile fst and dxy table (table 1) for the manuscript</span></span>
<span id="cb30-2"><a href="figure-1.html#cb30-2"></a>table_all &lt;-<span class="st"> </span>dxy_data <span class="op">%&gt;%</span></span>
<span id="cb30-3"><a href="figure-1.html#cb30-3"></a><span class="st">  </span><span class="kw">select</span>(run, mean_dxy) <span class="op">%&gt;%</span></span>
<span id="cb30-4"><a href="figure-1.html#cb30-4"></a><span class="st">  </span><span class="kw">left_join</span>( vroom<span class="op">::</span><span class="kw">vroom</span>(fst_globals, <span class="dt">delim =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>,</span>
<span id="cb30-5"><a href="figure-1.html#cb30-5"></a>                          <span class="dt">col_names =</span> <span class="kw">c</span>(<span class="st">&#39;loc&#39;</span>,<span class="st">&#39;run&#39;</span>,<span class="st">&#39;mean&#39;</span>,<span class="st">&#39;weighted_fst&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb30-6"><a href="figure-1.html#cb30-6"></a><span class="st">               </span><span class="kw">mutate</span>(<span class="dt">run =</span> <span class="kw">str_c</span>(loc,<span class="st">&#39;-&#39;</span>,run) <span class="op">%&gt;%</span></span>
<span id="cb30-7"><a href="figure-1.html#cb30-7"></a><span class="st">                        </span><span class="kw">reformat_run_name</span>())  <span class="op">%&gt;%</span></span>
<span id="cb30-8"><a href="figure-1.html#cb30-8"></a><span class="st">               </span><span class="kw">select</span>(run, weighted_fst)) <span class="op">%&gt;%</span></span>
<span id="cb30-9"><a href="figure-1.html#cb30-9"></a><span class="st">  </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">&#39;stat&#39;</span>,<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>) <span class="op">%&gt;%</span></span>
<span id="cb30-10"><a href="figure-1.html#cb30-10"></a><span class="st">  </span><span class="kw">separate</span>(run, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&#39;pop1&#39;</span>, <span class="st">&#39;pop2&#39;</span>), <span class="dt">sep =</span> <span class="st">&#39;-&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb30-11"><a href="figure-1.html#cb30-11"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">prep1 =</span> <span class="kw">ifelse</span>(stat <span class="op">==</span><span class="st"> &quot;weighted_fst&quot;</span>, pop2,pop1),</span>
<span id="cb30-12"><a href="figure-1.html#cb30-12"></a>         <span class="dt">prep2 =</span> <span class="kw">ifelse</span>(stat <span class="op">==</span><span class="st"> &quot;weighted_fst&quot;</span>, pop1,pop2),</span>
<span id="cb30-13"><a href="figure-1.html#cb30-13"></a>         <span class="dt">pop1 =</span> <span class="kw">factor</span>(prep1, <span class="dt">levels =</span> pop_levels),</span>
<span id="cb30-14"><a href="figure-1.html#cb30-14"></a>         <span class="dt">pop2 =</span> <span class="kw">factor</span>(prep2, <span class="dt">levels =</span> pop_levels),</span>
<span id="cb30-15"><a href="figure-1.html#cb30-15"></a>         <span class="dt">value =</span> <span class="kw">sprintf</span>(<span class="st">&#39;%7.5f&#39;</span>, value) ) <span class="op">%&gt;%</span></span>
<span id="cb30-16"><a href="figure-1.html#cb30-16"></a><span class="st">  </span><span class="kw">select</span>(pop1,pop2,value) <span class="op">%&gt;%</span></span>
<span id="cb30-17"><a href="figure-1.html#cb30-17"></a><span class="st">  </span><span class="kw">arrange</span>(pop2,pop1) <span class="op">%&gt;%</span></span>
<span id="cb30-18"><a href="figure-1.html#cb30-18"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">pop2 =</span> <span class="kw">as.character</span>(pop2) <span class="op">%&gt;%</span></span>
<span id="cb30-19"><a href="figure-1.html#cb30-19"></a><span class="st">           </span><span class="kw">str_replace</span>(<span class="dt">pattern =</span> <span class="st">&#39;([a-z]{3})([a-z]{3})&#39;</span>,</span>
<span id="cb30-20"><a href="figure-1.html#cb30-20"></a>                       <span class="dt">replacement =</span> <span class="st">&#39;</span><span class="ch">\\</span><span class="st">1|</span><span class="ch">\\</span><span class="st">2&#39;</span>),</span>
<span id="cb30-21"><a href="figure-1.html#cb30-21"></a>         <span class="dt">pop1 =</span> <span class="kw">as.character</span>(pop1) <span class="op">%&gt;%</span></span>
<span id="cb30-22"><a href="figure-1.html#cb30-22"></a><span class="st">           </span><span class="kw">str_replace</span>(<span class="dt">pattern =</span> <span class="st">&#39;([a-z]{3})([a-z]{3})&#39;</span>,</span>
<span id="cb30-23"><a href="figure-1.html#cb30-23"></a>                       <span class="dt">replacement =</span> <span class="st">&#39;</span><span class="ch">\\</span><span class="st">1|</span><span class="ch">\\</span><span class="st">2&#39;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb30-24"><a href="figure-1.html#cb30-24"></a><span class="st">  </span><span class="kw">pivot_wider</span>(<span class="dt">values_from =</span> value,</span>
<span id="cb30-25"><a href="figure-1.html#cb30-25"></a>              <span class="dt">names_from =</span> pop2) <span class="op">%&gt;%</span></span>
<span id="cb30-26"><a href="figure-1.html#cb30-26"></a><span class="st">  </span><span class="kw">rename</span>( <span class="dt">Population =</span> <span class="st">&#39;pop1&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb30-27"><a href="figure-1.html#cb30-27"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">srt1 =</span> <span class="kw">str_sub</span>(Population,<span class="op">-</span><span class="dv">3</span>, <span class="dv">-1</span>),</span>
<span id="cb30-28"><a href="figure-1.html#cb30-28"></a>         <span class="dt">srt2 =</span> <span class="kw">str_sub</span>(Population,<span class="dv">1</span>, <span class="dv">3</span>))  <span class="op">%&gt;%</span></span>
<span id="cb30-29"><a href="figure-1.html#cb30-29"></a><span class="st">  </span><span class="kw">arrange</span>(srt1,srt2) <span class="op">%&gt;%</span></span>
<span id="cb30-30"><a href="figure-1.html#cb30-30"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>srt1,<span class="op">-</span>srt2)</span></code></pre></div>
<p>Then, missing data is formated to be displayed as dashes.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="figure-1.html#cb31-1"></a><span class="co"># replace &quot;NA&quot; by dashes &quot;-&quot;</span></span>
<span id="cb31-2"><a href="figure-1.html#cb31-2"></a>table_all[<span class="kw">is.na</span>(table_all)] &lt;-<span class="st"> &#39;-&#39;</span></span></code></pre></div>
<p>Finally, the table is subset for each location and exported as a <code>.tex</code> file to be easily integrated within the <code>latex</code> document of the manuscript.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="figure-1.html#cb32-1"></a><span class="co"># export sub-tables 1 a - c</span></span>
<span id="cb32-2"><a href="figure-1.html#cb32-2"></a>table_all[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span><span class="op">:</span><span class="dv">6</span>)] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">export_2_latex</span>(<span class="dt">name =</span> <span class="st">&#39;tables/suppl_tab3a.tex&#39;</span>)</span>
<span id="cb32-3"><a href="figure-1.html#cb32-3"></a>table_all[<span class="dv">6</span><span class="op">:</span><span class="dv">11</span>,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">7</span><span class="op">:</span><span class="dv">12</span>)] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">export_2_latex</span>(<span class="dt">name =</span> <span class="st">&#39;tables/suppl_tab3b.tex&#39;</span>)</span>
<span id="cb32-4"><a href="figure-1.html#cb32-4"></a>table_all[<span class="dv">12</span><span class="op">:</span><span class="dv">14</span>, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">13</span><span class="op">:</span><span class="dv">15</span>)] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">export_2_latex</span>(<span class="dt">name =</span> <span class="st">&#39;tables/suppl_tab3c.tex&#39;</span>)</span></code></pre></div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="git-11-visualization.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
