<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10 (git 9) Analysis VII (hybridization) | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="10 (git 9) Analysis VII (hybridization) | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10 (git 9) Analysis VII (hybridization) | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench" />


<meta name="date" content="2020-06-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="git-8-analysis-vi-demographic-history.html"/>
<link rel="next" href="git-10-analysis-viii-admixture.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../">Script repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a><ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a><ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a><ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a><ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a><ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a><ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i> &amp; pi)</a><ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a><ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (phylogeny &amp; topology weighting)</a><ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a><ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a><ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a><ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (poptrees)</a><ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#details-of-analysis_xx.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_xx.nf</code></a><ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#data-preparation-4"><i class="fa fa-check"></i><b>8.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a><ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a><ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-5"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a><ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_xx.nf-1"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_xx.nf</code></a><ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-6"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a><ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_xx.nf-2"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_xx.nf</code></a><ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-7"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-visualization.html"><a href="git-11-visualization.html"><i class="fa fa-check"></i><b>12</b> (git 11) Visualization</a><ul>
<li class="chapter" data-level="12.1" data-path="git-11-visualization.html"><a href="git-11-visualization.html#all-at-once"><i class="fa fa-check"></i><b>12.1</b> All at once</a></li>
<li class="chapter" data-level="12.2" data-path="git-11-visualization.html"><a href="git-11-visualization.html#one-by-one"><i class="fa fa-check"></i><b>12.2</b> One by one</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="git-9-analysis-vii-hybridization" class="section level1">
<h1><span class="header-section-number">10</span> (git 9) Analysis VII (hybridization)</h1>
<div id="summary-8" class="section level2">
<h2><span class="header-section-number">10.1</span> Summary</h2>
<p>The population recombination rate is estimated within the <a href="https://www.nextflow.io/"><strong>nextflow</strong></a> script <code>analysis_XX.nf</code> (located under <code>$BASE_DIR/nf/09_analysis_hybridization/</code>), which runs on the XX data set.
Below is an overview of the steps involved in the analysis.
(The <span style="color:#4DAF4A">green dot</span> indicates the genotype input, <span style="color:#E41A1C">red arrows</span> depict output that is exported for further use.)</p>
<div style="max-width:500px; margin:auto;">

</div>
</div>
<div id="details-of-analysis_xx.nf-1" class="section level2">
<h2><span class="header-section-number">10.2</span> Details of <code>analysis_xx.nf</code></h2>
<div id="data-preparation-6" class="section level3">
<h3><span class="header-section-number">10.2.1</span> Data preparation</h3>
<p>The nextflow script starts by opening the genotype data.</p>
<div class="kclass">
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode">#<span class="hl opt">!/</span>usr<span class="hl opt">/</span>bin<span class="hl opt">/</span>env nextflow
<span class="hl slc">// git 9.1</span>
<span class="hl slc">// open genotype data</span>
<span class="hl kwa">Channel</span>
    .fromFilePairs<span class="hl opt">(</span><span class="hl str">&quot;../../1_genotyping/4_phased/phased_mac2.vcf.{gz,gz.tbi}&quot;</span><span class="hl opt">)</span>
    .into<span class="hl opt">{</span> vcf_loc1<span class="hl opt">;</span> vcf_loc2<span class="hl opt">;</span> vcf_loc3 <span class="hl opt">}</span>

<span class="hl slc">// git 9.2</span>
<span class="hl slc">// initialize location channel</span>
<span class="hl kwa">Channel</span>
    .from<span class="hl opt">(</span> <span class="hl str">&quot;bel&quot;</span><span class="hl opt">,</span> <span class="hl str">&quot;hon&quot;</span><span class="hl opt">,</span> <span class="hl str">&quot;pan&quot;</span><span class="hl opt">)</span>
    .set<span class="hl opt">{</span> locations_ch <span class="hl opt">}</span>

<span class="hl slc">// git 9.3</span>
<span class="hl slc">// define location specific sepcies set</span>
<span class="hl kwa">Channel</span>.from<span class="hl opt">( [[</span><span class="hl num">1</span><span class="hl opt">,</span> <span class="hl str">&quot;ind&quot;</span><span class="hl opt">], [</span><span class="hl num">2</span><span class="hl opt">,</span> <span class="hl str">&quot;may&quot;</span><span class="hl opt">], [</span><span class="hl num">3</span><span class="hl opt">,</span> <span class="hl str">&quot;nig&quot;</span><span class="hl opt">], [</span><span class="hl num">4</span><span class="hl opt">,</span> <span class="hl str">&quot;pue&quot;</span><span class="hl opt">], [</span><span class="hl num">5</span><span class="hl opt">,</span> <span class="hl str">&quot;uni&quot;</span><span class="hl opt">]] )</span>.into<span class="hl opt">{</span> bel_spec1_ch<span class="hl opt">;</span> bel_spec2_ch <span class="hl opt">}</span>
<span class="hl kwa">Channel</span>.from<span class="hl opt">( [[</span><span class="hl num">1</span><span class="hl opt">,</span> <span class="hl str">&quot;abe&quot;</span><span class="hl opt">], [</span><span class="hl num">2</span><span class="hl opt">,</span> <span class="hl str">&quot;gum&quot;</span><span class="hl opt">], [</span><span class="hl num">3</span><span class="hl opt">,</span> <span class="hl str">&quot;nig&quot;</span><span class="hl opt">], [</span><span class="hl num">4</span><span class="hl opt">,</span> <span class="hl str">&quot;pue&quot;</span><span class="hl opt">], [</span><span class="hl num">5</span><span class="hl opt">,</span> <span class="hl str">&quot;ran&quot;</span><span class="hl opt">], [</span><span class="hl num">6</span><span class="hl opt">,</span> <span class="hl str">&quot;uni&quot;</span><span class="hl opt">]] )</span>.into<span class="hl opt">{</span> hon_spec1_ch<span class="hl opt">;</span> hon_spec2_ch <span class="hl opt">}</span>
<span class="hl kwa">Channel</span>.from<span class="hl opt">( [[</span><span class="hl num">1</span><span class="hl opt">,</span> <span class="hl str">&quot;nig&quot;</span><span class="hl opt">], [</span><span class="hl num">2</span><span class="hl opt">,</span> <span class="hl str">&quot;pue&quot;</span><span class="hl opt">], [</span><span class="hl num">3</span><span class="hl opt">,</span> <span class="hl str">&quot;uni&quot;</span><span class="hl opt">]] )</span>.into<span class="hl opt">{</span> pan_spec1_ch<span class="hl opt">;</span> pan_spec2_ch <span class="hl opt">}</span>

<span class="hl slc">// git 9.4</span>
<span class="hl slc">// prepare pairwise new_hybrids</span>
<span class="hl slc">// ------------------------------</span>
<span class="hl com">/* (create all possible species pairs depending on location</span>
<span class="hl com">   and combine with genotype subset (for the respective location))*/</span>
bel_pairs_ch <span class="hl opt">=</span> <span class="hl kwa">Channel</span>.from<span class="hl opt">(</span> <span class="hl str">&quot;bel&quot;</span> <span class="hl opt">)</span>
    .combine<span class="hl opt">(</span> vcf_loc1 <span class="hl opt">)</span>
    .combine<span class="hl opt">(</span>bel_spec1_ch<span class="hl opt">)</span>
    .combine<span class="hl opt">(</span>bel_spec2_ch<span class="hl opt">)</span>
    .filter<span class="hl opt">{</span> it<span class="hl opt">[</span><span class="hl num">3</span><span class="hl opt">] &lt;</span> it<span class="hl opt">[</span><span class="hl num">5</span><span class="hl opt">] }</span>
    .map<span class="hl opt">{</span> it<span class="hl opt">[</span><span class="hl num">0</span><span class="hl opt">,</span><span class="hl num">1</span><span class="hl opt">,</span><span class="hl num">2</span><span class="hl opt">,</span><span class="hl num">4</span><span class="hl opt">,</span><span class="hl num">6</span><span class="hl opt">]}</span>
hon_pairs_ch <span class="hl opt">=</span> <span class="hl kwa">Channel</span>.from<span class="hl opt">(</span> <span class="hl str">&quot;hon&quot;</span> <span class="hl opt">)</span>
    .combine<span class="hl opt">(</span> vcf_loc2 <span class="hl opt">)</span>
    .combine<span class="hl opt">(</span>hon_spec1_ch<span class="hl opt">)</span>
    .combine<span class="hl opt">(</span>hon_spec2_ch<span class="hl opt">)</span>
    .filter<span class="hl opt">{</span> it<span class="hl opt">[</span><span class="hl num">3</span><span class="hl opt">] &lt;</span> it<span class="hl opt">[</span><span class="hl num">5</span><span class="hl opt">] }</span>
    .map<span class="hl opt">{</span> it<span class="hl opt">[</span><span class="hl num">0</span><span class="hl opt">,</span><span class="hl num">1</span><span class="hl opt">,</span><span class="hl num">2</span><span class="hl opt">,</span><span class="hl num">4</span><span class="hl opt">,</span><span class="hl num">6</span><span class="hl opt">]}</span>
pan_pairs_ch <span class="hl opt">=</span> <span class="hl kwa">Channel</span>.from<span class="hl opt">(</span> <span class="hl str">&quot;pan&quot;</span> <span class="hl opt">)</span>
    .combine<span class="hl opt">(</span> vcf_loc3 <span class="hl opt">)</span>
    .combine<span class="hl opt">(</span>pan_spec1_ch<span class="hl opt">)</span>
    .combine<span class="hl opt">(</span>pan_spec2_ch<span class="hl opt">)</span>
    .filter<span class="hl opt">{</span> it<span class="hl opt">[</span><span class="hl num">3</span><span class="hl opt">] &lt;</span> it<span class="hl opt">[</span><span class="hl num">5</span><span class="hl opt">] }</span>
    .map<span class="hl opt">{</span> it<span class="hl opt">[</span><span class="hl num">0</span><span class="hl opt">,</span><span class="hl num">1</span><span class="hl opt">,</span><span class="hl num">2</span><span class="hl opt">,</span><span class="hl num">4</span><span class="hl opt">,</span><span class="hl num">6</span><span class="hl opt">]}</span>
bel_pairs_ch.concat<span class="hl opt">(</span> hon_pairs_ch<span class="hl opt">,</span> pan_pairs_ch  <span class="hl opt">)</span>.set <span class="hl opt">{</span> all_fst_pairs_ch <span class="hl opt">}</span>

<span class="hl slc">// git 9.5</span>
<span class="hl slc">// comute pairwise fsts for SNP filtering</span>
<span class="hl kwa">process</span> fst_run <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_20g45m_fst_run&#39;</span>
    <span class="hl kwb">tag</span> <span class="hl str">&quot;</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">&quot;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> loc <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> vcfidx <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec1 <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec2 <span class="hl opt">)</span> <span class="hl kwa">from</span> all_fst_pairs_ch

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> loc <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec1 <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec2 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;${vcf[0]}&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*.fst.tsv.gz&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">.pop&quot;</span><span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">.pop&quot;</span><span class="hl opt">)</span> <span class="hl kwa">into</span> fst_SNPS

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   vcfsamplenames ${vcf[0]} | grep</span> <span class="hl ipl">${spec1}${loc}</span> <span class="hl str">&gt;</span> <span class="hl ipl">${spec1}${loc}</span><span class="hl str">.pop</span>
<span class="hl str">   vcfsamplenames ${vcf[0]} | grep</span> <span class="hl ipl">${spec2}${loc}</span> <span class="hl str">&gt;</span> <span class="hl ipl">${spec2}${loc}</span><span class="hl str">.pop</span>
<span class="hl str"></span>
<span class="hl str">   vcftools --gzvcf ${vcf[0]} \</span>
<span class="hl str">        --weir-fst-pop</span> <span class="hl ipl">${spec1}${loc}</span><span class="hl str">.pop \</span>
<span class="hl str">        --weir-fst-pop</span> <span class="hl ipl">${spec2}${loc}</span><span class="hl str">.pop \</span>
<span class="hl str">        --stdout | gzip &gt;</span> <span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">.fst.tsv.gz</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>

<span class="hl slc">// git 9.6</span>
<span class="hl slc">// select the 800 most differentiated SNPs for each population pair</span>
<span class="hl kwa">process</span> filter_fst <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_8g15m_filter_fst&#39;</span>
    <span class="hl kwb">tag</span> <span class="hl str">&quot;</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">&quot;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> loc <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec1 <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec2 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> fst <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> pop1 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> pop2 <span class="hl opt">)</span> <span class="hl kwa">from</span> fst_SNPS

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> loc <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec1 <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec2 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> pop1 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> pop2 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*SNPs.snps&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> filter_SNPs


    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   Rscript --vanilla \$BASE_DIR/R/filter_snps.R</span> <span class="hl ipl">${fst}</span> <span class="hl str">800</span> <span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>

<span class="hl slc">// git 9.7</span>
<span class="hl slc">// filter the SNP set by min distance (5kb), than randomly pick 80 SNPs</span>
<span class="hl slc">// then reformat newhybrid input</span>
<span class="hl kwa">process</span> prep_nh_input <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_8g15m_prep_nh&#39;</span>
    <span class="hl kwb">tag</span> <span class="hl str">&quot;</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">&quot;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> loc <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec1 <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec2 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> pop1 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> pop2 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> snps <span class="hl opt">)</span> <span class="hl kwa">from</span> filter_SNPs


    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> loc <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec1 <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec2 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*_individuals.txt&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*.80SNPs.txt&quot;</span><span class="hl opt">)</span>  <span class="hl kwa">into</span> newhybrids_input


    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   vcftools \</span>
<span class="hl str">  --gzvcf</span> <span class="hl ipl">${vcf}</span> <span class="hl str">\</span>
<span class="hl str">   --keep</span> <span class="hl ipl">${pop1}</span> <span class="hl str">\</span>
<span class="hl str">   --keep</span> <span class="hl ipl">${pop2}</span> <span class="hl str">\</span>
<span class="hl str">   --thin 5000 \</span>
<span class="hl str">   --out newHyb.</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span> <span class="hl str">\</span>
<span class="hl str">   --positions</span> <span class="hl ipl">${snps}</span> <span class="hl str">\</span>
<span class="hl str">   --recode</span>
<span class="hl str"></span>
<span class="hl str">   grep &#39;#&#39; newHyb.</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">.recode.vcf &gt; newHyb.</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">.80SNPs.vcf</span>
<span class="hl str">   grep -v &#39;#&#39; newHyb.</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">.recode.vcf | \</span>
<span class="hl str">       shuf -n 80 | \</span>
<span class="hl str">       sort -k 1 -k2 &gt;&gt; newHyb.</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">.80SNPs.vcf</span>
<span class="hl str"></span>
<span class="hl str">   grep &#39;#CHROM&#39; newHyb.</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">.80SNPs.vcf | \</span>
<span class="hl str">       cut -f 10- | \</span>
<span class="hl str">       sed &#39;s/</span><span class="hl esc">\\</span><span class="hl str">t/</span><span class="hl esc">\\</span><span class="hl str">n/g&#39; &gt; newHyb.</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">.80SNPs_individuals.txt</span>
<span class="hl str"></span>
<span class="hl str">   /usr/bin/java -Xmx1024m -Xms512M \</span>
<span class="hl str">       -jar \$SFTWR/PGDSpider/PGDSpider2-cli.jar \</span>
<span class="hl str">       -inputfile newHyb.</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">.80SNPs.vcf \</span>
<span class="hl str">       -inputformat VCF \</span>
<span class="hl str">       -outputfile newHyb.</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">.80SNPs.txt \</span>
<span class="hl str">       -outputformat NEWHYBRIDS \</span>
<span class="hl str">       -spid \$BASE_DIR/ressources/vcf2nh.spid</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>

<span class="hl slc">// git 9.8</span>
<span class="hl slc">// Run new hybrids</span>
<span class="hl slc">// (copy of nh_input is needed because nh can&#39;t read links)</span>
<span class="hl kwa">process</span> run_nh <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_20g15h4x_run_nh&#39;</span>
    <span class="hl kwb">tag</span> <span class="hl str">&quot;</span><span class="hl ipl">${spec1}${loc}</span><span class="hl str">-</span><span class="hl ipl">${spec2}${loc}</span><span class="hl str">&quot;</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../2_analysis/newhyb/&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> loc <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec1 <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> spec2 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> inds <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> snps <span class="hl opt">)</span> <span class="hl kwa">from</span> newhybrids_input

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;nh_input/NH.Results/newHyb.*/*_individuals.txt&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;nh_input/NH.Results/newHyb.*/*_PofZ.txt&quot;</span> <span class="hl opt">)</span>  <span class="hl kwa">into</span> newhybrids_output

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   mkdir -p nh_input</span>
<span class="hl str">   cp</span> <span class="hl ipl">${snps}</span> <span class="hl str">nh_input/</span><span class="hl ipl">${snps}</span>
<span class="hl str">   cp</span> <span class="hl ipl">${inds}</span> <span class="hl str">nh_input/</span><span class="hl ipl">${inds}</span>
<span class="hl str"></span>
<span class="hl str">   Rscript --vanilla \$BASE_DIR/R/run_newhybrids.R</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
</div>
<p>Finally, we are done with XX.</p>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="git-8-analysis-vi-demographic-history.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="git-10-analysis-viii-admixture.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
