<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>22 Supplementary Figure 5 | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="22 Supplementary Figure 5 | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="22 Supplementary Figure 5 | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench" />


<meta name="date" content="2020-06-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="supplementary-figure-4.html"/>
<link rel="next" href="supplementary-figure-6.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a><ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a><ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a><ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a><ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a><ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a><ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i> &amp; pi)</a><ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a><ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (phylogeny &amp; topology weighting)</a><ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a><ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a><ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a><ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (poptrees)</a><ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#details-of-analysis_fst_poptree.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_fst_poptree.nf</code></a><ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#data-preparation-4"><i class="fa fa-check"></i><b>8.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a><ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a><ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-5"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a><ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a><ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-6"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a><ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a><ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-7"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-visualization.html"><a href="git-11-visualization.html"><i class="fa fa-check"></i><b>12</b> (git 11) Visualization</a></li>
<li class="chapter" data-level="13" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>13</b> Figure 1</a><ul>
<li class="chapter" data-level="13.1" data-path="figure-1.html"><a href="figure-1.html#summary-10"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>13.2</b> Details of <code>plot_F1.R</code></a><ul>
<li class="chapter" data-level="13.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>13.2.1</b> Config</a></li>
<li class="chapter" data-level="13.2.2" data-path="figure-1.html"><a href="figure-1.html#data-import"><i class="fa fa-check"></i><b>13.2.2</b> Data import</a></li>
<li class="chapter" data-level="13.2.3" data-path="figure-1.html"><a href="figure-1.html#plotting"><i class="fa fa-check"></i><b>13.2.3</b> Plotting</a></li>
<li class="chapter" data-level="13.2.4" data-path="figure-1.html"><a href="figure-1.html#table-export"><i class="fa fa-check"></i><b>13.2.4</b> Table export</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>14</b> Figure 2</a><ul>
<li class="chapter" data-level="14.1" data-path="figure-2.html"><a href="figure-2.html#summary-11"><i class="fa fa-check"></i><b>14.1</b> Summary</a></li>
<li class="chapter" data-level="14.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>14.2</b> Details of <code>plot_F2.R</code></a><ul>
<li class="chapter" data-level="14.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>14.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>15</b> Figure 3</a><ul>
<li class="chapter" data-level="15.1" data-path="figure-3.html"><a href="figure-3.html#summary-12"><i class="fa fa-check"></i><b>15.1</b> Summary</a></li>
<li class="chapter" data-level="15.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>15.2</b> Details of <code>plot_F3.R</code></a><ul>
<li class="chapter" data-level="15.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>15.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>16</b> Figure 4</a><ul>
<li class="chapter" data-level="16.1" data-path="figure-4.html"><a href="figure-4.html#summary-13"><i class="fa fa-check"></i><b>16.1</b> Summary</a></li>
<li class="chapter" data-level="16.2" data-path="figure-4.html"><a href="figure-4.html#details-of-plot_f4.r"><i class="fa fa-check"></i><b>16.2</b> Details of <code>plot_F4.R</code></a><ul>
<li class="chapter" data-level="16.2.1" data-path="figure-4.html"><a href="figure-4.html#config-3"><i class="fa fa-check"></i><b>16.2.1</b> Config</a></li>
<li class="chapter" data-level="16.2.2" data-path="figure-4.html"><a href="figure-4.html#data-import-1"><i class="fa fa-check"></i><b>16.2.2</b> Data import</a></li>
<li class="chapter" data-level="16.2.3" data-path="figure-4.html"><a href="figure-4.html#plotting-1"><i class="fa fa-check"></i><b>16.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>17</b> Figure 5</a><ul>
<li class="chapter" data-level="17.1" data-path="figure-5.html"><a href="figure-5.html#summary-14"><i class="fa fa-check"></i><b>17.1</b> Summary</a></li>
<li class="chapter" data-level="17.2" data-path="figure-5.html"><a href="figure-5.html#details-of-plot_f5.r"><i class="fa fa-check"></i><b>17.2</b> Details of <code>plot_F5.R</code></a><ul>
<li class="chapter" data-level="17.2.1" data-path="figure-5.html"><a href="figure-5.html#config-4"><i class="fa fa-check"></i><b>17.2.1</b> Config</a></li>
<li class="chapter" data-level="17.2.2" data-path="figure-5.html"><a href="figure-5.html#data-import-2"><i class="fa fa-check"></i><b>17.2.2</b> Data import</a></li>
<li class="chapter" data-level="17.2.3" data-path="figure-5.html"><a href="figure-5.html#plotting-2"><i class="fa fa-check"></i><b>17.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html"><i class="fa fa-check"></i><b>18</b> Supplementary Figure 1</a><ul>
<li class="chapter" data-level="18.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#summary-15"><i class="fa fa-check"></i><b>18.1</b> Summary</a></li>
<li class="chapter" data-level="18.2" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#details-of-plot_sf1.r"><i class="fa fa-check"></i><b>18.2</b> Details of <code>plot_SF1.R</code></a><ul>
<li class="chapter" data-level="18.2.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#config-5"><i class="fa fa-check"></i><b>18.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html"><i class="fa fa-check"></i><b>19</b> Supplementary Figure 2</a><ul>
<li class="chapter" data-level="19.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#summary-16"><i class="fa fa-check"></i><b>19.1</b> Summary</a></li>
<li class="chapter" data-level="19.2" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#details-of-plot_sf2.r"><i class="fa fa-check"></i><b>19.2</b> Details of <code>plot_SF2.R</code></a><ul>
<li class="chapter" data-level="19.2.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#config-6"><i class="fa fa-check"></i><b>19.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html"><i class="fa fa-check"></i><b>20</b> Supplementary Figure 3</a><ul>
<li class="chapter" data-level="20.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#summary-17"><i class="fa fa-check"></i><b>20.1</b> Summary</a></li>
<li class="chapter" data-level="20.2" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#details-of-plot_sf3.r"><i class="fa fa-check"></i><b>20.2</b> Details of <code>plot_SF3.R</code></a><ul>
<li class="chapter" data-level="20.2.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#config-7"><i class="fa fa-check"></i><b>20.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html"><i class="fa fa-check"></i><b>21</b> Supplementary Figure 4</a><ul>
<li class="chapter" data-level="21.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#summary-18"><i class="fa fa-check"></i><b>21.1</b> Summary</a></li>
<li class="chapter" data-level="21.2" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#details-of-plot_sf4.r"><i class="fa fa-check"></i><b>21.2</b> Details of <code>plot_SF4.R</code></a><ul>
<li class="chapter" data-level="21.2.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#config-8"><i class="fa fa-check"></i><b>21.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html"><i class="fa fa-check"></i><b>22</b> Supplementary Figure 5</a><ul>
<li class="chapter" data-level="22.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#summary-19"><i class="fa fa-check"></i><b>22.1</b> Summary</a></li>
<li class="chapter" data-level="22.2" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#details-of-plot_sf5.r"><i class="fa fa-check"></i><b>22.2</b> Details of <code>plot_SF5.R</code></a><ul>
<li class="chapter" data-level="22.2.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#config-9"><i class="fa fa-check"></i><b>22.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html"><i class="fa fa-check"></i><b>23</b> Supplementary Figure 6</a><ul>
<li class="chapter" data-level="23.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#summary-20"><i class="fa fa-check"></i><b>23.1</b> Summary</a></li>
<li class="chapter" data-level="23.2" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#details-of-plot_sf6.r"><i class="fa fa-check"></i><b>23.2</b> Details of <code>plot_SF6.R</code></a><ul>
<li class="chapter" data-level="23.2.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#config-10"><i class="fa fa-check"></i><b>23.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html"><i class="fa fa-check"></i><b>24</b> Supplementary Figure 7</a><ul>
<li class="chapter" data-level="24.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#summary-21"><i class="fa fa-check"></i><b>24.1</b> Summary</a></li>
<li class="chapter" data-level="24.2" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#details-of-plot_sf7.r"><i class="fa fa-check"></i><b>24.2</b> Details of <code>plot_SF7.R</code></a><ul>
<li class="chapter" data-level="24.2.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#config-11"><i class="fa fa-check"></i><b>24.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html"><i class="fa fa-check"></i><b>25</b> Supplementary Figure 8</a><ul>
<li class="chapter" data-level="25.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#summary-22"><i class="fa fa-check"></i><b>25.1</b> Summary</a></li>
<li class="chapter" data-level="25.2" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#details-of-plot_sf8.r"><i class="fa fa-check"></i><b>25.2</b> Details of <code>plot_SF8.R</code></a><ul>
<li class="chapter" data-level="25.2.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#config-12"><i class="fa fa-check"></i><b>25.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html"><i class="fa fa-check"></i><b>26</b> Supplementary Figure 9</a><ul>
<li class="chapter" data-level="26.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#summary-23"><i class="fa fa-check"></i><b>26.1</b> Summary</a></li>
<li class="chapter" data-level="26.2" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#details-of-plot_sf9.r"><i class="fa fa-check"></i><b>26.2</b> Details of <code>plot_SF9.R</code></a><ul>
<li class="chapter" data-level="26.2.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#config-13"><i class="fa fa-check"></i><b>26.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="supplementary-figure-5" class="section level1">
<h1><span class="header-section-number">22</span> Supplementary Figure 5</h1>
<div id="summary-19" class="section level2">
<h2><span class="header-section-number">22.1</span> Summary</h2>
<p>This is the accessory documentation of Supplementary Figure 5.
The Figure can be recreated by running the <strong>R</strong> script <code>plot_SF5.R</code>:</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb152-1"><a href="supplementary-figure-5.html#cb152-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span></span>
<span id="cb152-2"><a href="supplementary-figure-5.html#cb152-2"></a></span>
<span id="cb152-3"><a href="supplementary-figure-5.html#cb152-3"></a><span class="ex">Rscript</span> --vanilla R/fig/plot_SF5.R 2_analysis/pi/50k/ \</span>
<span id="cb152-4"><a href="supplementary-figure-5.html#cb152-4"></a>  2_analysis/fasteprr/step4/fasteprr.all.rho.txt.gz</span></code></pre></div>
</div>
<div id="details-of-plot_sf5.r" class="section level2">
<h2><span class="header-section-number">22.2</span> Details of <code>plot_SF5.R</code></h2>
<p>In the following, the individual steps of the R script are documented.
It is an executable R script that depends on the accessory R package <a href="https://k-hench.github.io/GenomicOriginsScripts"><strong>GenomicOriginsScripts</strong></a> as well as the packages <a href="https://k-hench.github.io/hypoimg"><strong>hypoimg</strong></a> and <a href="https://vroom.r-lib.org/"><strong>vroom</strong></a>.</p>
<div id="config-9" class="section level3">
<h3><span class="header-section-number">22.2.1</span> Config</h3>
<p>The scripts start with a header that contains copy &amp; paste templates to execute or debug the script:</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="supplementary-figure-5.html#cb153-1"></a><span class="co">#!/usr/bin/env Rscript</span></span>
<span id="cb153-2"><a href="supplementary-figure-5.html#cb153-2"></a><span class="co"># run from terminal:</span></span>
<span id="cb153-3"><a href="supplementary-figure-5.html#cb153-3"></a><span class="co"># Rscript --vanilla R/fig/plot_SF5.R 2_analysis/pi/50k/ \</span></span>
<span id="cb153-4"><a href="supplementary-figure-5.html#cb153-4"></a><span class="co">#   2_analysis/fasteprr/step4/fasteprr.all.rho.txt.gz</span></span>
<span id="cb153-5"><a href="supplementary-figure-5.html#cb153-5"></a><span class="co"># ===============================================================</span></span>
<span id="cb153-6"><a href="supplementary-figure-5.html#cb153-6"></a><span class="co"># This script produces Suppl. Figure 5 of the study &quot;Ancestral variation,</span></span>
<span id="cb153-7"><a href="supplementary-figure-5.html#cb153-7"></a><span class="co"># hybridization and modularity fuel a marine radiation&quot;</span></span>
<span id="cb153-8"><a href="supplementary-figure-5.html#cb153-8"></a><span class="co"># by Hench, McMillan and Puebla</span></span>
<span id="cb153-9"><a href="supplementary-figure-5.html#cb153-9"></a><span class="co"># ---------------------------------------------------------------</span></span>
<span id="cb153-10"><a href="supplementary-figure-5.html#cb153-10"></a><span class="co"># ===============================================================</span></span>
<span id="cb153-11"><a href="supplementary-figure-5.html#cb153-11"></a><span class="co"># args &lt;- c(&#39;2_analysis/pi/50k/&#39;,</span></span>
<span id="cb153-12"><a href="supplementary-figure-5.html#cb153-12"></a><span class="co">#           &#39;2_analysis/fasteprr/step4/fasteprr.all.rho.txt.gz&#39;)</span></span>
<span id="cb153-13"><a href="supplementary-figure-5.html#cb153-13"></a><span class="co"># script_name &lt;- &quot;R/fig/plot_SF5.R&quot;</span></span></code></pre></div>
<p>The next section processes the input from the command line.
It stores the arguments in the vector <code>args</code>.
The needed R packages are loaded and the script name and the current working directory are stored inside variables (<code>script_name</code>, <code>plot_comment</code>).
This information will later be written into the meta data of the figure to help us tracing back the scripts that created the figures in the future.</p>
<p>Then we drop all the imported information besides the arguments following the script name and print the information to the terminal.</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="supplementary-figure-5.html#cb154-1"></a>args &lt;-<span class="st"> </span><span class="kw">commandArgs</span>(<span class="dt">trailingOnly=</span><span class="ot">FALSE</span>)</span>
<span id="cb154-2"><a href="supplementary-figure-5.html#cb154-2"></a><span class="co"># setup -----------------------</span></span>
<span id="cb154-3"><a href="supplementary-figure-5.html#cb154-3"></a><span class="kw">library</span>(GenomicOriginsScripts)</span>
<span id="cb154-4"><a href="supplementary-figure-5.html#cb154-4"></a><span class="kw">library</span>(vroom)</span>
<span id="cb154-5"><a href="supplementary-figure-5.html#cb154-5"></a><span class="kw">library</span>(hypoimg)</span>
<span id="cb154-6"><a href="supplementary-figure-5.html#cb154-6"></a></span>
<span id="cb154-7"><a href="supplementary-figure-5.html#cb154-7"></a><span class="kw">cat</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>)</span>
<span id="cb154-8"><a href="supplementary-figure-5.html#cb154-8"></a>script_name &lt;-<span class="st"> </span>args[<span class="dv">5</span>] <span class="op">%&gt;%</span></span>
<span id="cb154-9"><a href="supplementary-figure-5.html#cb154-9"></a><span class="st">  </span><span class="kw">str_remove</span>(.,<span class="st">&#39;--file=&#39;</span>)</span>
<span id="cb154-10"><a href="supplementary-figure-5.html#cb154-10"></a></span>
<span id="cb154-11"><a href="supplementary-figure-5.html#cb154-11"></a>plot_comment &lt;-<span class="st"> </span>script_name <span class="op">%&gt;%</span></span>
<span id="cb154-12"><a href="supplementary-figure-5.html#cb154-12"></a><span class="st">  </span><span class="kw">str_c</span>(<span class="st">&#39;mother-script = &#39;</span>,<span class="kw">getwd</span>(),<span class="st">&#39;/&#39;</span>,.)</span>
<span id="cb154-13"><a href="supplementary-figure-5.html#cb154-13"></a></span>
<span id="cb154-14"><a href="supplementary-figure-5.html#cb154-14"></a>args &lt;-<span class="st"> </span><span class="kw">process_input</span>(script_name, args)</span></code></pre></div>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="supplementary-figure-5.html#cb155-1"></a><span class="co">#&gt; ── Script: scripts/plot_SF5.R ────────────────────────────────────────────</span></span>
<span id="cb155-2"><a href="supplementary-figure-5.html#cb155-2"></a><span class="co">#&gt; Parameters read:</span></span>
<span id="cb155-3"><a href="supplementary-figure-5.html#cb155-3"></a><span class="co">#&gt; ★ 1: 2_analysis/pi/50k/</span></span>
<span id="cb155-4"><a href="supplementary-figure-5.html#cb155-4"></a><span class="co">#&gt; ★ 2: 2_analysis/fasteprr/step4/fasteprr.all.rho.txt.gz</span></span>
<span id="cb155-5"><a href="supplementary-figure-5.html#cb155-5"></a><span class="co">#&gt; ─────────────────────────────────────────── /current/working/directory ──</span></span></code></pre></div>
<p>The paths containing the <span class="math inline">\(\pi\)</span> and <span class="math inline">\(\rho\)</span> data are received and stored inside more descriptive variables.</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="supplementary-figure-5.html#cb156-1"></a><span class="co"># config -----------------------</span></span>
<span id="cb156-2"><a href="supplementary-figure-5.html#cb156-2"></a>pi_path &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">1</span>])</span>
<span id="cb156-3"><a href="supplementary-figure-5.html#cb156-3"></a>rho_path &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">2</span>])</span></code></pre></div>
<p>The <span class="math inline">\(\pi\)</span>-path is screened for data files in the desired resolution (50 kb windows).</p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="supplementary-figure-5.html#cb157-1"></a><span class="co"># locate pi data files</span></span>
<span id="cb157-2"><a href="supplementary-figure-5.html#cb157-2"></a>files &lt;-<span class="st"> </span><span class="kw">dir</span>(pi_path, <span class="dt">pattern =</span> <span class="st">&#39;^[a-z]{6}.50k&#39;</span>)</span></code></pre></div>
<p>Then, the <span class="math inline">\(\pi\)</span> data is loaded and compiled into a single table containing all populations.</p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="supplementary-figure-5.html#cb158-1"></a><span class="co"># load pi data</span></span>
<span id="cb158-2"><a href="supplementary-figure-5.html#cb158-2"></a>data &lt;-<span class="st"> </span><span class="kw">str_c</span>(pi_path, files) <span class="op">%&gt;%</span></span>
<span id="cb158-3"><a href="supplementary-figure-5.html#cb158-3"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">map</span>(get_pi) <span class="op">%&gt;%</span></span>
<span id="cb158-4"><a href="supplementary-figure-5.html#cb158-4"></a><span class="st">  </span><span class="kw">bind_rows</span>()</span></code></pre></div>
<p>To be able to order the subplots of the final figure by the genome wide average <span class="math inline">\(\pi\)</span> of the populations, we create a second dta table containing the summary of the <span class="math inline">\(\pi\)</span> data for all populations.</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="supplementary-figure-5.html#cb159-1"></a><span class="co"># compute genome wide average pi for the subplot order</span></span>
<span id="cb159-2"><a href="supplementary-figure-5.html#cb159-2"></a>global_bar &lt;-<span class="st"> </span>data <span class="op">%&gt;%</span></span>
<span id="cb159-3"><a href="supplementary-figure-5.html#cb159-3"></a><span class="st">  </span><span class="kw">filter</span>( BIN_START <span class="op">%%</span><span class="st"> </span><span class="dv">50000</span> <span class="op">==</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span></span>
<span id="cb159-4"><a href="supplementary-figure-5.html#cb159-4"></a><span class="st">  </span><span class="kw">select</span>(N_VARIANTS, PI, spec) <span class="op">%&gt;%</span></span>
<span id="cb159-5"><a href="supplementary-figure-5.html#cb159-5"></a><span class="st">  </span><span class="kw">group_by</span>(spec) <span class="op">%&gt;%</span></span>
<span id="cb159-6"><a href="supplementary-figure-5.html#cb159-6"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">genome_wide_pi =</span> <span class="kw">sum</span>(N_VARIANTS<span class="op">*</span>PI)<span class="op">/</span><span class="kw">sum</span>(N_VARIANTS)) <span class="op">%&gt;%</span></span>
<span id="cb159-7"><a href="supplementary-figure-5.html#cb159-7"></a><span class="st">  </span><span class="kw">arrange</span>(genome_wide_pi) <span class="op">%&gt;%</span></span>
<span id="cb159-8"><a href="supplementary-figure-5.html#cb159-8"></a><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></span>
<span id="cb159-9"><a href="supplementary-figure-5.html#cb159-9"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">spec =</span> <span class="kw">fct_reorder</span>(<span class="dt">.f =</span> spec, <span class="dt">.x =</span> genome_wide_pi),</span>
<span id="cb159-10"><a href="supplementary-figure-5.html#cb159-10"></a>         <span class="dt">scaled_pi =</span> genome_wide_pi<span class="op">/</span><span class="kw">max</span>(genome_wide_pi))</span></code></pre></div>
<p>Then, we load the <span class="math inline">\(\rho\)</span> data.</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="supplementary-figure-5.html#cb160-1"></a><span class="co"># load recombination data</span></span>
<span id="cb160-2"><a href="supplementary-figure-5.html#cb160-2"></a>rho_data &lt;-<span class="st"> </span><span class="kw">vroom</span>(rho_path, <span class="dt">delim =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb160-3"><a href="supplementary-figure-5.html#cb160-3"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>BIN_END)</span></code></pre></div>
<p>To be able to regress <span class="math inline">\(\pi\)</span> on <span class="math inline">\(\rho\)</span> the two data sets are merged based on the window position on the hamlet reference genome.</p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="supplementary-figure-5.html#cb161-1"></a><span class="co"># merge pi and recombination data</span></span>
<span id="cb161-2"><a href="supplementary-figure-5.html#cb161-2"></a>combined_data &lt;-<span class="st"> </span>data <span class="op">%&gt;%</span></span>
<span id="cb161-3"><a href="supplementary-figure-5.html#cb161-3"></a><span class="st">  </span><span class="co"># filter pi data to &quot;non-overlapping&quot; windows</span></span>
<span id="cb161-4"><a href="supplementary-figure-5.html#cb161-4"></a><span class="st">  </span><span class="kw">filter</span>(BIN_START <span class="op">%%</span><span class="st"> </span><span class="dv">50000</span> <span class="op">==</span><span class="st"> </span><span class="dv">1</span> ) <span class="op">%&gt;%</span></span>
<span id="cb161-5"><a href="supplementary-figure-5.html#cb161-5"></a><span class="st">  </span><span class="co"># reorder populations by genome wide average pi</span></span>
<span id="cb161-6"><a href="supplementary-figure-5.html#cb161-6"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">spec =</span> <span class="kw">factor</span>(spec, <span class="dt">levels =</span> <span class="kw">levels</span>(global_bar<span class="op">$</span>spec))) <span class="op">%&gt;%</span></span>
<span id="cb161-7"><a href="supplementary-figure-5.html#cb161-7"></a><span class="st">  </span><span class="co"># merge with recombination data</span></span>
<span id="cb161-8"><a href="supplementary-figure-5.html#cb161-8"></a><span class="st">  </span><span class="kw">left_join</span>(rho_data, <span class="dt">by =</span> <span class="kw">c</span>(<span class="dt">CHROM =</span> <span class="st">&#39;CHROM&#39;</span>, <span class="dt">BIN_START =</span> <span class="st">&#39;BIN_START&#39;</span>))</span></code></pre></div>
<p>Then the data set is nested and for each hamlet population a linear regression of the dependence of <span class="math inline">\(\pi\)</span> on <span class="math inline">\(\rho\)</span> is modeled.
Then the model statistics are stored in additionnal columns.</p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="supplementary-figure-5.html#cb162-1"></a><span class="co"># nest data to run linear regression on all runs in one go</span></span>
<span id="cb162-2"><a href="supplementary-figure-5.html#cb162-2"></a>model_data &lt;-<span class="st"> </span>combined_data <span class="op">%&gt;%</span></span>
<span id="cb162-3"><a href="supplementary-figure-5.html#cb162-3"></a><span class="st">  </span><span class="kw">group_by</span>(spec) <span class="op">%&gt;%</span></span>
<span id="cb162-4"><a href="supplementary-figure-5.html#cb162-4"></a><span class="st">  </span><span class="kw">nest</span>() <span class="op">%&gt;%</span></span>
<span id="cb162-5"><a href="supplementary-figure-5.html#cb162-5"></a><span class="st">  </span><span class="kw">left_join</span>(., global_bar) <span class="op">%&gt;%</span></span>
<span id="cb162-6"><a href="supplementary-figure-5.html#cb162-6"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">mod =</span>  <span class="kw">map</span>(data, <span class="op">~</span><span class="st"> </span><span class="kw">lm</span>(.<span class="op">$</span>PI <span class="op">~</span><span class="st"> </span>.<span class="op">$</span>RHO))) <span class="op">%&gt;%</span></span>
<span id="cb162-7"><a href="supplementary-figure-5.html#cb162-7"></a><span class="st">  </span><span class="kw">bind_cols</span>(., <span class="kw">summarise_model</span>(.))</span></code></pre></div>
<p>To indicate the hamlet population on the sub-plots, hamlet illustrations are loaded.</p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="supplementary-figure-5.html#cb163-1"></a><span class="co"># create table with fish annotations</span></span>
<span id="cb163-2"><a href="supplementary-figure-5.html#cb163-2"></a>grob_tibble2 &lt;-<span class="st"> </span>global_bar<span class="op">$</span>spec <span class="op">%&gt;%</span></span>
<span id="cb163-3"><a href="supplementary-figure-5.html#cb163-3"></a><span class="st">  </span>purrr<span class="op">::</span><span class="kw">map</span>(fish_plot2) <span class="op">%&gt;%</span></span>
<span id="cb163-4"><a href="supplementary-figure-5.html#cb163-4"></a><span class="st">  </span><span class="kw">bind_rows</span>()</span></code></pre></div>
<p>Then, the final figure is created.</p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="supplementary-figure-5.html#cb164-1"></a><span class="co"># compose final figure</span></span>
<span id="cb164-2"><a href="supplementary-figure-5.html#cb164-2"></a>p &lt;-<span class="st"> </span>combined_data <span class="op">%&gt;%</span></span>
<span id="cb164-3"><a href="supplementary-figure-5.html#cb164-3"></a><span class="st">  </span><span class="kw">ggplot</span>()<span class="op">+</span></span>
<span id="cb164-4"><a href="supplementary-figure-5.html#cb164-4"></a><span class="st">  </span><span class="co"># add fish annotations</span></span>
<span id="cb164-5"><a href="supplementary-figure-5.html#cb164-5"></a><span class="st">  </span><span class="kw">geom_hypo_grob2</span>(<span class="dt">data =</span> grob_tibble2,</span>
<span id="cb164-6"><a href="supplementary-figure-5.html#cb164-6"></a>                  <span class="kw">aes</span>(<span class="dt">grob =</span> grob, <span class="dt">rel_x =</span> <span class="fl">.25</span>,<span class="dt">rel_y =</span> <span class="fl">.75</span>),</span>
<span id="cb164-7"><a href="supplementary-figure-5.html#cb164-7"></a>                  <span class="dt">angle =</span> <span class="dv">0</span>, <span class="dt">height =</span> <span class="fl">.5</span>,<span class="dt">width =</span> <span class="fl">.5</span>)<span class="op">+</span></span>
<span id="cb164-8"><a href="supplementary-figure-5.html#cb164-8"></a><span class="st">  </span><span class="co"># add hex-bin desity layer</span></span>
<span id="cb164-9"><a href="supplementary-figure-5.html#cb164-9"></a><span class="st">  </span><span class="kw">geom_hex</span>(<span class="dt">bins =</span> <span class="dv">30</span>,<span class="dt">color =</span> <span class="kw">rgb</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,.<span class="dv">3</span>),</span>
<span id="cb164-10"><a href="supplementary-figure-5.html#cb164-10"></a>           <span class="kw">aes</span>(<span class="dt">fill=</span><span class="kw">log10</span>(..count..), <span class="dt">x =</span> RHO, <span class="dt">y =</span> PI))<span class="op">+</span></span>
<span id="cb164-11"><a href="supplementary-figure-5.html#cb164-11"></a><span class="st"> </span><span class="co"># general plot structure (separated by run)</span></span>
<span id="cb164-12"><a href="supplementary-figure-5.html#cb164-12"></a><span class="st">  </span><span class="kw">facet_wrap</span>(spec <span class="op">~</span>., <span class="dt">ncol =</span> <span class="dv">3</span>)<span class="op">+</span></span>
<span id="cb164-13"><a href="supplementary-figure-5.html#cb164-13"></a><span class="st">  </span><span class="co"># set axis layout and color scheme</span></span>
<span id="cb164-14"><a href="supplementary-figure-5.html#cb164-14"></a><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">name =</span> <span class="kw">expression</span>(rho))<span class="op">+</span></span>
<span id="cb164-15"><a href="supplementary-figure-5.html#cb164-15"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">name =</span> <span class="kw">expression</span>(pi))<span class="op">+</span></span>
<span id="cb164-16"><a href="supplementary-figure-5.html#cb164-16"></a><span class="st">  </span>scico<span class="op">::</span><span class="kw">scale_fill_scico</span>(<span class="dt">palette =</span> <span class="st">&#39;berlin&#39;</span>) <span class="op">+</span></span>
<span id="cb164-17"><a href="supplementary-figure-5.html#cb164-17"></a><span class="st">  </span><span class="co"># customize legend</span></span>
<span id="cb164-18"><a href="supplementary-figure-5.html#cb164-18"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_colorbar</span>(<span class="dt">direction =</span> <span class="st">&#39;horizontal&#39;</span>,</span>
<span id="cb164-19"><a href="supplementary-figure-5.html#cb164-19"></a>                               <span class="dt">title.position =</span> <span class="st">&#39;top&#39;</span>,</span>
<span id="cb164-20"><a href="supplementary-figure-5.html#cb164-20"></a>                               <span class="dt">barheight =</span> <span class="kw">unit</span>(<span class="dv">7</span>,<span class="st">&#39;pt&#39;</span>),</span>
<span id="cb164-21"><a href="supplementary-figure-5.html#cb164-21"></a>                               <span class="dt">barwidth =</span> <span class="kw">unit</span>(<span class="dv">130</span>,<span class="st">&#39;pt&#39;</span>)))<span class="op">+</span></span>
<span id="cb164-22"><a href="supplementary-figure-5.html#cb164-22"></a><span class="st">  </span><span class="co"># general plot layout</span></span>
<span id="cb164-23"><a href="supplementary-figure-5.html#cb164-23"></a><span class="st">  </span><span class="kw">theme_minimal</span>()<span class="op">+</span></span>
<span id="cb164-24"><a href="supplementary-figure-5.html#cb164-24"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(.<span class="dv">84</span>,.<span class="dv">01</span>),</span>
<span id="cb164-25"><a href="supplementary-figure-5.html#cb164-25"></a>        <span class="dt">strip.text =</span> <span class="kw">element_blank</span>())</span></code></pre></div>
<center>
<img src="plot_SF5_files/figure-html/unnamed-chunk-12-1.png" width="768" />
</center>
<p>Finally, we can export Supplementary Figure 5.</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="supplementary-figure-5.html#cb165-1"></a><span class="co"># export final figure</span></span>
<span id="cb165-2"><a href="supplementary-figure-5.html#cb165-2"></a><span class="kw">hypo_save</span>(<span class="dt">filename =</span> <span class="st">&#39;figures/SF5.pdf&#39;</span>,</span>
<span id="cb165-3"><a href="supplementary-figure-5.html#cb165-3"></a>          <span class="dt">plot =</span> p,</span>
<span id="cb165-4"><a href="supplementary-figure-5.html#cb165-4"></a>          <span class="dt">width =</span> <span class="dv">8</span>,</span>
<span id="cb165-5"><a href="supplementary-figure-5.html#cb165-5"></a>          <span class="dt">height =</span> <span class="dv">10</span>,</span>
<span id="cb165-6"><a href="supplementary-figure-5.html#cb165-6"></a>          <span class="dt">comment =</span> plot_comment)</span></code></pre></div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="supplementary-figure-4.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="supplementary-figure-6.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
